--- kdelibs-trinity-3.5.13.2/kdecore/klibloader.cpp~	2014-05-27 23:28:54.291178525 +0300
+++ kdelibs-trinity-3.5.13.2/kdecore/klibloader.cpp	2014-05-27 23:30:56.593147558 +0300
@@ -25,6 +25,7 @@
 #include <tqtimer.h>
 #include <tqobjectdict.h>
 
+#define BOOST_FILESYSTEM_VERSION 3
 #include <boost/filesystem/exception.hpp>
 #include <boost/filesystem/operations.hpp>
 #include <boost/filesystem/path.hpp>
@@ -350,7 +350,7 @@
 			i != boost::filesystem::directory_iterator(); ++i )
 		{
 			boost::smatch m;
-			if ( boost::regex_match( i->string(), m, re ) )
+			if ( boost::regex_match( i->path().string(), m, re ) )
 				return m.str().c_str();
 		}
 	}
