--- kdelibs/kdefx/kstyle.cpp~	2011-01-05 00:23:08.000000000 +0200
+++ kdelibs/kdefx/kstyle.cpp	2011-01-05 00:23:43.787007235 +0200
@@ -2265,7 +2265,14 @@
 
 // HACK for gtk-qt-engine
 
-KDE_EXPORT extern "C"
+/* FIXME temp hack to get it compiled
+kstyle.cpp: At global scope:
+kstyle.cpp:2268:19: error: expected unqualified-id before string constant
+make[2]: *** [kstyle.lo] Error 1
+
+// KDE_EXPORT
+*/ 
+extern "C"
 void kde_kstyle_set_scrollbar_type_windows( void* style )
 {
     ((KStyle*)style)->setScrollBarType( KStyle::WindowsStyleScrollBar );
#--- kdelibs/networkstatus/serviceiface_stub.cpp~	2011-01-05 01:24:06.000000000 +0200
#+++ kdelibs/networkstatus/serviceiface_stub.cpp	2011-01-05 01:24:38.660091691 +0200
#@@ -55,7 +55,11 @@
#     TQCString replyType;
#     TQDataStream arg( data, IO_WriteOnly );
#     arg << arg0;
#-    arg << arg1;
#+    // FIXME temp hack to get it compiled
#+    // serviceiface_stub.o: In function `ServiceIface_stub::registerNetwork(QString const&, NetworkStatus::Properties)':
#+    // serviceiface_stub.cpp:58: undefined reference to `operator<<(QDataStream&, NetworkStatus::Properties)'
#+    // collect2: ld returned 1 exit status
#+    arg << arg1.status;
#     if ( dcopClient()->call( app(), obj(), "registerNetwork(TQString,NetworkStatus::Properties)", data, replyType, replyData ) ) {
# 	setStatus( CallSucceeded );
#     } else { 
--- kdelibs/networkstatus/networkstatuscommon.cpp~	2010-09-04 21:08:12.000000000 +0300
+++ kdelibs/networkstatus/networkstatuscommon.cpp	2011-01-05 01:44:47.862346338 +0200
@@ -22,7 +22,7 @@
 #include "networkstatuscommon.h"
 #include <kdebug.h>
 
-TQDataStream & operator<< ( TQDataStream & s, const NetworkStatus::Properties p )
+TQDataStream & operator<< ( TQDataStream & s, NetworkStatus::Properties p )
 {
 	kdDebug() << k_funcinfo << "status is: " << (int)p.status << endl;
 	s << (int)p.status;
--- kdelibs/networkstatus/networkstatuscommon.h~	2010-09-04 21:08:12.000000000 +0300
+++ kdelibs/networkstatus/networkstatuscommon.h	2011-01-05 01:44:57.519989113 +0200
@@ -49,6 +49,6 @@
 }
 
 TQDataStream & operator>> ( TQDataStream & s, NetworkStatus::Properties &p );
-TQDataStream & operator<< ( TQDataStream & s, const NetworkStatus::Properties p );
+TQDataStream & operator<< ( TQDataStream & s, NetworkStatus::Properties p );
 
 #endif
