--- kde-common/admin/cvs.sh.orig	2003-12-18 12:51:14.000000000 +0100
+++ kde-common/admin/cvs.sh	2003-12-18 12:52:01.000000000 +0100
@@ -65,7 +65,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* ) : ;;
+  automake*1.6.* | automake*1.7* | automake*1.8* ) : ;;
   "" )
     echo "*** AUTOMAKE NOT FOUND!."
     echo "*** KDE requires automake $required_automake_version"
