diff -urN kdelibs-3.5.4.org/dnssd/remoteservice.cpp kdelibs-3.5.4/dnssd/remoteservice.cpp
--- kdelibs-3.5.4.org/dnssd/remoteservice.cpp	2006-01-19 17:07:20.000000000 +0000
+++ kdelibs-3.5.4/dnssd/remoteservice.cpp	2006-09-09 18:24:23.850607000 +0000
@@ -102,7 +102,7 @@
 #ifdef HAVE_DNSSD
 	DNSServiceRef ref;
 	if (DNSServiceResolve(&ref,0,0,m_serviceName.utf8(), m_type.ascii(), 
-		domainToDNS(m_domain),resolve_callback,reinterpret_cast<void*>(this))
+		domainToDNS(m_domain),(DNSServiceResolveReply)resolve_callback,reinterpret_cast<void*>(this))
 		== kDNSServiceErr_NoError) d->setRef(ref);
 #endif
 	if (!d->isRunning()) emit resolved(false);
