diff -urN -x CVS kdelibs.orig/kate/Makefile.am kdelibs/kate/Makefile.am
--- kdelibs.orig/kate/Makefile.am	Tue Feb 24 20:14:48 2004
+++ kdelibs/kate/Makefile.am	Fri Aug 20 23:47:59 2004
@@ -4,7 +4,7 @@
 messages:
 	$(EXTRACTRC) ./*/*.rc >> ./rc.cpp
 	perl -n kate-extract-i18n-xml.pl data/*.xml >> rc.cpp
-	$(XGETTEXT) `find . -name "*.cpp"` -o $(podir)/katepart.pot
+	$(XGETTEXT) `find . -name "*.cpp"` part/*.h -o $(podir)/katepart.pot
 
 DOXYGEN_REFERENCES = kdecore dcop kio kdeui kparts
 include ../admin/Doxyfile.am
diff -urN -x CVS kdelibs.orig/kate/part/katedocument.h kdelibs/kate/part/katedocument.h
--- kdelibs.orig/kate/part/katedocument.h	Sun May 30 10:53:11 2004
+++ kdelibs/kate/part/katedocument.h	Fri Aug 20 23:46:10 2004
@@ -847,7 +847,7 @@
       else if ( m_modOnHdReason == 3 )
         reason = i18n("deleted");
 
-      return i18n("The file '%1' was changed (%2) on disc by another program!").arg( url().prettyURL() ).arg( reason );
+      return i18n("The file '%1' was changed (%2) on disk by another program!").arg( url().prettyURL() ).arg( reason );
     }
 
     /**
diff -urN -x CVS kdelibs.orig/kate/part/katehighlight.cpp kdelibs/kate/part/katehighlight.cpp
--- kdelibs.orig/kate/part/katehighlight.cpp	Wed Jun 30 19:11:38 2004
+++ kdelibs/kate/part/katehighlight.cpp	Sun Aug 15 13:17:48 2004
@@ -1727,7 +1727,7 @@
   // If no highlighting is selected we need only one default.
   if (noHl)
   {
-    list.append(new KateHlItemData(I18N_NOOP("Normal Text"), KateHlItemData::dsNormal));
+    list.append(new KateHlItemData(i18n("Normal Text"), KateHlItemData::dsNormal));
     return;
   }
 
diff -urN -x CVS kdelibs.orig/kdecore/kapplication.cpp kdelibs/kdecore/kapplication.cpp
--- kdelibs.orig/kdecore/kapplication.cpp	Thu Jul 22 18:50:04 2004
+++ kdelibs/kdecore/kapplication.cpp	Tue Aug 17 15:44:00 2004
@@ -18,7 +18,7 @@
     Boston, MA 02111-1307, USA.
         */
 
-// $Id$
+// $Id$
 
 #include "config.h"
 
@@ -2171,6 +2171,9 @@
      if (allowAttachments && q.startsWith("attach="))
        attachURLs.push_back(KURL::decode_string((*it).mid(7)));
      else
+     if (allowAttachments && q.startsWith("attachment="))
+       attachURLs.push_back(KURL::decode_string((*it).mid(11)));
+     else
      if (q.startsWith("to="))
        address = address.isEmpty()? KURL::decode_string((*it).mid(3)): address + ',' + KURL::decode_string((*it).mid(3));
    }
diff -urN -x CVS kdelibs.orig/kdecore/kextsock.cpp kdelibs/kdecore/kextsock.cpp
--- kdelibs.orig/kdecore/kextsock.cpp	Sun Jul 11 20:30:41 2004
+++ kdelibs/kdecore/kextsock.cpp	Sun Aug 15 20:52:31 2004
@@ -1369,7 +1369,7 @@
       QByteArray *a = outBuf.first();
       unsigned count = 0;
 
-      while (a && count + (a->size() - offset) < buf.size())
+      while (a && count + (a->size() - offset) <= buf.size())
 	{
 	  memcpy(buf.data() + count, a->data() + offset, a->size() - offset);
 	  count += a->size() - offset;
@@ -1377,6 +1377,16 @@
 	  a = outBuf.next();
 	}
 
+      // see if we can still fit more
+      if (a && count < buf.size())
+	{
+	  // getting here means this buffer (a) is larger than
+	  // (buf.size() - count) (even for count == 0).
+	  memcpy(buf.data() + count, a->data() + offset, buf.size() - count);
+	  offset += buf.size() - count;
+	  count = buf.size();
+	}
+
       // now try to write those bytes
       int wrote = KSocks::self()->write(sockfd, buf, count);
 
diff -urN -x CVS kdelibs.orig/kdecore/network/ksocketbuffer.cpp kdelibs/kdecore/network/ksocketbuffer.cpp
--- kdelibs.orig/kdecore/network/ksocketbuffer.cpp	Sat Jul 31 16:36:27 2004
+++ kdelibs/kdecore/network/ksocketbuffer.cpp	Sun Aug 15 20:40:04 2004
@@ -248,7 +248,7 @@
       // better by concatenating a few of them into a big buffer
       // question is: how big should that buffer be? 2 kB should be enough
 
-      Q_ULONG bufsize = 2048;
+      Q_ULONG bufsize = 1460;
       if (len != -1 && len < bufsize)
 	bufsize = len;
       QByteArray buf(bufsize);
@@ -262,6 +262,16 @@
 	  ++it;
 	}
 
+      // see if we can still fit more
+      if (count < bufsize && it != end)
+	{
+	  // getting here means this buffer (*it) is larger than
+	  // (bufsize - count) (even for count == 0).
+	  memcpy(buf.data() + count, (*it).data() + offset, bufsize - count);
+	  offset += bufsize - count;
+	  count = bufsize;
+	}
+
       // now try to write those bytes
       Q_LONG wrote = dev->writeBlock(buf, count);
 
diff -urN -x CVS kdelibs.orig/kdeui/kcolordialog.cpp kdelibs/kdeui/kcolordialog.cpp
--- kdelibs.orig/kdeui/kcolordialog.cpp	Tue Jul 20 16:25:00 2004
+++ kdelibs/kdeui/kcolordialog.cpp	Fri Aug 20 09:23:44 2004
@@ -532,10 +532,17 @@
      }
 }
 
+class KPaletteTable::KPaletteTablePrivate
+{
+public:
+    QMap<QString,QColor> m_namedColorMap;
+};
 
 KPaletteTable::KPaletteTable( QWidget *parent, int minWidth, int cols)
 	: QWidget( parent ), mMinWidth(minWidth), mCols(cols)
 {
+  d = new KPaletteTablePrivate;
+  
   cells = 0;
   mPalette = 0;
   i18n_customColors = i18n("* Custom Colors *");
@@ -580,6 +587,7 @@
 KPaletteTable::~KPaletteTable()
 {
    delete mPalette;
+   delete d;
 }
 
 QString
@@ -652,7 +660,14 @@
 	{
 	  continue;
 	}
-	list.append( i18n("color", name.latin1() ) );
+
+        const QColor color ( red, green, blue );
+        if ( color.isValid() )
+        {
+            const QString colorName( i18n("color", name.latin1() ) );
+            list.append( colorName );
+            d->m_namedColorMap[ colorName ] = color;
+        }
       }
     }
 
@@ -825,7 +840,7 @@
 void
 KPaletteTable::slotColorTextSelected( const QString &colorText )
 {
-  emit colorSelected( QColor (colorText), colorText );
+  emit colorSelected( d->m_namedColorMap[ colorText ], colorText );
 }
 
 
diff -urN -x CVS kdelibs.orig/kdeui/klistview.cpp kdelibs/kdeui/klistview.cpp
--- kdelibs.orig/kdeui/klistview.cpp	Mon Aug  9 06:28:33 2004
+++ kdelibs/kdeui/klistview.cpp	Tue Aug 17 21:40:43 2004
@@ -461,10 +461,7 @@
 {
   QListViewItem* item = itemAt( point );
   if ( item ) {
-    int offset = treeStepSize() * ( item->depth() + ( rootIsDecorated() ? 1 : 0) );
-    if (point.x() > (item->width( fontMetrics() , this, 0 ) + offset ))
-      return false;
-    return isExecuteArea( point.x() );
+    return isExecuteArea( point.x(), item );
   }
 
   return false;
@@ -472,6 +469,11 @@
 
 bool KListView::isExecuteArea( int x )
 {
+  return isExecuteArea( x, 0 );
+}
+
+bool KListView::isExecuteArea( int x, QListViewItem* item )
+{
   if( allColumnsShowFocus() )
     return true;
   else {
@@ -483,6 +485,16 @@
       offset += columnWidth( header()->mapToSection( index ) );
 
     x += contentsX(); // in case of a horizontal scrollbar
+
+    if ( item )
+    {
+	width = treeStepSize()*( item->depth() + ( rootIsDecorated() ? 1 : 0 ) );
+	width += itemMargin();
+	int ca = AlignHorizontal_Mask & columnAlignment( 0 );
+	if ( ca == AlignRight || ca == AlignAuto )
+	    width += item->width( fontMetrics(), this, 0 );
+    }
+
     return ( x > offset && x < ( offset + width ) );
   }
 }
diff -urN -x CVS kdelibs.orig/kdeui/klistview.h kdelibs/kdeui/klistview.h
--- kdelibs.orig/kdeui/klistview.h	Sat Jul 10 16:13:47 2004
+++ kdelibs/kdeui/klistview.h	Tue Aug 17 21:40:43 2004
@@ -968,6 +968,7 @@
 private:
   class KListViewPrivate;
   KListViewPrivate *d;
+  bool isExecuteArea( int x, QListViewItem* item );
 };
 
 /**
diff -urN -x CVS kdelibs.orig/kinit/configure.in.in kdelibs/kinit/configure.in.in
--- kdelibs.orig/kinit/configure.in.in	Thu Jul  1 18:05:11 2004
+++ kdelibs/kinit/configure.in.in	Tue Aug 17 10:37:18 2004
@@ -3,14 +3,37 @@
 KDE_CHECK_LIB(qt-mt,XftInit,[],[KDEINIT_USE_XFT=])
 
 dnl Xft requires freetype to compile
+KDE_FIND_PATH(fontconfig-config, FONTCONFIG_CONFIG, [${prefix}/bin ${exec_prefix}/bin /usr/bin /usr/local/bin /opt/local/bin], [ KDE_FIND_PATH(pkg-config, PKGCONFIG, [${prefix}/bin ${exec_prefix}/bin /usr/bin /usr/local/bin /opt/local/bin], [AC_MSG_WARN([Could not find neither pkg-config nor fontconfig-config, check http://www.fontconfig.org/ ])
+])
+])
+
+if test -n "$PKGCONFIG"; then
+  vers=`$PKGCONFIG fontconfig --modversion 2>/dev/null | sed -e 's/libfontconfig //' | awk 'BEGIN { FS = "."; } { printf "%d", ($1 * 1000 + $2) * 1000 + $3;}'`
+  if test -n "$vers" && test "$vers" -ge 1000000
+  then
+     FONTCONFIG_CFLAGS="`$PKGCONFIG fontconfig --cflags`"
+     KDEINIT_FONTCONFIG=1
+  fi
+fi
+
+if test -n "$FONTCONFIG_CONFIG"; then
+  vers=`$FONTCONFIG_CONFIG --version 2>/dev/null | sed -e 's/libfontconfig //' | awk 'BEGIN { FS = "."; } { printf "%d", ($1 * 1000 + $2) * 1000 + $3;}'`
+  if test -n "$vers" && test "$vers" -ge 1000000
+  then
+     FONTCONFIG_CFLAGS="`$FONTCONFIG_CONFIG --cflags`"
+     KDEINIT_FONTCONFIG=1
+  fi
+fi
+
 KDE_FIND_PATH(freetype-config, KDEINIT_FREETYPE_CONFIG, [${prefix}/bin ${exec_prefix}/bin /usr/bin /usr/local/bin /opt/local/bin])
 if test -n "$KDEINIT_FREETYPE_CONFIG"; then
-    KDEINIT_XFT_INCLUDES=`$KDEINIT_FREETYPE_CONFIG --cflags`
+   if test -n "$KDEINIT_FONTCONFIG"; then
+      KDEINIT_XFT_INCLUDES="`$KDEINIT_FREETYPE_CONFIG --cflags` $FONTCONFIG_CFLAGS"
     kdeinit_cppflags=$CPPFLAGS
-    CPPFLAGS="$CPPFLAGS $KDEINIT_XFT_INCLUDES"
+      CPPFLAGS="$CPPFLAGS $KDEINIT_XFT_INCLUDES $FONTCONFIG_CFLAGS"
     KDE_CHECK_HEADER(X11/Xft/Xft.h,[],[KDEINIT_USE_XFT=])
     CPPFLAGS=$kdeinit_cppflags
-    KDE_CHECK_HEADER(fontconfig/fontconfig.h,[],[KDEINIT_USE_XFT=])
+   fi
 else
     KDEINIT_USE_XFT=
 fi
diff -urN -x CVS kdelibs.orig/kio/kfile/kopenwith.cpp kdelibs/kio/kfile/kopenwith.cpp
--- kdelibs.orig/kio/kfile/kopenwith.cpp	Mon Jul 19 16:57:10 2004
+++ kdelibs/kio/kfile/kopenwith.cpp	Mon Aug 16 16:02:32 2004
@@ -101,15 +101,27 @@
 }
 
 
-// ----------------------------------------------------------------------
-// Ensure that dirs are sorted in front of files and case is ignored
+/* Ensures that directories sort before non-directories */
+int KAppTreeListItem::compare(QListViewItem *i, int col, bool ascending) const
+{
+	KAppTreeListItem *other = dynamic_cast<KAppTreeListItem *>(i);
+
+	// Directories sort first
+	if (directory && !other->directory)
+		return -1;
+
+	else if (!directory && other->directory)
+		return 1;
+
+	else // both directories or both not
+		return QListViewItem::compare(i, col, ascending);
+}
 
+// ----------------------------------------------------------------------
+// Ensure that case is ignored
 QString KAppTreeListItem::key(int column, bool /*ascending*/) const
 {
-    if (directory)
-        return QString::fromLatin1(" ") + text(column).upper();
-    else
-        return text(column).upper();
+	return text(column).upper();
 }
 
 void KAppTreeListItem::activate()
@@ -141,6 +153,7 @@
     setRootIsDecorated( true );
 
     addDesktopGroup( QString::null );
+	cleanupTree();
 
     connect( this, SIGNAL( currentChanged(QListViewItem*) ),
             SLOT( slotItemHighlighted(QListViewItem*) ) );
@@ -273,6 +286,40 @@
     KListView::resizeEvent(e);
 }
 
+// Prune empty directories from the tree
+void KApplicationTree::cleanupTree()
+{
+	QListViewItem *item=firstChild();
+	while(item!=0)
+	{
+		if(item->isExpandable())
+		{
+			item->setOpen(true);
+			if(item->childCount()==0) {
+				QListViewItem *current=item;
+				item=item->itemBelow();
+				delete current;
+				continue;
+			}
+			item=item->itemBelow();
+			continue;
+		}
+		item=item->itemBelow();
+	}
+	item=firstChild();
+	while(item!=0)
+	{
+		if(item->isExpandable())
+		{
+			QListViewItem *temp=item->itemBelow();
+			if(item->text(0)!=i18n("Applications"))
+				item->setOpen(false);
+			item=temp;
+			continue;
+		}
+		item=item->itemBelow();
+	}
+}
 
 /***************************************************************
  *
@@ -289,7 +336,7 @@
 };
 
 KOpenWithDlg::KOpenWithDlg( const KURL::List& _urls, QWidget* parent )
-             :QDialog( parent, 0L, true )
+             :QDialog( parent, "openwith", true )
 {
     setCaption( i18n( "Open With" ) );
     QString text;
@@ -308,7 +355,7 @@
 
 KOpenWithDlg::KOpenWithDlg( const KURL::List& _urls, const QString&_text,
                             const QString& _value, QWidget *parent)
-             :QDialog( parent, 0L, true )
+             :QDialog( parent, "openwith", true )
 {
   QString caption = KStringHandler::csqueeze( _urls.first().prettyURL() );
   if (_urls.count() > 1)
@@ -320,7 +367,7 @@
 
 KOpenWithDlg::KOpenWithDlg( const QString &serviceType, const QString& value,
                             QWidget *parent)
-             :QDialog( parent, 0L, true )
+             :QDialog( parent, "openwith", true )
 {
     setCaption(i18n("Choose Application for %1").arg(serviceType));
   QString text = i18n("<qt>Select the program for the file type: <b>%1</b>. "
@@ -333,7 +380,7 @@
 }
 
 KOpenWithDlg::KOpenWithDlg( QWidget *parent)
-             :QDialog( parent, 0L, true )
+             :QDialog( parent, "openwith", true )
 {
   setCaption(i18n("Choose Application"));
   QString text = i18n("<qt>Select a program. "
diff -urN -x CVS kdelibs.orig/kio/kfile/kopenwith_p.h kdelibs/kio/kfile/kopenwith_p.h
--- kdelibs.orig/kio/kfile/kopenwith_p.h	Fri Feb 28 02:03:58 2003
+++ kdelibs/kio/kfile/kopenwith_p.h	Sun Aug 15 00:24:18 2004
@@ -1,4 +1,4 @@
-// "$Id$"
+// "$Id$"
 /* This file is part of the KDE libraries
     Copyright (C) 2000 David Faure <faure@kde.org>
 
@@ -45,6 +45,7 @@
     QString exec;
 
 protected:
+	int compare(QListViewItem *i, int col, bool ascending ) const;
     QString key(int column, bool ascending) const;
 
     void init(const QPixmap& pixmap, bool parse, bool dir, const QString &_path, const QString &exec);
@@ -84,6 +85,7 @@
 protected:
     void resizeEvent( QResizeEvent *_ev );
     KAppTreeListItem* currentitem;
+	void cleanupTree();
 
 public slots:
     void slotItemHighlighted(QListViewItem* i);
diff -urN -x CVS kdelibs.orig/kio/kfile/kpropertiesdialog.cpp kdelibs/kio/kfile/kpropertiesdialog.cpp
--- kdelibs.orig/kio/kfile/kpropertiesdialog.cpp	Sat Aug  7 13:00:48 2004
+++ kdelibs/kio/kfile/kpropertiesdialog.cpp	Fri Aug 20 18:35:26 2004
@@ -863,6 +863,8 @@
 
     connect( button, SIGNAL( clicked() ), SLOT( slotEditFileType() ));
 
+    if (!kapp->authorizeKAction("editfiletype"))
+       button->hide();
 
     grid->addWidget(box, curRow++, 2);
   }
diff -urN -x CVS kdelibs.orig/kio/misc/uiserver.cpp kdelibs/kio/misc/uiserver.cpp
--- kdelibs.orig/kio/misc/uiserver.cpp	Wed Jul 14 00:10:23 2004
+++ kdelibs/kio/misc/uiserver.cpp	Fri Aug 20 16:21:47 2004
@@ -189,7 +189,7 @@
   m_sAppId = app_id;
   m_iJobId = job_id;
   m_visible = true;
-  m_defaultProgressVisible = true;
+  m_defaultProgressVisible = showDefault;
 
   // create dialog, but don't show it
   defaultProgress = new KIO::DefaultProgress( false );
diff -urN -x CVS kdelibs.orig/kioslave/http/kcookiejar/kcookieserver.cpp kdelibs/kioslave/http/kcookiejar/kcookieserver.cpp
--- kdelibs.orig/kioslave/http/kcookiejar/kcookieserver.cpp	Tue Jul 20 17:29:24 2004
+++ kdelibs/kioslave/http/kcookiejar/kcookieserver.cpp	Mon Aug 16 16:05:11 2004
@@ -23,7 +23,7 @@
 //----------------------------------------------------------------------------
 //
 // KDE Cookie Server
-// $Id$
+// $Id$
 
 #define SAVE_DELAY 3 // Save after 3 minutes
 
@@ -80,6 +80,7 @@
 {
    mOldCookieServer = new DCOPClient(); // backwards compatibility.
    mOldCookieServer->registerAs("kcookiejar", false);
+   mOldCookieServer->setDaemonMode( true );
    mCookieJar = new KCookieJar;
    mPendingCookies = new KHttpCookieList;
    mPendingCookies->setAutoDelete(true);
