--- kdelibs-3.0.2/kdeui/ktip.cpp.than	2002-08-02 14:40:57.000000000 +0200
+++ kdelibs-3.0.2/kdeui/ktip.cpp	2002-08-02 16:55:54.000000000 +0200
@@ -148,7 +148,7 @@
   titlePane->setBackgroundPixmap(locate("data", "kdeui/pics/ktip-background.png"));
   titlePane->setText(i18n("Did you know...?\n"));
   titlePane->setFont(QFont(KGlobalSettings::generalFont().family(), 20,  QFont::Bold));
-  titlePane->setAlignment(QLabel::AlignCenter);
+  titlePane->setAlignment(Qt::AlignCenter|Qt::WordBreak);
   hbox->addWidget(titlePane,100);
 
   _tipText = new KTextBrowser(this);
