Index: kurlrequester.cpp
===================================================================
RCS file: /home/kde/kdelibs/kfile/kurlrequester.cpp,v
retrieving revision 1.22
diff -u -p -r1.22 kurlrequester.cpp
--- kdelibs-2.1/kfile/kurlrequester.cpp	2001/02/28 09:56:59	1.22
+++ kdelibs-2.1/kfile/kurlrequester.cpp	2001/03/09 15:16:33
@@ -34,6 +34,7 @@
 #include <klocale.h>
 #include <kurlcompletion.h>
 #include <kurldrag.h>
+#include <kprotocolinfo.h>
 
 #include "kurlrequester.h"
 
@@ -228,7 +229,10 @@ void KURLRequester::slotOpenDialog()
     
     KFileDialog *dlg = fileDialog();
     if ( !d->text().isEmpty() ) {
-	dlg->setSelection( url() );
+        KURL u( url() );
+        // If we won't be able to list it (e.g. http), then don't try :)
+        if ( KProtocolInfo::supportsListing( u.protocol() ) )
+	    dlg->setSelection( u.url() );
     }
 
     if ( dlg->exec() == QDialog::Accepted )
