cvs server: Diffing .
Index: knumvalidator.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdeui/knumvalidator.cpp,v
retrieving revision 1.4
diff -u -p -r1.4 knumvalidator.cpp
--- kdelibs/kdeui/knumvalidator.cpp	2001/07/17 13:22:31	1.4
+++ kdelibs/kdeui/knumvalidator.cpp	2002/02/18 14:56:21
@@ -73,16 +73,15 @@ QValidator::State KIntValidator::validat
     val = 0;
     ok = true;
   }
-
   if (! ok)
     return QValidator::Invalid;
-
   if ((! _min && ! _max) || (val >= _min && val <= _max))
     return QValidator::Acceptable;
 
   if (_max && _min >= 0 && val < 0)
     return QValidator::Invalid;
-
+  if( val > _max || val <_min)
+    return QValidator::Invalid;
   return QValidator::Valid;
 }
 
cvs server: Diffing tests
