Index: cupsdconf.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdeprint/cups/cupsdconf/cupsdconf.cpp,v
retrieving revision 1.5
diff -u -3 -p -r1.5 cupsdconf.cpp
--- kdelibs/kdeprint/cups/cupsdconf/cupsdconf.cpp	2001/08/03 19:59:21	1.5
+++ kdelibs/kdeprint/cups/cupsdconf/cupsdconf.cpp	2002/03/11 14:04:20
@@ -297,6 +297,14 @@ bool CupsdConf::saveToFile(const QString
                         }
 			t << "</Location>" << endl;
 		}
+		if (unknown_.count() > 0)
+		{
+			t << endl << "# Options not supported by KDEPrint" << endl;
+			for (QStringList::Iterator it=unknown_.begin(); it!=unknown_.end(); ++it)
+			{
+				t << *it << endl;
+			}
+		}
 		return true;
 	}
 }
@@ -403,7 +411,10 @@ kdDebug() << "warning: empty option \"" 
 	else if (opt.lower() == "browseinterval") browseinterval_ = (*(wordlist_.at(1))).toInt();
 	else if (opt.lower() == "browsetimeout") browsetimeout_ = (*(wordlist_.at(1))).toInt();
 	else if (opt.lower() == "systemgroup") systemgroup_ = *(wordlist_.at(1));
-	else return false;
+	else
+	{
+		unknown_.append(wordlist_.join(" "));
+	}
 	return true;
 }
 
Index: cupsdconf.h
===================================================================
RCS file: /home/kde/kdelibs/kdeprint/cups/cupsdconf/cupsdconf.h,v
retrieving revision 1.1
diff -u -3 -p -r1.1 cupsdconf.h
--- kdelibs/kdeprint/cups/cupsdconf/cupsdconf.h	2001/03/30 21:48:47	1.1
+++ kdelibs/kdeprint/cups/cupsdconf/cupsdconf.h	2002/03/11 14:04:20
@@ -130,6 +130,9 @@ struct CupsdConf
 
 	// cupsd.conf file comments
 	CupsdComment	comments_;
+
+	// unsupported settings
+	QStringList	unknown_;
 };
 
 struct CupsLocation
Index: cupsddialog.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdeprint/cups/cupsdconf/cupsddialog.cpp,v
retrieving revision 1.1
diff -u -3 -p -r1.1 cupsddialog.cpp
--- kdelibs/kdeprint/cups/cupsdconf/cupsddialog.cpp	2001/03/30 21:48:47	1.1
+++ kdelibs/kdeprint/cups/cupsdconf/cupsddialog.cpp	2002/03/11 14:04:20
@@ -185,6 +185,12 @@ bool CupsdDialog::setConfigFile(const QS
 		KMessageBox::error(this, i18n("Error while loading configuration file !"), i18n("CUPS configuration error"));
 		return false;
 	}
+	else if (conf_->unknown_.count() > 0)
+	{
+		KMessageBox::sorry(this, i18n("Unsupported options detected. You will not be able to\n"
+					      "modify those settings in this configuration tool, but they\n"
+					      "will be restored untouched:\n\n") + conf_->unknown_.join("\n"));
+	}
 	bool	ok(true);
 	QString	msg;
 	for (pagelist_.first();pagelist_.current() && ok;pagelist_.next())
@@ -253,6 +259,7 @@ void CupsdDialog::slotApply()
 		CupsdConf	newconf_;
 		for (pagelist_.first();pagelist_.current() && ok;pagelist_.next())
 			ok = pagelist_.current()->saveConfig(&newconf_, msg);
+		newconf_.unknown_ = conf_->unknown_;
 		if (!ok)
 		{
 			; // do nothing
