--- kdelibs-2.2.2/kfile/kicondialog.cpp--	Tue Nov 27 14:54:58 2001
+++ kdelibs-2.2.2/kfile/kicondialog.cpp	Tue Nov 27 14:56:21 2001
@@ -327,17 +327,23 @@ void KIconDialog::slotButtonClicked(int 
     switch (id)
     {
     case 0:
-	mType = 0;
-	mpBrowseBut->setEnabled(false);
-	mpCombo->setEnabled(true);
-	showIcons();
+      if(mType!=0)
+        {
+            mType = 0;
+            mpBrowseBut->setEnabled(false);
+            mpCombo->setEnabled(true);
+            showIcons();
+        }
 	break;
 
     case 1:
-	mType = 1;
-	mpBrowseBut->setEnabled(true);
-	mpCombo->setEnabled(false);
-	showIcons();
+	if(mType!=1)
+	{
+	    mType = 1;
+	    mpBrowseBut->setEnabled(true);
+	    mpCombo->setEnabled(false);
+	    showIcons();
+	}
 	break;
 
     case 2:
