diff -urN -x CVS kdelibs.orig/admin/acinclude.m4.in kdelibs/admin/acinclude.m4.in
--- kdelibs.orig/admin/acinclude.m4.in	Fri Nov 12 23:48:09 2004
+++ kdelibs/admin/acinclude.m4.in	Thu Dec  9 22:22:14 2004
@@ -4254,7 +4254,8 @@
 
 AC_DEFUN([KDE_CHECK_PYTHON],
 [
-  KDE_CHECK_PYTHON_INTERN("2.3", 
+  KDE_CHECK_PYTHON_INTERN("2.4",
+   [KDE_CHECK_PYTHON_INTERN("2.3", 
    [KDE_CHECK_PYTHON_INTERN("2.2", 
      [KDE_CHECK_PYTHON_INTERN("2.1", 
        [KDE_CHECK_PYTHON_INTERN("2.0", 
@@ -4262,6 +4263,7 @@
        ])
      ])
    ])
+  ])
 ])
 
 AC_DEFUN([KDE_CHECK_STL],
diff -urN -x CVS kdelibs.orig/dcop/dcopclient.cpp kdelibs/dcop/dcopclient.cpp
--- kdelibs.orig/dcop/dcopclient.cpp	Tue Jul 13 14:45:47 2004
+++ kdelibs/dcop/dcopclient.cpp	Fri Dec 10 17:02:57 2004
@@ -1846,7 +1846,7 @@
             tv.tv_sec = msecs / 1000;
             tv.tv_usec = (msecs % 1000) * 1000;
             if ( select( socket() + 1, &fds, 0, 0, &tv ) <= 0 ) {
-                if( useEventLoop && (time_left > guiTimeout)) {
+                if( useEventLoop && (timeout < 0 || time_left > guiTimeout)) {
                     // nothing was available, we got a timeout. Reactivate
                     // the GUI in blocked state.
                     bool old_lock = d->non_blocking_call_lock;
@@ -1854,7 +1854,8 @@
                         d->non_blocking_call_lock = true;
                         emit blockUserInput( true );
                     }
-                    d->eventLoopTimer.start(time_left - guiTimeout, true);
+                    if( timeout >= 0 )
+                        d->eventLoopTimer.start(time_left - guiTimeout, true);
                     qApp->enter_loop();
                     d->eventLoopTimer.stop();
                     if ( !old_lock ) {
diff -urN -x CVS kdelibs.orig/dcop/dcopidlng/kalyptusCxxToDcopIDL.pm kdelibs/dcop/dcopidlng/kalyptusCxxToDcopIDL.pm
--- kdelibs.orig/dcop/dcopidlng/kalyptusCxxToDcopIDL.pm	Sun Oct 19 14:18:57 2003
+++ kdelibs/dcop/dcopidlng/kalyptusCxxToDcopIDL.pm	Sat Dec 11 14:37:42 2004
@@ -671,6 +671,7 @@
 	$argType =~ s/^\s*(.*?)\s*$/$1/;
 	$argType =~ s/</&lt;/g;
 	$argType =~ s/>/&gt;/g;
+	$argType =~ s/\s//g;
 
 	$args .= "        <ARG><TYPE$typeAttrs>$argType</TYPE><NAME>$arg->{ArgName}</NAME></ARG>\n";
 
diff -urN -x CVS kdelibs.orig/interfaces/kimproxy/interface/dcopinstantmessenger.desktop kdelibs/interfaces/kimproxy/interface/dcopinstantmessenger.desktop
--- kdelibs.orig/interfaces/kimproxy/interface/dcopinstantmessenger.desktop	Mon Nov 22 09:34:03 2004
+++ kdelibs/interfaces/kimproxy/interface/dcopinstantmessenger.desktop	Sat Dec 11 12:18:32 2004
@@ -4,6 +4,7 @@
 X-KDE-ServiceType=DCOP/InstantMessenger
 Comment=Instant Messenger with a DCOP interface
 Comment[af]=Oombliklike Boodskapper met DCOP intervlak
+Comment[ar]=مراسل فوري مع واجهة DCOP
 Comment[bg]=Програма за съобщения в реално време с интерфейс DCOP
 Comment[bn]=ডিকপ (DCOP) ইন্টারফেস সহ ইনস্ট্যান্ট মেসেঞ্জার
 Comment[bs]=Instant poruke sa DCOP interfejsom
diff -urN -x CVS kdelibs.orig/interfaces/kimproxy/interface/kcm_instantmessenger.desktop kdelibs/interfaces/kimproxy/interface/kcm_instantmessenger.desktop
--- kdelibs.orig/interfaces/kimproxy/interface/kcm_instantmessenger.desktop	Sat Nov 13 08:20:18 2004
+++ kdelibs/interfaces/kimproxy/interface/kcm_instantmessenger.desktop	Sat Dec 11 12:18:32 2004
@@ -38,6 +38,7 @@
 Name[zh_CN]=即时通讯程序
 Comment=The instant messenger allows two-way chat between individuals and groups.
 Comment[af]=Die oombliklike boodskappe laat twee rigting kommunikasie tussen individue en groepe toe.
+Comment[ar]=يسمح المراسل الفوري بمحادثة متبادلة بين أفراد ومجموعات
 Comment[bg]=Разговор между различни хора в реално време.
 Comment[bn]=ইনস্ট্যান্ট মেসেঞ্জার একাধিক ব্যক্তি বা গোষ্ঠীকে দ্বিমুখী বার্তালাপ বা আড্ডার সুযোগ দেয়। 
 Comment[bs]=Program za instant poruke omogućuje dvosmjernu komunikaciju između pojedinaca i grupa.
diff -urN -x CVS kdelibs.orig/kate/part/kateviewhelpers.cpp kdelibs/kate/part/kateviewhelpers.cpp
--- kdelibs.orig/kate/part/kateviewhelpers.cpp	Sat Jul 10 00:08:56 2004
+++ kdelibs/kate/part/kateviewhelpers.cpp	Wed Dec  8 19:49:37 2004
@@ -104,7 +104,7 @@
 
 void KateCmdLine::hideMe ()
 {
-  if (isVisibleTo(parentWidget())) {
+  if ( isVisibleTo(parentWidget()) && ! hasFocus() ) {
      m_view->toggleCmdLine ();
   }
   //m_view->/*show*/setCmdLine (false);
diff -urN -x CVS kdelibs.orig/kate/plugins/autobookmarker/ktexteditor_autobookmarker.desktop kdelibs/kate/plugins/autobookmarker/ktexteditor_autobookmarker.desktop
--- kdelibs.orig/kate/plugins/autobookmarker/ktexteditor_autobookmarker.desktop	Fri Nov 12 08:27:48 2004
+++ kdelibs/kate/plugins/autobookmarker/ktexteditor_autobookmarker.desktop	Sat Dec 11 12:18:32 2004
@@ -42,6 +42,7 @@
 Name[zh_CN]=KTextEditor 自动书签器
 Comment=Set bookmarks on lines matching a pattern when documents are loaded
 Comment[af]=Stel 'n boekmerk op lyne wat met 'n patroon ooreenstem wanneer die dokument gelaai word.
+Comment[ar]=ضبط المحفوظات على الأسطر الموافقة لنمط عند تحميل المستندات
 Comment[bg]=Установяване на отметки на редовете, които отговарят на зададен шаблон по време на зареждане на файла
 Comment[bn]=ডকুমেন্ট লোড করাকালীন প্রদত্ত নকশার সাথে মেলে এমন সমস্ত লাইন বুকমার্ক করে
 Comment[bs]=Podešava zabilješke na linijama koje odgovaraju uzorku pri učitavanju dokumenta
diff -urN -x CVS kdelibs.orig/kate/plugins/isearch/ktexteditor_isearch.desktop kdelibs/kate/plugins/isearch/ktexteditor_isearch.desktop
--- kdelibs.orig/kate/plugins/isearch/ktexteditor_isearch.desktop	Fri Nov 12 08:27:49 2004
+++ kdelibs/kate/plugins/isearch/ktexteditor_isearch.desktop	Sat Dec 11 12:18:32 2004
@@ -63,6 +63,7 @@
 Name[zu]=I-Plugin Yosesho Lokwenyuka ngezigaba ze-KTextEditor
 Comment=Also known as "As you type search"
 Comment[af]=Ook bekend as "Soos jy tik soektog"
+Comment[ar]=يعرف أيضاً بـ "البحث أثناء الكتابة"
 Comment[bg]=Последователно търсене, известно още като търсене по време на въвеждане на низа
 Comment[bn]="টাইপ করাকালীন সন্ধান" হিসাবেও পরিচিত
 Comment[bs]=Takođe poznat kao "Pretraga dok kucate"
@@ -75,6 +76,7 @@
 Comment[et]=Tuntud ka kui "otsimine vastavalt kirjutamisele"
 Comment[fi]=Tunnetaan myös nimellä "Hae kun kirjoitat"
 Comment[fr]=Aussi connu comme « Recherche pendant la saisie »
+Comment[ga]=Darbh ainm "Cuardach Beo" freisin
 Comment[he]=ידוע גם בתור "חיפוש תוך כדי כתיבה"
 Comment[hr]=Poznat i kao "As you type search"
 Comment[hu]=Más néven "Beírásos keresés"
diff -urN -x CVS kdelibs.orig/kate/plugins/wordcompletion/ktexteditor_docwordcompletion.desktop kdelibs/kate/plugins/wordcompletion/ktexteditor_docwordcompletion.desktop
--- kdelibs.orig/kate/plugins/wordcompletion/ktexteditor_docwordcompletion.desktop	Fri Nov 12 08:27:49 2004
+++ kdelibs/kate/plugins/wordcompletion/ktexteditor_docwordcompletion.desktop	Sat Dec 11 12:18:32 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Name=KTextEditor Word Completion Plugin
 Name[af]=KTextEditor Woord Voltooiïng Inprop Module
+Name[ar]=KTextEditor وظيفة إضافيّة لإكمال الكلمات
 Name[bn]=কে-টেক্সট-এডিটর শব্দ পরিপূরণ প্লাগ-ইন
 Name[bs]=KTextEditor dodatak za dovršavanje riječi
 Name[ca]=Endollable de completar paraules del KTextEditor
@@ -43,6 +44,7 @@
 Name[zh_CN]=KTextEditor 单词补全插件
 Comment=Directional or popup-based completion from words in the document
 Comment[af]=Direksionele of opspring gebaseerde woord voltooiïng van woorde in die dokument
+Comment[ar]=إكمال متجهي أو مبني على الانبثاق من الكلمات في المستند
 Comment[bg]=Автоматично завършване на думи, базирано на използваните вече думи в документа
 Comment[bn]=ডকুমেন্ট-এর অন্যান্য শব্দের উপর ভিত্তি করে পরিপূরণ
 Comment[bs]=Direkcionalno ili popup-bazirano dovršavanje za riječi u dokumentu
diff -urN -x CVS kdelibs.orig/kdecore/all_languages.desktop kdelibs/kdecore/all_languages.desktop
--- kdelibs.orig/kdecore/all_languages.desktop	Sun Nov 28 09:33:02 2004
+++ kdelibs/kdecore/all_languages.desktop	Sat Dec 11 12:18:33 2004
@@ -223,6 +223,7 @@
 Name[fa]=امهری
 Name[fi]=Amhara
 Name[fr]=Amharique
+Name[ga]=Amarais
 Name[he]=אמהרית
 Name[hi]=अम्हारिक
 Name[hr]=Amarski
@@ -462,6 +463,7 @@
 Name[et]=Vanapärsia
 Name[fa]=اوستایی
 Name[fi]=Avesta
+Name[ga]=Aivéistis
 Name[he]=אווסטית
 Name[hi]=अवेस्तन
 Name[hr]=Avestanski
@@ -504,6 +506,7 @@
 Name[et]=Aimaraa
 Name[fa]=آیمارا
 Name[fi]=Aimara
+Name[ga]=Adhmarais
 Name[gl]=Aimara
 Name[he]=איימרה
 Name[hi]=अयमारा
@@ -840,6 +843,7 @@
 Name[cy]=Bihareg
 Name[de]=Biharisch
 Name[fa]=بیحاری
+Name[ga]=Bihairis
 Name[he]=ביהרי
 Name[hi]=बिहारी
 Name[hr]=Biharski
@@ -873,6 +877,7 @@
 Name[cy]=Bislameg
 Name[fa]=بیسلاما
 Name[fr]=Bichlamar
+Name[ga]=Bioslamais
 Name[he]=ביסלמה
 Name[hi]=बिस्लामा
 Name[hr]=Bislamski
@@ -1285,6 +1290,7 @@
 Name[fa]=چچن
 Name[fi]=Tšetšeeni
 Name[fr]=Tchétchène
+Name[ga]=Seisnis
 Name[gl]=Checheno
 Name[he]=צ'צ'נית
 Name[hi]=चेचेन
@@ -1473,6 +1479,7 @@
 Name[fa]=چووش
 Name[fi]=Tšuvassi
 Name[fr]=Tchouvache
+Name[ga]=Suvais
 Name[he]=צ'ובשית
 Name[hi]=चाउवेश
 Name[hr]=Čuvaški
@@ -2646,6 +2653,7 @@
 Name[fa]=گالگان
 Name[fi]=Galicia
 Name[fr]=Galicien
+Name[ga]=Gailísis
 Name[gl]=Galego
 Name[he]=גליציאנית
 Name[hi]=गैलियन
@@ -2806,6 +2814,7 @@
 Name[et]=Guaranii
 Name[eu]=Guaraniera
 Name[fa]=گورانی
+Name[ga]=Guaráinis
 Name[gl]=Guaraní
 Name[he]=גוארני
 Name[hi]=गौरानी
@@ -3087,17 +3096,19 @@
 Name[zu]=Isi-Hiri Motu
 [hsb]
 Name=Upper Sorbian
+Name[ar]=الصربية العليا
 Name[bs]=Gornji lužičkosrpski
 Name[ca]=Alt sòrab
 Name[cs]=Hornolužicko srbský
 Name[cy]=Sorbieg Uchaf
 Name[da]=Øvresorbisk
+Name[de]=Obersorbisch
 Name[es]=Sorbio alto
 Name[et]=Ülemsorbi
 Name[fr]=Haut Sorabe
 Name[ga]=Sorbais Uachtarach
 Name[hu]=felső szorb
-Name[it]=Alto sorbian
+Name[it]=Alto sorabo
 Name[ja]=ソルブ語
 Name[nb]=Øvresorbisk
 Name[nds]=Böversorbsch
@@ -3112,7 +3123,7 @@
 [hu]
 Name=Hungarian
 Name[af]=Hongaars
-Name[ar]=الهنغارية (المجرية)
+Name[ar]=الهنغارية
 Name[az]=Macarca
 Name[be]=Вугорская
 Name[bg]=Унгарски
@@ -3290,6 +3301,7 @@
 Name[bn]=ইনাক্টিটুট
 Name[fa]=اینوکتیتوت
 Name[fi]=Eskimo
+Name[ga]=Ionúitis
 Name[he]=אינוקטיטוט
 Name[hi]=इनुक्तितु
 Name[hr]=Inuktitutski
@@ -3874,6 +3886,7 @@
 Name[eo]=Kmera
 Name[et]=Khmeeri
 Name[fa]=خمر
+Name[ga]=Címéiris
 Name[he]=חמר
 Name[hi]=ख्मेर
 Name[hr]=Kmerski
@@ -3914,6 +3927,7 @@
 Name[et]=Kikuju
 Name[fa]=کیکویو
 Name[fi]=Kikuju
+Name[ga]=Ciocúis
 Name[he]=קיקויו
 Name[hi]=किकूयू
 Name[hr]=Kikujski
@@ -3991,6 +4005,7 @@
 Name[fa]=قرقیز
 Name[fi]=Kirgiisi
 Name[fr]=Kirghize
+Name[ga]=Cirgeastáinis
 Name[he]=קירגיזית
 Name[hi]=किर्गिज
 Name[hr]=Kirgiski
@@ -4721,6 +4736,7 @@
 Name[et]=Malajalami
 Name[fa]=مالایالامی
 Name[fi]=Malajalam
+Name[ga]=Mailéalaimis
 Name[he]=מליאלאם
 Name[hi]=मलयालम
 Name[hr]=Malajamski
@@ -4812,6 +4828,7 @@
 Name[es]=Maratí
 Name[fa]=مراتی
 Name[fr]=Marathe
+Name[ga]=Maraitis
 Name[he]=מאראתי
 Name[hi]=मराठी
 Name[hr]=Maratski
@@ -5181,11 +5198,13 @@
 Name[zu]=Isi-Nauru
 [nds]
 Name=Low Saxon
+Name[ar]=الساكسونيّة الدنيا
 Name[bs]=Niži saksonski
 Name[ca]=Baix saxó
 Name[cs]=Dolnosaský
 Name[cy]=Sacsoneg Isel
 Name[da]=Nedresaksisk
+Name[de]=Plattdeutsch
 Name[es]=Bajo sajón
 Name[et]=Alamsaksi
 Name[fr]=Bas saxon (ou francique, ou plattdeutsch)
@@ -6306,6 +6325,7 @@
 Name[eu]=Kitxua
 Name[fa]=کوچوا
 Name[fi]=Ketšua
+Name[ga]=Ceatsuais
 Name[he]=קצ'ואה
 Name[hi]=क्वेचुआ
 Name[hu]=kecsua
@@ -6741,6 +6761,7 @@
 Name[fa]=سینهالسیی
 Name[fi]=Singali
 Name[fr]=Singhalais
+Name[ga]=Siolóinis
 Name[he]=סינהלזית
 Name[hi]=सिंहली
 Name[hr]=Sinhalski
@@ -7077,6 +7098,7 @@
 Name[de]=Sindhi-Sprache
 Name[eo]=Sinda
 Name[fa]=سیندهی
+Name[ga]=Sindis
 Name[he]=סינדהי
 Name[hi]=सिंधी
 Name[hu]=szindi
@@ -7400,6 +7422,7 @@
 Name[fa]=سوندانزی
 Name[fi]=Sunda
 Name[fr]=Soudanais
+Name[ga]=Sundais
 Name[he]=סודנית
 Name[hi]=सूडानी
 Name[hu]=szundanéz
@@ -7452,6 +7475,7 @@
 Name[eu]=Swahiliera
 Name[fa]=سواحیلی
 Name[fi]=Suahili
+Name[ga]=Svahaílis
 Name[he]=סוואהילי
 Name[hi]=स्वाहिली
 Name[hr]=Svahili
@@ -7680,6 +7704,7 @@
 Name[eu]=Tatarera
 Name[fa]=تاتاری
 Name[fi]=Tataari
+Name[ga]=Tatairis
 Name[he]=טטרית
 Name[hi]=तातार
 Name[hr]=Tatarski
@@ -7723,6 +7748,7 @@
 Name[eo]=Telugua
 Name[fa]=تلوگویی
 Name[fr]=Télougou
+Name[ga]=Teileagúilis
 Name[he]=טלוגו
 Name[hi]=तेलुगु
 Name[hr]=Telugski
@@ -7937,6 +7963,7 @@
 Name[fa]=تیگرینیایی
 Name[fi]=Tigrinja
 Name[fr]=Tigrigna
+Name[ga]=Tigrinis
 Name[he]=טיגריניה
 Name[hi]=टिग्रिन्या
 Name[hr]=Tigrinski
@@ -8086,6 +8113,7 @@
 Name[fa]=ترکمنی
 Name[fi]=Turkmeeni
 Name[fr]=Turkmène
+Name[ga]=Turcamáinis
 Name[he]=טורקמנית
 Name[hi]=तुर्कमेन
 Name[hr]=Turkmenski
@@ -8697,6 +8725,7 @@
 Name[eo]=Volofa
 Name[et]=Volofi
 Name[fa]=ولفی
+Name[ga]=Volaifis
 Name[he]=וולוף
 Name[hi]=वालाफ़
 Name[hr]=Volofski
diff -urN -x CVS kdelibs.orig/kdecore/eventsrc kdelibs/kdecore/eventsrc
--- kdelibs.orig/kdecore/eventsrc	Fri Nov 12 08:27:50 2004
+++ kdelibs/kdecore/eventsrc	Sat Dec 11 12:18:33 2004
@@ -146,6 +146,7 @@
 Name[zu]=Ukuqedwa kombhalo:ukujikelezisa
 Comment=The end of the list of matches has been reached
 Comment[af]=Die einde van die lys van ooreenkomste is bereik
+Comment[ar]=تم الوصول إلى نهاية قائمة التماثلات
 Comment[bg]=Достигнат е края на списъка от съвпадения
 Comment[bn]=মিলের তালিকার শেষে উপস্থিত
 Comment[bs]=Došao sam do kraja liste pogodaka
@@ -260,6 +261,7 @@
 Name[zu]=Ukuqedwa kokumbhalo: akukho okufanayo
 Comment=No matching completion was found
 Comment[af]=Geen ooreenstemming was gevind
+Comment[ar]=لا توجد تكملة مطابقة
 Comment[bg]=Не е намерено съвпадение
 Comment[bn]=কোনো যুতসই পরিপূরক পাওয়া যায় নি
 Comment[bs]=Nijedno dovršavanje nije pronađeno
@@ -374,6 +376,7 @@
 Name[zu]=Ukuqedwa kokubhaliweyo: ukufaniswa kwesikhashana
 Comment=There is more than one possible match
 Comment[af]=Daar is meer as een moontlik ooreenkoms
+Comment[ar]=هنالك أكثر من تطابق واحد ممكن
 Comment[bg]=Повече от едно съвпадение са намерени
 Comment[bn]=একাধিক সম্ভাব্য পরিপূরক বর্তমান
 Comment[bs]=Postoji više od jednog mogućeg pogotka
@@ -458,7 +461,7 @@
 Name[ms]=Tidak boleh Buka Fail
 Name[mt]=Ma jistax jinfetaħ fajl
 Name[nb]=Kan ikke åpne fil
-Name[nds]=Datei laat sik nich opmaken
+Name[nds]=Datei lett sik nich opmaken
 Name[nl]=Kan het bestand niet openen
 Name[nn]=Kan ikkje opna fil
 Name[nso]=Ekase Bule Faele
@@ -529,7 +532,7 @@
 Comment[ms]=Fail dipilih tidak boleh dibaca atau ditulis
 Comment[mt]=Il-fajl magħżul ma setax jinfetaħ għall-qari jew kitba.
 Comment[nb]=Den valgte fila kan ikke åpnes for lesing eller skriving
-Comment[nds]=De utsöchte Datei laat sik nich lesen oder schrieven
+Comment[nds]=De utsöchte Datei lett sik nich lesen oder schrieven
 Comment[nl]=Het geselecteerde bestand kan niet worden geopend om te lezen of naar te schrijven.
 Comment[nn]=Den valte fila kan ikkje opnast for lesing eller skriving
 Comment[nso]=Faele yeo e kgethilwego ekase bulelwe go bala goba go ngwala
@@ -637,6 +640,7 @@
 Name[zu]=Iphutha Lengozi
 Comment=There was a serious error causing the program to exit
 Comment[af]=Daar was 'n ernstige fout wat veroorsaak het dat die program beïendig is.
+Comment[ar]=لقد كان هناك خطأ كبير أدى الى خروج البرنامج
 Comment[bg]=Появи се сериозна грешка, която предизвика прекъсване на програмата
 Comment[bn]=একটি গুরুতর ত্রুটির দরুণ প্রোগ্রামটি থেমে গেছে
 Comment[bs]=Program je morao završiti zbog ozbiljne greške
@@ -753,6 +757,7 @@
 Name[zu]=Isaziso
 Comment=Something special happened in the program
 Comment[af]=Iets spesiaal het met die program gebeur
+Comment[ar]=حدث شيء خاص في البرنامج
 Comment[bg]=В програмата се случи нещо неочаквано
 Comment[bn]=প্রোগ্রামে বিশেষ কিছু একটা ঘটেছে
 Comment[bs]=Nešto posebno se desilo u programu
@@ -872,6 +877,7 @@
 Name[zu]=Isexwayiso
 Comment=There was an error in the program which may cause problems
 Comment[af]=Daar was 'n fout in die program wat dalk probleme kan veroorsaak
+Comment[ar]=هناك خطأ في البرنامج قد يؤدي الى حدوث مشاكل
 Comment[bg]=Появи се грешка, която може да предизвика проблеми
 Comment[bn]=প্রোগ্রামে একটি ত্রুটি ছিল যার ফলে সমস্যা হতে পারে
 Comment[bs]=Postoji greška u programu koja možda uzrokuje probleme.
@@ -1114,6 +1120,7 @@
 Name[zu]=Ukungena ngaphakathi
 Comment=KDE is starting up
 Comment[af]=Begin KDE Laai
+Comment[ar]=جاري بدء KDE
 Comment[bg]=Стартиране на системата
 Comment[bn]=কে.ডি.ই. চালু হচ্ছে
 Comment[bs]=KDE se pokreće
@@ -1228,6 +1235,7 @@
 Name[zu]=Phumela ngaphandle
 Comment=KDE is exiting
 Comment[af]=KDE is besig om toe te maak
+Comment[ar]=جارى إيقاف KDE
 Comment[bg]=Спиране на системата
 Comment[bn]=কে.ডি.ই. বন্ধ করা হচ্ছে
 Comment[bs]=KDE se zatvara
diff -urN -x CVS kdelibs.orig/kdecore/kaboutdata.cpp kdelibs/kdecore/kaboutdata.cpp
--- kdelibs.orig/kdecore/kaboutdata.cpp	Sat Feb 21 21:05:12 2004
+++ kdelibs/kdecore/kaboutdata.cpp	Sat Dec  4 16:44:55 2004
@@ -102,7 +102,8 @@
   mCopyrightStatement( copyrightStatement ),
   mOtherText( text ),
   mHomepageAddress( homePageAddress ),
-  mBugEmailAddress( bugsEmailAddress )
+  mBugEmailAddress( bugsEmailAddress ),
+  mLicenseText (0)
 {
    d = new KAboutDataPrivate;
    d->productName = 0;
@@ -119,6 +120,8 @@
 
 KAboutData::~KAboutData()
 {
+    if (mLicenseKey == License_File)
+        delete [] mLicenseText;
     delete d;
 }
 
diff -urN -x CVS kdelibs.orig/kdecore/kaccel.cpp kdelibs/kdecore/kaccel.cpp
--- kdelibs.orig/kdecore/kaccel.cpp	Thu May 27 10:43:47 2004
+++ kdelibs/kdecore/kaccel.cpp	Thu Dec  2 09:35:45 2004
@@ -20,6 +20,7 @@
 #include "kaccel.h"
 
 #include <qaccel.h>
+#include <qguardedptr.h>
 #include <qpopupmenu.h>
 #include <qstring.h>
 #include <qtimer.h>
@@ -316,9 +317,12 @@
 					KAccelAction* pAction = m_mapIDToAction[nID];
 					if( !pAction->isEnabled() )
 						continue;
+					QGuardedPtr<KAccelPrivate> me = this;
 					connect( this, SIGNAL(menuItemActivated()), pAction->objSlotPtr(), pAction->methodSlotPtr() );
 					emit menuItemActivated();
-					disconnect( this, SIGNAL(menuItemActivated()), pAction->objSlotPtr(), pAction->methodSlotPtr() );
+					if (me) {
+						disconnect( me, SIGNAL(menuItemActivated()), pAction->objSlotPtr(), pAction->methodSlotPtr() );
+					}
 				} else
 					slotKeyPressed( nID );
 
diff -urN -x CVS kdelibs.orig/kdecore/kdebug.areas kdelibs/kdecore/kdebug.areas
--- kdelibs.orig/kdecore/kdebug.areas	Thu Jun 17 15:28:27 2004
+++ kdelibs/kdecore/kdebug.areas	Mon Nov 29 18:09:51 2004
@@ -459,6 +459,9 @@
 30518        openoffice-filter
 30519        import-export-openoffice-filter-lib
 30520        kword-1.3-filter
+30521        gnumeric-filter
+30522        latex-filter
+30523        quattro-pro-filter
 31000        koffice
 32001        kword
 32002        kword (formatting)
diff -urN -x CVS kdelibs.orig/kdecore/network/kresolvermanager.cpp kdelibs/kdecore/network/kresolvermanager.cpp
--- kdelibs.orig/kdecore/network/kresolvermanager.cpp	Sat Oct  9 02:10:16 2004
+++ kdelibs/kdecore/network/kresolvermanager.cpp	Sun Dec  5 18:25:51 2004
@@ -197,6 +197,8 @@
 
 } resInit;
 
+} // anonymous namespace
+
 /*
  * parameters
  */
@@ -767,5 +769,3 @@
   QMutexLocker locker(&mutex);
   dequeueNew(obj);
 }
-
-} // anonymous namespace
diff -urN -x CVS kdelibs.orig/kdecore/network/ksocketdevice.cpp kdelibs/kdecore/network/ksocketdevice.cpp
--- kdelibs.orig/kdecore/network/ksocketdevice.cpp	Mon Aug 30 02:50:48 2004
+++ kdelibs/kdecore/network/ksocketdevice.cpp	Sun Dec  5 18:25:51 2004
@@ -804,58 +804,58 @@
     bool isNull() const
     { return obj == 0; }
   };
+}
 
-  static KSocketDeviceFactoryBase* defaultImplFactory;
-  static QMutex defaultImplFactoryMutex;
-  typedef QMap<int, KSocketDeviceFactoryBase* > factoryMap;
-  static factoryMap factories;
+static KSocketDeviceFactoryBase* defaultImplFactory;
+static QMutex defaultImplFactoryMutex;
+typedef QMap<int, KSocketDeviceFactoryBase* > factoryMap;
+static factoryMap factories;
  
-  KSocketDevice* KSocketDevice::createDefault(KSocketBase* parent)
-  {
-    KSocketDevice* device = dynamic_cast<KSocketDevice*>(parent);
-    if (device != 0L)
-      return device;
-
-    KSocksSocketDevice::initSocks();
-
-    if (defaultImplFactory)
-      return defaultImplFactory->create(parent);
-
-    // the really default
-    return new KSocketDevice(parent);
-  }
-
-  KSocketDevice* KSocketDevice::createDefault(KSocketBase* parent, int capabilities)
-  {
-    KSocketDevice* device = dynamic_cast<KSocketDevice*>(parent);
-    if (device != 0L)
-      return device;
-
-    QMutexLocker locker(&defaultImplFactoryMutex);
-    factoryMap::ConstIterator it = factories.constBegin();
-    for ( ; it != factories.constEnd(); ++it)
-      if ((it.key() & capabilities) == capabilities)
-	// found a match
-	return it.data()->create(parent);
-
-    return 0L;			// no default
-  }
-
-  KSocketDeviceFactoryBase*
-  KSocketDevice::setDefaultImpl(KSocketDeviceFactoryBase* factory)
-  {
-    QMutexLocker locker(&defaultImplFactoryMutex);
-    KSocketDeviceFactoryBase* old = defaultImplFactory;
-    defaultImplFactory = factory;
-    return old;
-  }
-
-  void KSocketDevice::addNewImpl(KSocketDeviceFactoryBase* factory, int capabilities)
-  {
-    QMutexLocker locker(&defaultImplFactoryMutex);
-    if (factories.contains(capabilities))
-      delete factories[capabilities];
-    factories.insert(capabilities, factory);
-  }
+KSocketDevice* KSocketDevice::createDefault(KSocketBase* parent)
+{
+  KSocketDevice* device = dynamic_cast<KSocketDevice*>(parent);
+  if (device != 0L)
+    return device;
+
+  KSocksSocketDevice::initSocks();
+
+  if (defaultImplFactory)
+    return defaultImplFactory->create(parent);
+
+  // the really default
+  return new KSocketDevice(parent);
+}
+
+KSocketDevice* KSocketDevice::createDefault(KSocketBase* parent, int capabilities)
+{
+  KSocketDevice* device = dynamic_cast<KSocketDevice*>(parent);
+  if (device != 0L)
+    return device;
+
+  QMutexLocker locker(&defaultImplFactoryMutex);
+  factoryMap::ConstIterator it = factories.constBegin();
+  for ( ; it != factories.constEnd(); ++it)
+    if ((it.key() & capabilities) == capabilities)
+      // found a match
+      return it.data()->create(parent);
 
+  return 0L;			// no default
 }
+
+KSocketDeviceFactoryBase*
+KSocketDevice::setDefaultImpl(KSocketDeviceFactoryBase* factory)
+{
+  QMutexLocker locker(&defaultImplFactoryMutex);
+  KSocketDeviceFactoryBase* old = defaultImplFactory;
+  defaultImplFactory = factory;
+  return old;
+}
+
+void KSocketDevice::addNewImpl(KSocketDeviceFactoryBase* factory, int capabilities)
+{
+  QMutexLocker locker(&defaultImplFactoryMutex);
+  if (factories.contains(capabilities))
+    delete factories[capabilities];
+  factories.insert(capabilities, factory);
+}
+
diff -urN -x CVS kdelibs.orig/kdeprint/kdeprintd.desktop kdelibs/kdeprint/kdeprintd.desktop
--- kdelibs.orig/kdeprint/kdeprintd.desktop	Fri Nov 12 08:27:50 2004
+++ kdelibs/kdeprint/kdeprintd.desktop	Sat Dec 11 12:18:34 2004
@@ -76,6 +76,7 @@
 Name[zu]= I-Daemon Yokushicilela kwe-KDE
 Comment=Print daemon for KDE
 Comment[af]=Druk Bediener vir KDE
+Comment[ar]=خادم طباعة لـ KDE
 Comment[bg]=Демон за печат в КДЕ
 Comment[bn]=কে.ডি.ই-র মুদ্রণ ডিমন
 Comment[bs]=Demon za štampu za KDE
diff -urN -x CVS kdelibs.orig/kdeprint/specials.desktop kdelibs/kdeprint/specials.desktop
--- kdelibs.orig/kdeprint/specials.desktop	Sun Nov 28 09:33:03 2004
+++ kdelibs/kdeprint/specials.desktop	Sat Dec 11 12:18:34 2004
@@ -790,6 +790,7 @@
 Icon=kmail
 Name=Mail PDF File
 Name[af]=Pos PDF Lêer
+Name[ar]=ارسال ملف PDF كبريد
 Name[be]=Даслаць PDF файл
 Name[bg]=Изпращане на файл PDF по е-поща
 Name[bn]=পি-ডি-এফ ফাইল মেইল করো
diff -urN -x CVS kdelibs.orig/kdeui/kdetrayproxy/kdetrayproxy.desktop kdelibs/kdeui/kdetrayproxy/kdetrayproxy.desktop
--- kdelibs.orig/kdeui/kdetrayproxy/kdetrayproxy.desktop	Fri Nov 12 08:27:51 2004
+++ kdelibs/kdeui/kdetrayproxy/kdetrayproxy.desktop	Sat Dec 11 12:18:34 2004
@@ -3,6 +3,7 @@
 Type=Service
 Name=KDE Tray Proxy Module
 Name[af]=KDE Skinkbord Proksie Module
+Name[ar]=وحدة تحكّم KDE Tray Proxy
 Name[bn]=কে.ডি.ই. ট্রে প্রক্সি মডিউল
 Name[bs]=KDE proxy modul za tray
 Name[ca]=Mòdul de safata d'intermediari del KDE
@@ -38,6 +39,7 @@
 Name[zh_CN]=KDE 托盘代理模块
 Comment=Proxy enabling KDE systray applications to work without KWin
 Comment[af]=Proksie wat KDE 'systray' programme toelaat om sonder KWin te werk
+Comment[ar]=ملقّم وكيل يسمح لتطبيقات KDE systray بالعمل من دون KWin
 Comment[bg]=Модул, който позволява на програмите от КДЕ да работят в системния поднос с други мениджъри на прозорци
 Comment[bn]=কে.ডি.ই. সিস-ট্রে অ্যাপলিকেশন কে-উইন ছাড়াই কাজ করাবার জন্য প্রক্সি
 Comment[bs]=Proxy koji omogućuje KDE systray programima da rade izvan KWin-a
@@ -56,7 +58,7 @@
 Comment[lt]=Proxy įgalinantis KDE sistemos dėklo taikomasias programas dirbti be KWin
 Comment[mk]=Прокси кое им овозможува на KDE апликациите од сис. лента да работат без KWin
 Comment[nb]=Modul som gjør at systemkurvprogrammet for KDE kan fungere uten KWin
-Comment[nds]=Disse Proxy laat Programmen ok ahn KWin as Finsterpleger binnen den Paneel-Systeemafsnitt lopen
+Comment[nds]=Disse Proxy lett Programmen ok ahn KWin as Finsterpleger binnen den Paneel-Systeemafsnitt lopen
 Comment[nl]=Proxy-module om applicaties in het KDE-systeemvak te plaatsen die geen KWin ondersteuning hebben.
 Comment[nn]=Modul som gjer at systemtrauprogram for KDE kan fungera uavhengig av KWin
 Comment[pa]=KWin ਬਿਨਾਂ ਕੰਮ ਕਰਨ ਲਈ KDE systray ਕਾਰਜਾਂ ਲਈ ਪਰਾਕਸੀ ਯੋਗ
diff -urN -x CVS kdelibs.orig/khtml/ecma/kjs_window.cpp kdelibs/khtml/ecma/kjs_window.cpp
--- kdelibs.orig/khtml/ecma/kjs_window.cpp	Wed Nov 17 14:23:07 2004
+++ kdelibs/khtml/ecma/kjs_window.cpp	Fri Dec 10 12:00:29 2004
@@ -1464,7 +1464,7 @@
       part->xmlDocImpl()->updateRendering();
     if ( part )
       emit part->browserExtension()->requestFocus(part);
-    KMessageBox::error(widget, QStyleSheet::convertFromPlainText(str), caption);
+    KMessageBox::error(widget, str, caption);
     return Undefined();
   case Window::Confirm:
     if (!widget->dialogsAllowed())
@@ -1473,7 +1473,7 @@
       part->xmlDocImpl()->updateRendering();
     if ( part )
       emit part->browserExtension()->requestFocus(part);
-    return Boolean((KMessageBox::warningYesNo(widget, QStyleSheet::convertFromPlainText(str), caption,
+    return Boolean((KMessageBox::warningYesNo(widget, str, caption,
                                                 KStdGuiItem::ok(), KStdGuiItem::cancel()) == KMessageBox::Yes));
   case Window::Prompt:
     if (!widget->dialogsAllowed())
diff -urN -x CVS kdelibs.orig/khtml/html/html_miscimpl.cpp kdelibs/khtml/html/html_miscimpl.cpp
--- kdelibs.orig/khtml/html/html_miscimpl.cpp	Wed Nov 17 14:59:16 2004
+++ kdelibs/khtml/html/html_miscimpl.cpp	Fri Dec  3 11:03:49 2004
@@ -523,6 +523,12 @@
 
 // -----------------------------------------------------------------------------
 
+NodeImpl *HTMLFormCollectionImpl::item( unsigned long index ) const
+{
+    int dummy;
+    return getItem(0L, index, dummy);
+}
+
 unsigned long HTMLFormCollectionImpl::calcLength(NodeImpl*) const
 {
     QPtrList<HTMLGenericFormElementImpl> l = static_cast<HTMLFormElementImpl*>( base )->formElements;
diff -urN -x CVS kdelibs.orig/khtml/html/html_miscimpl.h kdelibs/khtml/html/html_miscimpl.h
--- kdelibs.orig/khtml/html/html_miscimpl.h	Wed Nov 17 14:59:16 2004
+++ kdelibs/khtml/html/html_miscimpl.h	Fri Dec  3 11:03:49 2004
@@ -75,8 +75,8 @@
     virtual ~HTMLCollectionImpl();
     unsigned long length() const;
     // Only when iterating forward, it will use caching, ie. making it O(1).
-    NodeImpl *item ( unsigned long index ) const;
-    // obsolete and not domtree changes save, virtual too.
+    virtual NodeImpl *item ( unsigned long index ) const;
+    // obsolete and not domtree changes save
     virtual NodeImpl *firstItem() const;
     virtual NodeImpl *nextItem() const;
 
@@ -126,6 +126,7 @@
     {}
     ~HTMLFormCollectionImpl() { }
 
+    virtual NodeImpl *item ( unsigned long index ) const;
     virtual NodeImpl *firstItem() const;
     virtual NodeImpl *nextItem() const;
 protected:
diff -urN -x CVS kdelibs.orig/khtml/khtml_part.cpp kdelibs/khtml/khtml_part.cpp
--- kdelibs.orig/khtml/khtml_part.cpp	Wed Nov 17 14:46:56 2004
+++ kdelibs/khtml/khtml_part.cpp	Mon Dec 13 16:40:07 2004
@@ -517,6 +517,10 @@
   d->m_bPluginsEnabled = KHTMLFactory::defaultHTMLSettings()->isPluginsEnabled(url.host());
 
   m_url = url;
+  
+  d->m_restoreScrollPosition = true;
+  disconnect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(restoreScrollPosition()));
+  connect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(restoreScrollPosition()));
 
   KHTMLPageCache::self()->fetchData( d->m_cacheId, this, SLOT(slotRestoreData(const QByteArray &)));
 
@@ -602,25 +606,22 @@
         return true;
     }
   }
-
-  //jump to the anchor AFTER layouting is done, otherwise the position of the
-  //anchor is not known and we have no clue to which coordinates to jump
-  disconnect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(gotoAnchor()));
-  connect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(gotoAnchor()));
-
+  
   // Save offset of viewport when page is reloaded to be compliant
   // to every other capable browser out there.
   if (args.reload) {
     args.xOffset = d->m_view->contentsX();
     args.yOffset = d->m_view->contentsY();
     d->m_extension->setURLArgs(args);
-    disconnect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(gotoAnchor()));
-    connect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(restoreScrollPosition()));
-  }
+  } 
 
   if (!d->m_restored)
     closeURL();
 
+  d->m_restoreScrollPosition = d->m_restored;
+  disconnect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(restoreScrollPosition()));      
+  connect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(restoreScrollPosition()));
+
   // initializing m_url to the new url breaks relative links when opening such a link after this call and _before_ begin() is called (when the first
   // data arrives) (Simon)
   m_url = url;
@@ -1526,15 +1527,11 @@
     if (!baseURL.isEmpty())
       d->m_doc->setBaseURL(KURL( d->m_doc->completeURL(baseURL) ));
 
-
     if ( !m_url.isLocalFile() ) {
         // Support for http last-modified
         d->m_lastModified = d->m_job->queryMetaData("modified");
     } else
         d->m_lastModified = QString::null; // done on-demand by lastModified()
-
-    // Reset contents position
-    connect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(restoreScrollPosition()));
   }
 
   KHTMLPageCache::self()->addData(d->m_cacheId, data);
@@ -2115,12 +2112,6 @@
 
   checkEmitLoadEvent(); // if we didn't do it before
 
-  // check that the view has not been moved by the user
-
-  if ( m_url.encodedHtmlRef().isEmpty() && d->m_view->contentsY() == 0 )
-      d->m_view->setContentsPos( d->m_extension->urlArgs().xOffset,
-                                 d->m_extension->urlArgs().yOffset );
-
   bool pendingAction = false;
 
   if ( !d->m_redirectURL.isEmpty() )
@@ -2359,17 +2350,6 @@
     d->m_doc->setUserStyleSheet( styleSheet );
 }
 
-void KHTMLPart::gotoAnchor()
-{
-  if ( !d->m_doc || !d->m_doc->parsing() ) {
-    disconnect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(gotoAnchor()));
-  }
-
-  if ( m_url.hasRef() )
-    if ( !gotoAnchor(m_url.encodedHtmlRef()) )
-      gotoAnchor(m_url.htmlRef());
-}
-
 bool KHTMLPart::gotoAnchor( const QString &name )
 {
   if (!d->m_doc)
@@ -4758,7 +4738,7 @@
 KHTMLPart::findFrameParent( KParts::ReadOnlyPart *callingPart, const QString &f, khtml::ChildFrame **childFrame )
 {
 #ifdef DEBUG_FINDFRAME
-  kdDebug(6050) << "KHTMLPart::findFrameParent: this = " << this << " URL = " << m_url << " findFrameParent( " << f << " )" << endl;
+  kdDebug(6050) << "KHTMLPart::findFrameParent: this = " << this << " URL = " << m_url << " name = " << name() << " findFrameParent( " << f << " )" << endl;
 #endif
   // Check access
   KHTMLPart* const callingHtmlPart = dynamic_cast<KHTMLPart *>(callingPart);
@@ -4766,6 +4746,9 @@
   if (!checkFrameAccess(callingHtmlPart))
      return 0;
 
+  if (!childFrame && !parentPart() && (name() == f))
+     return this;
+
   FrameIt it = d->m_frames.find( f );
   const FrameIt end = d->m_frames.end();
   if ( it != end )
@@ -5154,8 +5137,6 @@
     args.yOffset = yOffset;
     args.docState = docState;
 
-    connect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(restoreScrollPosition()));
-
     d->m_extension->setURLArgs( args );
     if (!KHTMLPageCache::self()->isComplete(d->m_cacheId))
     {
@@ -6604,6 +6585,14 @@
 {
   KParts::URLArgs args = d->m_extension->urlArgs();
 
+  if ( m_url.hasRef() && !d->m_restoreScrollPosition && !args.reload) {
+    if ( !d->m_doc || !d->m_doc->parsing() )
+      disconnect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(restoreScrollPosition()));
+    if ( !gotoAnchor(m_url.encodedHtmlRef()) )
+      gotoAnchor(m_url.htmlRef());
+    return;
+  }
+  
   // Check whether the viewport has become large enough to encompass the stored
   // offsets. If the document has been fully loaded, force the new coordinates,
   // even if the canvas is too short (can happen when user resizes the window
@@ -6612,7 +6601,7 @@
       || d->m_bComplete) {
     d->m_view->setContentsPos(args.xOffset, args.yOffset);
     disconnect(d->m_view, SIGNAL(finishedLayout()), this, SLOT(restoreScrollPosition()));
-  }
+  }  
 }
 
 
diff -urN -x CVS kdelibs.orig/khtml/khtml_part.h kdelibs/khtml/khtml_part.h
--- kdelibs.orig/khtml/khtml_part.h	Wed Nov 10 10:00:14 2004
+++ kdelibs/khtml/khtml_part.h	Mon Dec 13 16:40:08 2004
@@ -1217,11 +1217,6 @@
                         const QString& boundary = QString::null );
 
 private slots:
-  /**
-   * @internal
-   * Jump to the anchor as defined in the url. Typically connected to KHTMLView::finishedLayout().
-   */
-  void gotoAnchor();
 
   /**
    * @internal
@@ -1455,6 +1450,8 @@
 
   /**
    * @internal
+   * used to restore or reset the view's scroll position (including positioning on anchors)
+   * once a sufficient portion of the document as been laid out.
    */
   void restoreScrollPosition();
 
diff -urN -x CVS kdelibs.orig/khtml/khtml_run.cpp kdelibs/khtml/khtml_run.cpp
--- kdelibs.orig/khtml/khtml_run.cpp	Sat Nov  1 22:14:59 2003
+++ kdelibs/khtml/khtml_run.cpp	Fri Dec 10 12:03:18 2004
@@ -33,6 +33,9 @@
                           false, false, hideErrorDialog ),
   m_child( child )
 {
+    // Don't use an external browser for parts of a webpage we are rendering. (iframes at least are one example)
+    setEnableExternalBrowser(false);
+
     // get the wheel to start spinning
     part->started(0L);
 }
diff -urN -x CVS kdelibs.orig/khtml/khtmlpart_p.h kdelibs/khtml/khtmlpart_p.h
--- kdelibs.orig/khtml/khtmlpart_p.h	Wed Nov 10 10:00:14 2004
+++ kdelibs/khtml/khtmlpart_p.h	Mon Dec 13 16:40:08 2004
@@ -217,6 +217,7 @@
     m_frameNameId = 1;
 
     m_restored = false;
+    m_restoreScrollPosition = false;
 
     m_focusNodeNumber = -1;
     m_focusNodeRestored = false;
@@ -326,6 +327,7 @@
   bool m_metaRefreshEnabled :1;
   bool m_bPluginsOverride :1;
   bool m_restored :1;
+  bool m_restoreScrollPosition :1;
   bool m_statusMessagesEnabled :1;
   bool m_bWalletOpened :1;
   int m_frameNameId;
diff -urN -x CVS kdelibs.orig/khtml/kmultipart/kmultipart.desktop kdelibs/khtml/kmultipart/kmultipart.desktop
--- kdelibs.orig/khtml/kmultipart/kmultipart.desktop	Fri Nov 12 08:27:51 2004
+++ kdelibs/khtml/kmultipart/kmultipart.desktop	Sat Dec 11 12:18:34 2004
@@ -4,6 +4,7 @@
 MimeType=multipart/mixed;multipart/x-mixed-replace
 Name=Embeddable Component for multipart/mixed
 Name[af]=Inlegbare Komponent vir multideel/gemeng
+Name[ar]=مكوّن مدمج لمتعدد الأجزاء أو مختلط.
 Name[bg]=Компонент за вграждане за преглед на съобщения multipart/mixed
 Name[bn]=multipart/mixed-এর জন্য অভ্যন্তরীণ উপাদান
 Name[bs]=Ugradiva komponenta za multipart/mixed
diff -urN -x CVS kdelibs.orig/khtml/rendering/render_block.cpp kdelibs/khtml/rendering/render_block.cpp
--- kdelibs.orig/khtml/rendering/render_block.cpp	Wed Nov 24 15:01:00 2004
+++ kdelibs/khtml/rendering/render_block.cpp	Tue Nov 30 16:05:46 2004
@@ -1038,7 +1038,7 @@
                     int cw = lineWidth( child->yPos() );
                     static_cast<RenderBox*>(child)->calcHorizontalMargins
                         ( child->style()->marginLeft(), child->style()->marginRight(), cw);
-                    
+                    chPos = leftOff + child->marginLeft();                    
                 }
             }
         } else {
diff -urN -x CVS kdelibs.orig/kio/kio/job.cpp kdelibs/kio/kio/job.cpp
--- kdelibs.orig/kio/kio/job.cpp	Tue Nov  9 01:50:35 2004
+++ kdelibs/kio/kio/job.cpp	Wed Dec  8 02:08:15 2004
@@ -3121,10 +3121,10 @@
                            f.close();
                            KSimpleConfig config( path );
                            config.setDesktopGroup();
-                           config.writePathEntry( QString::fromLatin1("URL"), (*it).uSource.url() );
-                           KURL urlName = (*it).uSource;
-                           urlName.setPass( "" );
-                           config.writeEntry( QString::fromLatin1("Name"), urlName.url() );
+                           KURL url = (*it).uSource;
+                           url.setPass( "" );
+                           config.writePathEntry( QString::fromLatin1("URL"), url.url() );
+                           config.writeEntry( QString::fromLatin1("Name"), url.url() );
                            config.writeEntry( QString::fromLatin1("Type"), QString::fromLatin1("Link") );
                            QString protocol = (*it).uSource.protocol();
                            if ( protocol == QString::fromLatin1("ftp") )
@@ -3375,9 +3375,8 @@
                 QString newPath;
                 // Offer overwrite only if the existing thing is a file
                 // If src==dest, use "overwrite-itself"
-                RenameDlg_Mode mode = (RenameDlg_Mode)
-                                      ( ( err == ERR_DIR_ALREADY_EXIST ? 0 :
-                                          ( m_currentSrcURL == dest ) ? M_OVERWRITE_ITSELF : M_OVERWRITE ) );
+                RenameDlg_Mode mode = (RenameDlg_Mode) (
+                                          ( m_currentSrcURL == dest ) ? M_OVERWRITE_ITSELF : M_OVERWRITE );
 
                 if ( m_srcList.count() > 1 )
                     mode = (RenameDlg_Mode) ( mode | M_MULTI | M_SKIP );
diff -urN -x CVS kdelibs.orig/kio/kio/kdirlister.cpp kdelibs/kio/kio/kdirlister.cpp
--- kdelibs.orig/kio/kio/kdirlister.cpp	Sat Jul  3 00:54:42 2004
+++ kdelibs/kio/kio/kdirlister.cpp	Sun Dec 12 02:17:31 2004
@@ -1883,6 +1883,9 @@
 bool KDirLister::matchesMimeFilter( const KFileItem *item ) const
 {
   Q_ASSERT( item );
+  // Don't lose time determining the mimetype if there is no filter
+  if ( d->mimeFilter.isEmpty() && d->mimeExcludeFilter.isEmpty() )
+      return true;
   return matchesMimeFilter( item->mimetype() );
 }
 
diff -urN -x CVS kdelibs.orig/kio/kio/kprotocolmanager.cpp kdelibs/kio/kio/kprotocolmanager.cpp
--- kdelibs.orig/kio/kio/kprotocolmanager.cpp	Thu Apr 15 16:49:05 2004
+++ kdelibs/kio/kio/kprotocolmanager.cpp	Wed Dec  1 15:32:17 2004
@@ -72,8 +72,8 @@
 
 // DEFAULT USERAGENT STRING
 #define CFG_DEFAULT_UAGENT(X) \
-QString("Mozilla/5.0 (compatible; Konqueror/%1.%2%3) (KHTML, like Gecko)") \
-        .arg(KDE_VERSION_MAJOR).arg(KDE_VERSION_MINOR).arg(X)
+QString("Mozilla/5.0 (compatible; Konqueror/%1.%2%3) KHTML/%4.%5.%6 (like Gecko)") \
+        .arg(KDE_VERSION_MAJOR).arg(KDE_VERSION_MINOR).arg(X).arg(KDE_VERSION_MAJOR).arg(KDE_VERSION_MINOR).arg(KDE_VERSION_RELEASE)
 
 void KProtocolManager::reparseConfiguration()
 {
diff -urN -x CVS kdelibs.orig/kio/kio/krun.cpp kdelibs/kio/kio/krun.cpp
--- kdelibs.orig/kio/kio/krun.cpp	Thu Sep 30 12:14:18 2004
+++ kdelibs/kio/kio/krun.cpp	Sat Dec 11 12:46:53 2004
@@ -65,6 +65,7 @@
     bool m_runExecutables;
 
     QString m_preferredService;
+    QString m_externalBrowser;
     QGuardedPtr <QWidget> m_window;
 };
 
@@ -707,6 +708,7 @@
   d = new KRunPrivate;
   d->m_runExecutables = true;
   d->m_window = window;
+  setEnableExternalBrowser(true);
 
   // Start the timer. This means we will return to the event
   // loop and do initialization afterwards.
@@ -750,8 +752,7 @@
   QString exec;
   if (m_strURL.protocol().startsWith("http"))
   {
-    KConfigGroup cfg(KGlobal::config(), "General");
-    exec = cfg.readEntry("BrowserApplication");
+    exec = d->m_externalBrowser;
   }
 
   if ( m_bIsLocalFile )
@@ -1124,6 +1125,14 @@
   m_timer.start( 0, true );
 }
 
+void KRun::setEnableExternalBrowser(bool b)
+{
+   if (b)
+      d->m_externalBrowser = KConfigGroup(KGlobal::config(), "General").readEntry("BrowserApplication");
+   else
+      d->m_externalBrowser = QString::null;
+}
+
 void KRun::setPreferredService( const QString& desktopEntryName )
 {
     d->m_preferredService = desktopEntryName;
diff -urN -x CVS kdelibs.orig/kio/kio/krun.h kdelibs/kio/kio/krun.h
--- kdelibs.orig/kio/kio/krun.h	Fri May  7 20:05:18 2004
+++ kdelibs/kio/kio/krun.h	Sat Dec 11 12:46:53 2004
@@ -161,6 +161,14 @@
   void setRunExecutables(bool b);
 
   /**
+   * Sets whether the external webbrowser setting should be honoured.
+   * This is enabled by default.
+   * This should only be disabled in webbrowser applications.
+   * @since 3.4
+   */
+  void setEnableExternalBrowser(bool b);
+
+  /**
    * Open a list of URLs with a certain service (application).
    *
    * @param _service the service to run
diff -urN -x CVS kdelibs.orig/kio/kpasswdserver.desktop kdelibs/kio/kpasswdserver.desktop
--- kdelibs.orig/kio/kpasswdserver.desktop	Fri Nov 12 08:27:52 2004
+++ kdelibs/kio/kpasswdserver.desktop	Sat Dec 11 12:18:35 2004
@@ -69,6 +69,7 @@
 Name[zu]=Ingxenye Yegama lokudlula le-KDED
 Comment=Password caching support
 Comment[af]=Wagwoord Kasberging ondersteuning
+Comment[ar]=دعم تخزين كلمة المرور بصورة مؤقّتة
 Comment[bg]=Кеширане на пароли
 Comment[bn]=পাসওয়ার্ড ক্যাশ-এ রাখা সমর্থন করে
 Comment[bs]=Podrška za keširanje šifre
diff -urN -x CVS kdelibs.orig/kio/misc/kpac/eventsrc kdelibs/kio/misc/kpac/eventsrc
--- kdelibs.orig/kio/misc/kpac/eventsrc	Mon Nov 15 08:54:58 2004
+++ kdelibs/kio/misc/kpac/eventsrc	Sat Dec 11 12:18:35 2004
@@ -271,7 +271,7 @@
 Comment[mn]=Итгэмжилэгч тохиргооны скрипт татагдахгүй байна
 Comment[ms]=Skrip penyelarasan proksi tidak boleh dimuat turun
 Comment[nb]=Klarte ikke å laste ned skiptet for mellomtjener-innstillinger
-Comment[nds]=Dat Skript för't Instellen vun den Proxy laat sik nich daalladen
+Comment[nds]=Dat Skript för't Instellen vun den Proxy lett sik nich daalladen
 Comment[nl]=Het proxyconfiguratiescript kon niet worden opgehaald
 Comment[nn]=Klarte ikkje lasta ned skript for mellomtenaroppsett
 Comment[pa]=ਪਰਾਕਸੀ ਸੰਰਚਨਾ ਸਕ੍ਰਿਪਟ ਨੂੰ ਲੋਡ ਨਹੀਂ ਕੀਤਾ ਜਾ ਸਕਦਾ ਹੈ
diff -urN -x CVS kdelibs.orig/kio/misc/kpac/proxyscout.desktop kdelibs/kio/misc/kpac/proxyscout.desktop
--- kdelibs.orig/kio/misc/kpac/proxyscout.desktop	Mon Nov 15 08:54:58 2004
+++ kdelibs/kio/misc/kpac/proxyscout.desktop	Sat Dec 11 12:18:35 2004
@@ -40,6 +40,7 @@
 Name[zh_CN]=代理搜索
 Comment=Automatic proxy configuration
 Comment[af]=Outomatiese proksie opstelling
+Comment[ar]=تهيئة تلقائيّة للخادم الوكيل
 Comment[bg]=Автоматично конфигуриране на прокси сървъра
 Comment[bn]=স্বয়ংক্রিয় প্রক্সি কনফিগারেশন
 Comment[bs]=Automatsko podešavanje proxy-ja
diff -urN -x CVS kdelibs.orig/kio/misc/kssld/kssld.desktop kdelibs/kio/misc/kssld/kssld.desktop
--- kdelibs.orig/kio/misc/kssld/kssld.desktop	Fri Nov 12 08:27:52 2004
+++ kdelibs/kio/misc/kssld/kssld.desktop	Sat Dec 11 12:18:35 2004
@@ -72,6 +72,7 @@
 Name[zu]=Ingxenye ye-daemon ye-KSSL
 Comment=KSSL daemon module for KDED
 Comment[af]=KSSL bediener module vir KDED
+Comment[ar]=وحدة خادم KSSL لـ KDED
 Comment[bg]=Модул демон за KSSL за KDED
 Comment[bn]=KDED-র জন্য KSSL ডিমন মডিউল 
 Comment[bs]=KSSL Daemon Module za KDED
diff -urN -x CVS kdelibs.orig/kio/misc/kwalletd/kwalletd.desktop kdelibs/kio/misc/kwalletd/kwalletd.desktop
--- kdelibs.orig/kio/misc/kwalletd/kwalletd.desktop	Fri Nov 12 08:27:52 2004
+++ kdelibs/kio/misc/kwalletd/kwalletd.desktop	Sat Dec 11 12:18:36 2004
@@ -65,6 +65,7 @@
 Name[zh_TW]=KWAllet 服務程式模組
 Comment=KWallet daemon module for KDED
 Comment[af]=KBeursie bediener bodule vir KDED
+Comment[ar]=وحدة خادم KWallet لـ KDED
 Comment[bg]=Модул демон за системата Портфейл за KDED
 Comment[bn]=KDED-র জন্য কে-ওয়ালেট ডিমন মডিউল 
 Comment[bs]=KWallet daemon modul za KDED
diff -urN -x CVS kdelibs.orig/kioslave/http/kcookiejar/kcookiejar.desktop kdelibs/kioslave/http/kcookiejar/kcookiejar.desktop
--- kdelibs.orig/kioslave/http/kcookiejar/kcookiejar.desktop	Fri Nov 12 08:27:53 2004
+++ kdelibs/kioslave/http/kcookiejar/kcookiejar.desktop	Sat Dec 11 12:18:36 2004
@@ -63,6 +63,7 @@
 Name[zu]=Ingxenye Yojeke ye-Cookie ye-KDED
 Comment=Keeps track of all cookies in the system
 Comment[af]=Hou tred van al die koekies in die stelsel
+Comment[ar]=تتبّع كل ملفّات الـ cookies على النظام
 Comment[bg]=Контрол над всички кукита в системата
 Comment[bn]=সিস্টেমে সমস্ত কুকি-র খোঁজখবর রাখে
 Comment[bs]=Prati sve kolačiće (cookije) na sistemu
diff -urN -x CVS kdelibs.orig/kresources/kresources.desktop kdelibs/kresources/kresources.desktop
--- kdelibs.orig/kresources/kresources.desktop	Fri Nov 12 08:27:53 2004
+++ kdelibs/kresources/kresources.desktop	Sat Dec 11 12:18:36 2004
@@ -74,6 +74,7 @@
 
 Comment=Configure KDE Resources
 Comment[af]=Stel KDE Hulpbronne op
+Comment[ar]=إعداد موارد KDE
 Comment[be]=Настроіць рэсурсы KDE
 Comment[bg]=Настройки на ресурсите на КДЕ
 Comment[bn]=কে.ডি.ই. রিসোর্সসমূহ কনফিগার করো
diff -urN -x CVS kdelibs.orig/ksmartcard/kardimpl/kcardgemsafe.desktop kdelibs/ksmartcard/kardimpl/kcardgemsafe.desktop
--- kdelibs.orig/ksmartcard/kardimpl/kcardgemsafe.desktop	Fri Nov 12 08:27:53 2004
+++ kdelibs/ksmartcard/kardimpl/kcardgemsafe.desktop	Sat Dec 11 12:18:36 2004
@@ -10,6 +10,7 @@
 X-KDE-Smartcard-SubSubType=gem3
 Name=GemSafe Card Implementation
 Name[af]=Gemsafe Kaart implementering
+Name[ar]=GemSafe Card تطبيق
 Name[bg]=Реализация на карта GemSafe
 Name[bn]=জেম-সেফ কার্ড ইমপ্লিমেন্টেশন
 Name[bs]=GemSafe Card implementacija
diff -urN -x CVS kdelibs.orig/ksmartcard/kardimpl/kcardgsm.desktop kdelibs/ksmartcard/kardimpl/kcardgsm.desktop
--- kdelibs.orig/ksmartcard/kardimpl/kcardgsm.desktop	Fri Nov 12 08:27:53 2004
+++ kdelibs/ksmartcard/kardimpl/kcardgsm.desktop	Sat Dec 11 12:18:36 2004
@@ -10,6 +10,7 @@
 X-KDE-Smartcard-SubSubType=gsm3
 Name=GSM Card Implementation
 Name[af]=GSM Kaart implementering
+Name[ar]=GSM Card تطبيق
 Name[bg]=Реализация на карта GSM
 Name[bn]=জি-এস-এম কার্ড ইমপ্লিমেন্টেশন
 Name[bs]=GSM Card implementacija
diff -urN -x CVS kdelibs.orig/kspell2/Makefile.am kdelibs/kspell2/Makefile.am
--- kdelibs.orig/kspell2/Makefile.am	Tue May  4 05:44:31 2004
+++ kdelibs/kspell2/Makefile.am	Sun Dec  5 01:54:31 2004
@@ -1,24 +1,23 @@
-SUBDIRS = ui . tests plugins
+SUBDIRS = . ui tests plugins
 METASOURCES = AUTO
 
-lib_LTLIBRARIES = libkspell2.la
+noinst_LTLIBRARIES = libkspell2base.la
 
 AM_CPPFLAGS = -DKDE_NO_COMPAT -DQT_NO_COMPAT -I$(top_srcdir)/kutils $(all_includes)
 
-kspell2includedir = $(includedir)/kspell2
+kspell2baseincludedir = $(includedir)/kspell2
 
-kspell2include_HEADERS = settings.h filter.h \
+kspell2baseinclude_HEADERS = settings.h filter.h \
 	dictionary.h broker.h \
 	backgroundchecker.h defaultdictionary.h 
 
-libkspell2_la_SOURCES = settings.cpp broker.cpp \
+libkspell2base_la_SOURCES = settings.cpp broker.cpp \
 	client.cpp filter.cpp backgroundchecker.cpp \
 	backgroundthread.cpp backgroundengine.cpp \
 	defaultdictionary.cpp 
 
-libkspell2_la_LDFLAGS = -no-undefined -version-info 1:0:0 $(all_libraries)
-libkspell2_la_LIBADD  = $(LIB_KPARTS) ../kutils/libkutils.la ./ui/libkspell2ui.la 
+libkspell2base_la_LDFLAGS = -no-undefined $(all_libraries)
+libkspell2base_la_LIBADD  = $(LIB_KPARTS) ../kutils/libkutils.la
 
 servicetype_DATA = kspellclient.desktop
 servicetypedir = $(kde_servicetypesdir)
-KDE_OPTIONS = nofinal
diff -urN -x CVS kdelibs.orig/kspell2/kspellclient.desktop kdelibs/kspell2/kspellclient.desktop
--- kdelibs.orig/kspell2/kspellclient.desktop	Fri Nov 12 08:27:53 2004
+++ kdelibs/kspell2/kspellclient.desktop	Sat Dec 11 12:18:37 2004
@@ -5,6 +5,7 @@
 X-KDE-Derived=KPluginInfo
 Comment=KSpell Client
 Comment[af]=KSpel Klient
+Comment[ar]=عميل KSpell
 Comment[be]=Кліент KSpell
 Comment[bg]=Клиент на KSpell
 Comment[bn]=কে-স্পেল ক্লায়েন্ট
diff -urN -x CVS kdelibs.orig/kspell2/plugins/aspell/Makefile.am kdelibs/kspell2/plugins/aspell/Makefile.am
--- kdelibs.orig/kspell2/plugins/aspell/Makefile.am	Sat May  1 07:32:17 2004
+++ kdelibs/kspell2/plugins/aspell/Makefile.am	Sun Dec  5 01:54:31 2004
@@ -7,7 +7,7 @@
 kspell_aspell_la_SOURCES = kspell_aspellclient.cpp kspell_aspelldict.cpp
 
 kspell_aspell_la_LDFLAGS = -module -no-undefined $(KDE_PLUGIN)
-kspell_aspell_la_LIBADD = ../../libkspell2.la -laspell
+kspell_aspell_la_LIBADD = ../../ui/libkspell2.la -laspell
 
 service_DATA = kspell_aspell.desktop
 servicedir = $(kde_servicesdir)
diff -urN -x CVS kdelibs.orig/kspell2/plugins/ispell/Makefile.am kdelibs/kspell2/plugins/ispell/Makefile.am
--- kdelibs.orig/kspell2/plugins/ispell/Makefile.am	Sat May  1 07:32:17 2004
+++ kdelibs/kspell2/plugins/ispell/Makefile.am	Sun Dec  5 01:54:31 2004
@@ -14,7 +14,7 @@
 	ispell_checker.cpp
 
 kspell_ispell_la_LDFLAGS = -module -no-undefined $(KDE_PLUGIN)
-kspell_ispell_la_LIBADD = ../../libkspell2.la 
+kspell_ispell_la_LIBADD = ../../ui/libkspell2.la
 
 service_DATA = kspell_ispell.desktop
 servicedir = $(kde_servicesdir)
diff -urN -x CVS kdelibs.orig/kspell2/tests/Makefile.am kdelibs/kspell2/tests/Makefile.am
--- kdelibs.orig/kspell2/tests/Makefile.am	Sat May  1 07:48:20 2004
+++ kdelibs/kspell2/tests/Makefile.am	Sun Dec  5 01:54:31 2004
@@ -6,19 +6,19 @@
 	test_dialog test_highlighter test_configdialog
 
 test_SOURCES = test.cpp
-test_LDADD = ../libkspell2.la $(LIB_KDECORE)
+test_LDADD = ../ui/libkspell2.la $(LIB_KDECORE)
 
 test_filter_SOURCES = test_filter.cpp
-test_filter_LDADD = ../libkspell2.la $(LIB_KDECORE)
+test_filter_LDADD = ../ui/libkspell2.la $(LIB_KDECORE)
 
 backgroundtest_SOURCES = backgroundtest.cpp
-backgroundtest_LDADD = ../libkspell2.la $(LIB_KDECORE)
+backgroundtest_LDADD = ../ui/libkspell2.la $(LIB_KDECORE)
 
 test_dialog_SOURCES = test_dialog.cpp
-test_dialog_LDADD = ../libkspell2.la $(LIB_KDECORE)
+test_dialog_LDADD = ../ui/libkspell2.la $(LIB_KDECORE)
 
 test_highlighter_SOURCES = test_highlighter.cpp
-test_highlighter_LDADD = ../libkspell2.la $(LIB_KDECORE)
+test_highlighter_LDADD = ../ui/libkspell2.la $(LIB_KDECORE)
 
 test_configdialog_SOURCES = test_configdialog.cpp
-test_configdialog_LDADD = ../libkspell2.la $(LIB_KDECORE)
+test_configdialog_LDADD = ../ui/libkspell2.la $(LIB_KDECORE)
diff -urN -x CVS kdelibs.orig/kspell2/ui/Makefile.am kdelibs/kspell2/ui/Makefile.am
--- kdelibs.orig/kspell2/ui/Makefile.am	Thu May 13 19:47:50 2004
+++ kdelibs/kspell2/ui/Makefile.am	Sun Dec  5 01:54:32 2004
@@ -1,6 +1,6 @@
 METASOURCES = AUTO
 
-noinst_LTLIBRARIES = libkspell2ui.la
+lib_LTLIBRARIES = libkspell2.la
 
 KDE_CXXFLAGS = -DKDE_NO_COMPAT -DQT_NO_COMPAT 
 INCLUDES = -I$(top_srcdir)/kspell2 $(all_includes)
@@ -9,8 +9,9 @@
 kspell2include_HEADERS = dialog.h highlighter.h configdialog.h \
 	configwidget.h
 
-libkspell2ui_la_SOURCES = kspell2ui.ui dialog.cpp \
+libkspell2_la_SOURCES = kspell2ui.ui dialog.cpp \
 	highlighter.cpp configui.ui configdialog.cpp \
 	configwidget.cpp
 
-libkspell2ui_la_LDFLAGS = -no-undefined  $(all_libraries)
+libkspell2_la_LDFLAGS = -no-undefined -version-info 1:0:0 $(all_libraries)
+libkspell2_la_LIBADD = ../libkspell2base.la
diff -urN -x CVS kdelibs.orig/kutils/kplugininfo.desktop kdelibs/kutils/kplugininfo.desktop
--- kdelibs.orig/kutils/kplugininfo.desktop	Sat Oct 30 08:38:19 2004
+++ kdelibs/kutils/kplugininfo.desktop	Sat Dec 11 12:18:37 2004
@@ -23,6 +23,7 @@
 Name[fa]=اطلاعات متصل‌کننده‌ی KDE
 Name[fi]=KDE:n lisäosatiedot
 Name[fr]=Information sur le module KDE
+Name[ga]=Eolas Bhreiseán KDE
 Name[gl]=Información de Plugin de KDE
 Name[he]=מידע על תוספים של KDE
 Name[hi]=केडीई प्लगिन जानकारी
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.mozilla.xul+xml.desktop kdelibs/mimetypes/application/vnd.mozilla.xul+xml.desktop
--- kdelibs.orig/mimetypes/application/vnd.mozilla.xul+xml.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/vnd.mozilla.xul+xml.desktop	Sat Dec 11 12:18:37 2004
@@ -7,6 +7,7 @@
 Patterns=*.xul;*.XUL;
 Comment=Mozilla XUL File
 Comment[af]=Mozilla XUL Lêer
+Comment[ar]=ملف Mozilla XUL
 Comment[be]=Файл Mozilla XUL
 Comment[bg]=Файл XUL на Мозила
 Comment[bn]=মোজিলা এক্স-ইউ-এল ফাইল
@@ -19,6 +20,7 @@
 Comment[et]=Mozilla XUL fail
 Comment[fi]=Mozilla XUL-tiedosto
 Comment[fr]=Fichier XUL Mozilla
+Comment[ga]=Comhad Mozilla XUL
 Comment[he]=קובץ XUL של מוזילה
 Comment[hu]=Mozilla XUL-fájl
 Comment[is]=Mozilla XUL skrá
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.stardivision.chart.desktop kdelibs/mimetypes/application/vnd.stardivision.chart.desktop
--- kdelibs.orig/mimetypes/application/vnd.stardivision.chart.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/vnd.stardivision.chart.desktop	Sat Dec 11 12:18:37 2004
@@ -6,6 +6,7 @@
 Icon=kchart_chrt
 Comment=StarOffice Chart
 Comment[af]=StarOffice Kaart
+Comment[ar]=تخطيط StarOffice
 Comment[bg]=Документ на StarOffice Chart
 Comment[bn]=স্টার-অফিস চার্ট
 Comment[bs]=StarOffice grafikon
@@ -18,6 +19,7 @@
 Comment[es]=Hoja de cálculo de StarOffice
 Comment[fi]=StarOffice-kaavio
 Comment[fr]=Diagramme StarOffice
+Comment[ga]=Cairt StarOffice
 Comment[hu]=StarOfice-diagram
 Comment[is]=StarOffice tafla
 Comment[it]=Diagramma StarOffice
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.stardivision.draw.desktop kdelibs/mimetypes/application/vnd.stardivision.draw.desktop
--- kdelibs.orig/mimetypes/application/vnd.stardivision.draw.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/vnd.stardivision.draw.desktop	Sat Dec 11 12:18:37 2004
@@ -20,6 +20,7 @@
 Comment[et]=StarOffice joonis
 Comment[fi]=StarOffice-piirros
 Comment[fr]=Dessin StarOffice
+Comment[ga]=Dearadh StarOffice
 Comment[he]=ציור של StarOffice
 Comment[hu]=StarOffice-rajz
 Comment[is]=StarOffice teikning
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.stardivision.math.desktop kdelibs/mimetypes/application/vnd.stardivision.math.desktop
--- kdelibs.orig/mimetypes/application/vnd.stardivision.math.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/vnd.stardivision.math.desktop	Sat Dec 11 12:18:37 2004
@@ -6,6 +6,7 @@
 Icon=kformula_kfo
 Comment=StarOffice Math Document
 Comment[af]=StarOffice Wiskunde Dokument
+Comment[ar]=مستند StarOffice Math
 Comment[be]=Дакумэнт StarOffice Math
 Comment[bg]=Документ на StarOffice Math
 Comment[bn]=স্টার-অফিস ম্যাথ নথী
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.stardivision.writer-global.desktop kdelibs/mimetypes/application/vnd.stardivision.writer-global.desktop
--- kdelibs.orig/mimetypes/application/vnd.stardivision.writer-global.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/vnd.stardivision.writer-global.desktop	Sat Dec 11 12:18:37 2004
@@ -6,6 +6,7 @@
 Icon=wordprocessing
 Comment=StarOffice Writer Master Document
 Comment[af]=StarOffice Writer Meester Dokument
+Comment[ar]=مستند StarOffice Writer Master
 Comment[bg]=Главен документ на StarOffice Writer
 Comment[bn]=স্টার-অফিস রাইটার মাস্টার নথী
 Comment[bs]=StarOffice Writer glavni dokument
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.sun.xml.calc.desktop kdelibs/mimetypes/application/vnd.sun.xml.calc.desktop
--- kdelibs.orig/mimetypes/application/vnd.sun.xml.calc.desktop	Sat Oct 30 08:38:20 2004
+++ kdelibs/mimetypes/application/vnd.sun.xml.calc.desktop	Sat Dec 11 12:18:37 2004
@@ -22,6 +22,7 @@
 Comment[fa]=صفحه‌گسترده‌ی OpenOffice.org
 Comment[fi]=OpenOffice.org-taulukko
 Comment[fr]=Feuille de calcul OpenOffice.org
+Comment[ga]=Scarbhileog OpenOffice.org
 Comment[gl]=Folla de Cálculo de OpenOffice.org
 Comment[he]=גיליון אלקטרוני של OpenOffice.org
 Comment[hi]=स्टारआफिस.ऑर्ग स्प्रेडशीट
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.sun.xml.calc.template.desktop kdelibs/mimetypes/application/vnd.sun.xml.calc.template.desktop
--- kdelibs.orig/mimetypes/application/vnd.sun.xml.calc.template.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/vnd.sun.xml.calc.template.desktop	Sat Dec 11 12:18:37 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Comment=OpenOffice.org Spreadsheet Template
 Comment[af]=OpenOffice.org Spreiblad Voorbeeld
+Comment[ar]=قالب OpenOffice.org Spreadsheet
 Comment[be]=Шаблён электроннай табліцы OpenOffice.org
 Comment[bg]=Шаблон на OpenOffice.org Spreadsheet
 Comment[bn]=ওপেন-অফিস.অর্গ স্প্রেডশিট টেম্‌প্লেট
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.sun.xml.draw.desktop kdelibs/mimetypes/application/vnd.sun.xml.draw.desktop
--- kdelibs.orig/mimetypes/application/vnd.sun.xml.draw.desktop	Mon Nov 22 09:34:04 2004
+++ kdelibs/mimetypes/application/vnd.sun.xml.draw.desktop	Sat Dec 11 12:18:37 2004
@@ -21,6 +21,7 @@
 Comment[fa]=ترسیم OpenOffice.org
 Comment[fi]=OpenOffice.org-piirros
 Comment[fr]=Dessin OpenOffice.org
+Comment[ga]=Dearadh OpenOffice.org
 Comment[gl]=Debuxo de OpenOffice.org
 Comment[he]=ציור של OpenOffice.org
 Comment[hi]=ओपनआफिस.ऑर्ग ड्राइंग
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.sun.xml.draw.template.desktop kdelibs/mimetypes/application/vnd.sun.xml.draw.template.desktop
--- kdelibs.orig/mimetypes/application/vnd.sun.xml.draw.template.desktop	Mon Nov 15 08:54:58 2004
+++ kdelibs/mimetypes/application/vnd.sun.xml.draw.template.desktop	Sat Dec 11 12:18:37 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Comment=OpenOffice.org Drawing Template
 Comment[af]=OpenOffice.org Tekening Voorbeeld
+Comment[ar]=قالب OpenOffice.org Drawing
 Comment[be]=Шаблён малюнку OpenOffice.org
 Comment[bg]=Шаблон на OpenOffice.org Drawing
 Comment[bn]=ওপেন-অফিস.অর্গ ড্রয়িং টেম্‌প্লেট
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.sun.xml.impress.template.desktop kdelibs/mimetypes/application/vnd.sun.xml.impress.template.desktop
--- kdelibs.orig/mimetypes/application/vnd.sun.xml.impress.template.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/vnd.sun.xml.impress.template.desktop	Sat Dec 11 12:18:37 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Comment=OpenOffice.org Presentation Template
 Comment[af]=OpenOffice.org Voorlegging 
+Comment[ar]=قالب OpenOffice.org Presentation
 Comment[be]=Шаблён прэзэнтацыі OpenOffice.org
 Comment[bg]=Шаблон на OpenOffice.org Presentation
 Comment[bn]=ওপেন-অফিস.অর্গ প্রেসেন্টেশন টেম্‌প্লেট
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.sun.xml.writer.master.desktop kdelibs/mimetypes/application/vnd.sun.xml.writer.master.desktop
--- kdelibs.orig/mimetypes/application/vnd.sun.xml.writer.master.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/vnd.sun.xml.writer.master.desktop	Sat Dec 11 12:18:37 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Comment=OpenOffice.org Master Text Document
 Comment[af]=OpenOffice.org Meester Teks Dokument
+Comment[ar]=مستند نصّي OpenOffice.org Master
 Comment[bg]=Главен документ на OpenOffice.org Writer
 Comment[bn]=ওপেন-অফিস.অর্গ মাস্টার টেক্সট নথী
 Comment[bs]=OpenOffice.org glavni tekst dokument
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.sun.xml.writer.template.desktop kdelibs/mimetypes/application/vnd.sun.xml.writer.template.desktop
--- kdelibs.orig/mimetypes/application/vnd.sun.xml.writer.template.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/vnd.sun.xml.writer.template.desktop	Sat Dec 11 12:18:37 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Comment=OpenOffice.org Text Document Template
 Comment[af]=OpenOffice.org Teks Dokument Voorbeeld
+Comment[ar]=قالب مستند نصّي OpenOffice.org 
 Comment[be]=Шаблён тэкставы дакумэнт OpenOffice.org
 Comment[bg]=Шаблон на OpenOffice.org Text Document
 Comment[bn]=ওপেন-অফিস.অর্গ টেক্সট নথী টেম্‌প্লেট
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-bz2dvi.desktop kdelibs/mimetypes/application/x-bz2dvi.desktop
--- kdelibs.orig/mimetypes/application/x-bz2dvi.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/x-bz2dvi.desktop	Sat Dec 11 12:18:37 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Comment=Bzip2 Compressed DVI File
 Comment[af]=Bzip2 gekompakteerde DVI Lêer
+Comment[ar]=ملف DVI مضغوط بـ Bzip2
 Comment[bg]=Компресиран файл DVI с bzip2
 Comment[bn]=বি-জিপ-২ (Bzip2) কম্প্রেসকৃত ডি-ভি-আই ফাইল
 Comment[bs]=Bzip2 komprimirana DVI datoteka
@@ -13,6 +14,7 @@
 Comment[et]=Bzip2 tihendusega DVI fail
 Comment[fi]=Bzip2-pakattu DVI-tiedosto
 Comment[fr]=Fichier DVI compacté avec Bzip2
+Comment[ga]=Comhad DVI comhbhrúite le bzip2
 Comment[he]=קובץ DVI דחוס באמצעות Bzip2
 Comment[hu]=Bzip2-vel tömörített DVI-fájl
 Comment[is]=Bzip2 þjöppuð DVI skrá
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-executable-script.desktop kdelibs/mimetypes/application/x-executable-script.desktop
--- kdelibs.orig/mimetypes/application/x-executable-script.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/x-executable-script.desktop	Sat Dec 11 12:18:38 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Comment=Script (possibly executable)
 Comment[af]=Skrip (moointlik uitvoerbaar)
+Comment[ar]=Script (ربما قابلة للتنفيذ)
 Comment[bg]=Скрипт (вероятно изпълним)
 Comment[bn]=স্ক্রিপ্ট (সম্ভবত এক্সিকিউটেবল)
 Comment[bs]=Skripta (možda izvršiva)
@@ -13,6 +14,7 @@
 Comment[et]=Käivitatav skript
 Comment[fi]=Skripti (mahdollisesti suoritettava)
 Comment[fr]=Script (peut-être exécutable)
+Comment[ga]=Script (b'fhéidir inrite)
 Comment[he]=תסריט הנראה כניתן להרצה
 Comment[hu]=Szkript (általában futtatható)
 Comment[is]=Skeljaforrit (sem er ef til vill keyranlegt)
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-executable.desktop kdelibs/mimetypes/application/x-executable.desktop
--- kdelibs.orig/mimetypes/application/x-executable.desktop	Sat Oct 30 08:38:21 2004
+++ kdelibs/mimetypes/application/x-executable.desktop	Sat Dec 11 12:18:38 2004
@@ -36,7 +36,7 @@
 Comment[ms]=Fail Boleh Laksana
 Comment[mt]=Programm
 Comment[nb]=Program
-Comment[nds]=Datei, laat sik utföhren
+Comment[nds]=Datei, lett sik utföhren
 Comment[nl]=Uitvoerbaar bestand
 Comment[nn]=Køyrbar fil
 Comment[nso]=Faele yeo e Phethagatsegago
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-gzdvi.desktop kdelibs/mimetypes/application/x-gzdvi.desktop
--- kdelibs.orig/mimetypes/application/x-gzdvi.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/x-gzdvi.desktop	Sat Dec 11 12:18:38 2004
@@ -2,6 +2,7 @@
 Encoding=UTF-8
 Comment=Gzip Compressed DVI File
 Comment[af]=Gzip Gekompakteerde DVI Lêer
+Comment[ar]=ملف DVI مضغوط بـ Gzip
 Comment[bg]=Компресиран файл DVI с gzip
 Comment[bn]=জি-জিপকৃত (gzipped) ডি-ভি-আই ফাইল
 Comment[bs]=Gzip komprimirana DVI datoteka
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-ica.desktop kdelibs/mimetypes/application/x-ica.desktop
--- kdelibs.orig/mimetypes/application/x-ica.desktop	Sat Nov 13 08:20:19 2004
+++ kdelibs/mimetypes/application/x-ica.desktop	Sat Dec 11 12:18:38 2004
@@ -3,6 +3,7 @@
 MimeType=application/x-ica
 Comment=Citrix ICA Client Configuration
 Comment[af]=Citrix ICA Klient Opstelling
+Comment[ar]=إعداد عميل Citrix ICA
 Comment[bg]=Конфигурация на клиент Citrix ICA
 Comment[bn]=সাইট্রিক্স ICA ক্লায়েন্ট কনফিগারেশন
 Comment[bs]=Citrix ICA Client postavke
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-iso.desktop kdelibs/mimetypes/application/x-iso.desktop
--- kdelibs.orig/mimetypes/application/x-iso.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/x-iso.desktop	Sat Dec 11 12:18:38 2004
@@ -5,6 +5,7 @@
 Icon=cdtrack
 Comment=ISO9660 Image File
 Comment[af]=ISO9660 CD-Beeld Leêr
+Comment[ar]=ملف صورة ISO9660
 Comment[be]=Файл вобразу ISO9660
 Comment[bg]=Файл снимка ISO9660
 Comment[bn]=ISO9660 ইমেজ ফাইল
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-javascript.desktop kdelibs/mimetypes/application/x-javascript.desktop
--- kdelibs.orig/mimetypes/application/x-javascript.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/x-javascript.desktop	Sat Dec 11 12:18:38 2004
@@ -5,6 +5,7 @@
 Patterns=*.js;
 Comment=JavaScript File
 Comment[af]=JavaScript Lêer
+Comment[ar]=ملف جافا سكريبت
 Comment[be]=Файл JavaScript
 Comment[bg]=Изходен код на JavaScript
 Comment[bn]=জাভাস্ক্রিপ্ট ফাইল
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-msmetafile.desktop kdelibs/mimetypes/application/x-msmetafile.desktop
--- kdelibs.orig/mimetypes/application/x-msmetafile.desktop	Sat Oct 30 08:38:21 2004
+++ kdelibs/mimetypes/application/x-msmetafile.desktop	Sat Dec 11 12:18:38 2004
@@ -17,6 +17,7 @@
 Comment[et]=Microsoft MetaFile
 Comment[fi]=Microsoft Metatiedosto
 Comment[fr]=Métafichier Microsoft
+Comment[ga]=Meiteachomhad Microsoft
 Comment[he]=מטא-קובץ של Microsoft
 Comment[hr]=Microsoft metadatoteka
 Comment[hu]=Microsoft-metafájl
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-object.desktop kdelibs/mimetypes/application/x-object.desktop
--- kdelibs.orig/mimetypes/application/x-object.desktop	Tue Aug 24 08:29:49 2004
+++ kdelibs/mimetypes/application/x-object.desktop	Sat Dec 11 12:18:38 2004
@@ -25,6 +25,7 @@
 Comment[fa]=کد موضوعی
 Comment[fi]=Objektikoodi
 Comment[fr]=Code objet
+Comment[ga]=Cód Oibiachta
 Comment[gl]=Código Obxecto
 Comment[he]=קוד אובייקט
 Comment[hi]=ऑब्जेक्ट कोड
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-pak.desktop kdelibs/mimetypes/application/x-pak.desktop
--- kdelibs.orig/mimetypes/application/x-pak.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/x-pak.desktop	Sat Dec 11 12:18:38 2004
@@ -6,6 +6,7 @@
 Patterns=*.pak;*.PAK;
 Comment=PAK File Archive
 Comment[af]=PAK Argief
+Comment[ar]=أرشيف ملف PAK
 Comment[be]=Архіў PAK
 Comment[bg]=Архив PAK
 Comment[bn]=PAK ফাইল আর্কাইভ
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-python-bytecode.desktop kdelibs/mimetypes/application/x-python-bytecode.desktop
--- kdelibs.orig/mimetypes/application/x-python-bytecode.desktop	Sat Oct 30 08:38:21 2004
+++ kdelibs/mimetypes/application/x-python-bytecode.desktop	Sat Dec 11 12:18:38 2004
@@ -25,6 +25,7 @@
 Comment[fa]=کدبایت Python
 Comment[fi]=Python-tavukoodi
 Comment[fr]=Bytecode Python
+Comment[ga]=Beartchód Python
 Comment[gl]=Bytecode de Python
 Comment[he]=קוד בתים של Python
 Comment[hi]=पायथन बाइटकोड
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-tex-pk.desktop kdelibs/mimetypes/application/x-tex-pk.desktop
--- kdelibs.orig/mimetypes/application/x-tex-pk.desktop	Tue Nov  9 08:35:29 2004
+++ kdelibs/mimetypes/application/x-tex-pk.desktop	Sat Dec 11 12:18:38 2004
@@ -21,6 +21,7 @@
 Comment[fa]=قلم جمع
 Comment[fi]=Pakattu kirjasin
 Comment[fr]=Police de caractères compactée
+Comment[ga]=Clófhoireann Phacáilte
 Comment[gl]=Fonte Empaquetada
 Comment[he]=גופן ארוז
 Comment[hi]=पैक्ड फ़ॉन्ट
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-vnd.kde.kplato.desktop kdelibs/mimetypes/application/x-vnd.kde.kplato.desktop
--- kdelibs.orig/mimetypes/application/x-vnd.kde.kplato.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/application/x-vnd.kde.kplato.desktop	Sat Dec 11 12:18:38 2004
@@ -4,6 +4,7 @@
 MimeType=application/x-vnd.kde.kplato
 Comment=KPlato Project Management Document
 Comment[af]=KPlato Projek Bestuur Dokument
+Comment[ar]=مستند إدارة مشاريع KPlato
 Comment[bg]=Документ на KPlato Project Management
 Comment[bn]=কে-প্লেটো প্রোজেক্ট ম্যানেজমেন্ট নথী
 Comment[bs]=KPlato Menadžment projektima dokument
diff -urN -x CVS kdelibs.orig/mimetypes/audio/mpeg.desktop kdelibs/mimetypes/audio/mpeg.desktop
--- kdelibs.orig/mimetypes/audio/mpeg.desktop	Mon Nov 22 09:34:05 2004
+++ kdelibs/mimetypes/audio/mpeg.desktop	Sat Dec 11 12:18:38 2004
@@ -8,6 +8,7 @@
 Icon=sound
 Comment=MPEG Audio
 Comment[af]=MPEG Klank
+Comment[ar]=صوت MPEG
 Comment[be]=MPEG аўдыё
 Comment[bg]=Аудио файл MPEG
 Comment[bn]=এমপেগ অডিও
@@ -23,6 +24,7 @@
 Comment[et]=MPEG audiofail
 Comment[fi]=MPEG-ääni
 Comment[fr]=Son MPEG
+Comment[ga]=Fuaim MPEG
 Comment[he]=שמע MPEG
 Comment[hu]=MPEG-hangfájl
 Comment[is]=MPEG hljóðskrá
diff -urN -x CVS kdelibs.orig/mimetypes/audio/x-mp2.desktop kdelibs/mimetypes/audio/x-mp2.desktop
--- kdelibs.orig/mimetypes/audio/x-mp2.desktop	Fri Nov 12 08:27:54 2004
+++ kdelibs/mimetypes/audio/x-mp2.desktop	Sat Dec 11 12:18:38 2004
@@ -7,6 +7,7 @@
 X-KDE-IsAlso=audio/mpeg
 Comment=MPEG Layer 2 Audio
 Comment[af]=MPEG Vlak 2 Klank
+Comment[ar]=صوت MPEG Layer 2
 Comment[be]=MPEG Layer 2 аўдыё
 Comment[bg]=Аудио файл MP2
 Comment[bn]=এমপেগ লেয়ার ২ অডিও
diff -urN -x CVS kdelibs.orig/mimetypes/audio/x-musepack.desktop kdelibs/mimetypes/audio/x-musepack.desktop
--- kdelibs.orig/mimetypes/audio/x-musepack.desktop	Sat Oct 30 08:38:26 2004
+++ kdelibs/mimetypes/audio/x-musepack.desktop	Sat Dec 11 12:18:38 2004
@@ -19,6 +19,7 @@
 Comment[et]=Musepack audiofail
 Comment[fi]=Musepack-ääni
 Comment[fr]=Son Musepack
+Comment[ga]=Fuaim Musepack
 Comment[he]=MusePack שמע
 Comment[hu]=Musepack-hangfájl
 Comment[is]=Musepack hljóðskrá
diff -urN -x CVS kdelibs.orig/mimetypes/image/x-exr.desktop kdelibs/mimetypes/image/x-exr.desktop
--- kdelibs.orig/mimetypes/image/x-exr.desktop	Fri Nov 12 08:27:55 2004
+++ kdelibs/mimetypes/image/x-exr.desktop	Sat Dec 11 12:18:39 2004
@@ -21,6 +21,7 @@
 Comment[et]=ILM EXR pildifail
 Comment[fi]=ILM EXR -kuva
 Comment[fr]=Image ILM EXR
+Comment[ga]=Íomhá ILM EXR
 Comment[he]=תמונת ILM EXR
 Comment[hr]=ILM EXR slika
 Comment[hu]=ILM EXR-kép
diff -urN -x CVS kdelibs.orig/mimetypes/text/x-ldif.desktop kdelibs/mimetypes/text/x-ldif.desktop
--- kdelibs.orig/mimetypes/text/x-ldif.desktop	Sat Oct 30 08:38:30 2004
+++ kdelibs/mimetypes/text/x-ldif.desktop	Sat Dec 11 12:18:39 2004
@@ -22,6 +22,7 @@
 Comment[fa]=دفتر‌نشانی LDAP
 Comment[fi]=LDAP-osoitekirja
 Comment[fr]=Carnet d'adresses LDAP
+Comment[ga]=Leabhar Seoltaí LDAP
 Comment[gl]=Caderno de Enderezos LDAP
 Comment[he]=ספר כתובת מבוסס LDAP
 Comment[hi]=LDAP पता पुस्तिका
diff -urN -x CVS kdelibs.orig/mimetypes/text/x-xslt.desktop kdelibs/mimetypes/text/x-xslt.desktop
--- kdelibs.orig/mimetypes/text/x-xslt.desktop	Mon Nov 22 09:34:05 2004
+++ kdelibs/mimetypes/text/x-xslt.desktop	Sat Dec 11 12:18:39 2004
@@ -21,6 +21,7 @@
 Comment[fa]=پرونده‌ی صفحه‌گسترده‌ی XSLT
 Comment[fi]=XSLT-tyylitiedosto
 Comment[fr]=Feuille de style XSLT
+Comment[ga]=Stílbhileog XSLT
 Comment[he]=קובץ גיליון סגנון XSLT
 Comment[hi]=XSLT स्टाइलशीट फ़ाइल
 Comment[hr]=XSLT datoteka s opisom stila
diff -urN -x CVS kdelibs.orig/pics/crystalsvg/index.theme kdelibs/pics/crystalsvg/index.theme
--- kdelibs.orig/pics/crystalsvg/index.theme	Fri Nov 12 08:27:55 2004
+++ kdelibs/pics/crystalsvg/index.theme	Sat Dec 11 12:18:39 2004
@@ -17,6 +17,7 @@
 Name[uz]=Кристал SVG бета1
 Comment=Icon Theme by Everaldo.com Design Studio
 Comment[af]=Ikoon Tema deur Everaldo.com Design Studio
+Comment[ar]=سمات أيقونات من استديوهات تصميم Everaldo.com
 Comment[be]=Тэма піктаграмаў ад студыі дызайну Everaldo.com
 Comment[bg]=Тема с икони от Евералдо (www.everaldo.com)
 Comment[bn]=এভারাল্ডোর (www.everaldo.com) তৈরি আইকন থিম
