--- kdefx/kcpuinfo.cpp~	2004-05-29 22:04:47.665679096 +0200
+++ kdefx/kcpuinfo.cpp	2004-05-29 22:04:47.666678944 +0200
@@ -160,19 +160,6 @@
         //       so we don't have to do any additional tests for that.
     }
 #endif // HAVE_X86_SSE
-#elif defined __PPC__ && defined HAVE_PPC_ALTIVEC
-    signal( SIGILL, sigill_handler );
-    if ( sigsetjmp( jmpbuf, 1 ) ) {
-        signal( SIGILL, SIG_DFL );
-    } else {
-        canjump = 1;
-        __asm__ __volatile__( "mtspr 256, %0\n\t"
-                              "vand %%v0, %%v0, %%v0"
-                              : /* none */
-                              : "r" (-1) );
-        signal( SIGILL, SIG_DFL );
-        features |= KCPUInfo::AltiVec;
-    }
 #endif // __i386__
 #endif //HAVE_GNU_INLINE_ASM
 
