diff -urN kdelibs-2.2.1.orig/kdecore/kstddirs.cpp kdelibs-2.2.1/kdecore/kstddirs.cpp
--- kdelibs-2.2.1.orig/kdecore/kstandarddirs.cpp	Wed Jan 24 23:29:38 2001
+++ kdelibs-2.2.1/kdecore/kstandarddirs.cpp	Thu Nov  1 19:50:21 2001
@@ -724,9 +724,9 @@
     if (!strcmp(type, "data"))
 	return "share/apps/";
     if (!strcmp(type, "html"))
-	return "share/doc/HTML/";
+	return "share/doc/kde/HTML/";
     if (!strcmp(type, "icon"))
-	return "share/icons/";
+	return "share/pixmaps/";
     if (!strcmp(type, "config"))
 	return "share/config/";
     if (!strcmp(type, "pixmap"))
@@ -964,6 +964,7 @@
     addResourceDir("cache", dir);

     addResourceDir("home", QDir::homeDirPath());
+    addResourceDir("html", "/usr/share/doc/kde/HTML");
 }
 
 void KStandardDirs::checkConfig() const
