diff -ruN kdelibs-1.93.orig/configure.in kdelibs-1.93/configure.in
--- kdelibs-1.93.orig/configure.in	Fri Aug 18 13:14:20 2000
+++ kdelibs-1.93/configure.in	Fri Aug 25 10:32:18 2000
@@ -599,11 +599,6 @@
 AC_SUBST(KDE_CATALOG)
 AC_SUBST(KDE_STYLESHEET)
 
-KDE_FIND_PATH(install-catalog, INSTALL_CATALOG,
-	AC_MSG_WARN([no install-catalog found])
-)
-AC_SUBST(INSTALL_CATALOG)
-
 catdir=$kde_datadir/ksgmltools
 AC_SUBST(catdir)
 
