--- kdelibs-R14.0.0/tdecore/kiconloader.cpp~	2014-10-18 21:00:57.000000000 +0300
+++ kdelibs-R14.0.0/tdecore/kiconloader.cpp	2015-03-22 13:44:00.492426502 +0200
@@ -239,6 +239,11 @@
 	    d->mpGroups[i].size = d->mpThemeRoot->theme->defaultSize(i);
     }
 
+    // Make the app icons themable!!!
+    d->mpDirs->addResourceType("appicon", TDEStandardDirs::kde_default("icon") +
+		TDEIconTheme::current() + appname + "/pics/");
+    d->mpDirs->addResourceType("appicon", TDEStandardDirs::kde_default("icon") +
+		TDEIconTheme::current() + appname + "/toolbar/");
     // Insert application specific themes at the top.
     d->mpDirs->addResourceType("appicon", TDEStandardDirs::kde_default("data") +
 		appname + "/pics/");
