--- kdelibs-3.5.6/kio/kio/kprotocolmanager.cpp.orig	2006-01-19 18:06:11.000000000 +0100
+++ kdelibs-3.5.6/kio/kio/kprotocolmanager.cpp	2007-03-22 01:37:08.000000000 +0100
@@ -72,7 +72,7 @@
 
 // DEFAULT USERAGENT STRING
 #define CFG_DEFAULT_UAGENT(X) \
-QString("Mozilla/5.0 (compatible; Konqueror/%1.%2%3) KHTML/%4.%5.%6 (like Gecko)") \
+QString("Mozilla/5.0 (compatible; Konqueror/%1.%2%3) KHTML/%4.%5.%6 (like Gecko) PLD/Titanium") \
         .arg(KDE_VERSION_MAJOR).arg(KDE_VERSION_MINOR).arg(X).arg(KDE_VERSION_MAJOR).arg(KDE_VERSION_MINOR).arg(KDE_VERSION_RELEASE)
 
 void KProtocolManager::reparseConfiguration()
