Index: cups/cupsdconf/cupsdbrowsingtimeoutpage.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdeprint/cups/cupsdconf/cupsdbrowsingtimeoutpage.cpp,v
retrieving revision 1.2
diff -u -u -p -r1.2 cupsdbrowsingtimeoutpage.cpp
--- kdelibs/kdeprint/cups/cupsdconf/cupsdbrowsingtimeoutpage.cpp	2001/04/20 05:16:56	1.2
+++ kdelibs/kdeprint/cups/cupsdconf/cupsdbrowsingtimeoutpage.cpp	2002/03/09 14:49:09
@@ -24,7 +24,7 @@
 #include <qlineedit.h>
 #include <qlabel.h>
 #include <qwhatsthis.h>
-
+#include <knumvalidator.h>
 #include "cupsdconf.h"
 #include "cupsdoption.h"
 
@@ -39,8 +39,9 @@ CupsdBrowsingTimeoutPage::CupsdBrowsingT
 		opt_[i] = new CupsdOption(this);
 
 	browseinterval_ = new QLineEdit(opt_[0]);
+	browseinterval_->setValidator(new KIntValidator(1,9999,browseinterval_));
 	browsetimeout_ = new QLineEdit(opt_[1]);
-
+	browsetimeout_->setValidator(new KIntValidator(1,9999,browsetimeout_));
 	QLabel	*l1 = new QLabel(i18n("Browse interval:"), this);
 	QLabel	*l2 = new QLabel(i18n("Browse timeout:"), this);
 
Index: cups/cupsdconf/cupsdnetworkclientspage.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdeprint/cups/cupsdconf/cupsdnetworkclientspage.cpp,v
retrieving revision 1.1
diff -u -u -p -r1.1 cupsdnetworkclientspage.cpp
--- kdelibs/kdeprint/cups/cupsdconf/cupsdnetworkclientspage.cpp	2001/03/30 21:48:47	1.1
+++ kdelibs/kdeprint/cups/cupsdconf/cupsdnetworkclientspage.cpp	2002/03/09 14:49:09
@@ -26,6 +26,7 @@
 #include <qcheckbox.h>
 #include <qframe.h>
 #include <qwhatsthis.h>
+#include <knumvalidator.h>
 
 #include "cupsdconf.h"
 #include "cupsdoption.h"
@@ -42,7 +43,10 @@ CupsdNetworkClientsPage::CupsdNetworkCli
 
 	keepalive_ = new QCheckBox(i18n("Accept \"Keep Alive\" requests"), opt_[0]);
 	keepalivetimeout_ = new QLineEdit(opt_[1]);
+	keepalivetimeout_->setValidator(new KIntValidator(1,9999,keepalivetimeout_));
+	
 	maxclients_ = new QLineEdit(opt_[2]);
+	maxclients_->setValidator(new KIntValidator(1,9999,maxclients_));
 
 	QLabel	*l1 = new QLabel(i18n("Keep alive timeout:"), this);
 	QLabel	*l2 = new QLabel(i18n("Max number of clients:"), this);
