--- kdelibs-3.1rc6/kdecore/kstandarddirs.cpp.orig	Sat Nov 23 16:39:26 2002
+++ kdelibs-3.1rc6/kdecore/kstandarddirs.cpp	Tue Jan 14 03:18:06 2003
@@ -50,6 +50,7 @@
 #include "kinstance.h"
 #include <sys/param.h>
 #include <unistd.h>
+#include "plddirs.h"
 
 template class QDict<QStringList>;
 
@@ -1039,6 +1040,9 @@
     addResourceDir("cache", dir);
 
     addResourceDir("home", QDir::homeDirPath());
+    addResourceDir("apps", kde_appsdir);
+    addResourceDir("html", kde_htmldir);
+    addResourceDir("icon", kde_icondir);
 }
 
 void KStandardDirs::checkConfig() const
