--- kdelibs-2.0/dcop/dcopidl/main.cpp.nodebug	Sun Oct 22 17:43:27 2000
+++ kdelibs-2.0/dcop/dcopidl/main.cpp	Sun Oct 22 17:43:59 2000
@@ -29,7 +29,7 @@
 
 int idl_line_no;
 
-#ifdef YYDEBUG
+#if YYDEBUG != 0
 extern int yydebug;
 #endif
 
@@ -52,7 +52,7 @@
     arr.resize( len + 1 );
     arr[ len ] = 0;
 
-#ifdef YYDEBUG
+#if YYDEBUG != 0
     char *debug = getenv("DEBUG");
     if (debug)
 	yydebug = 1;
