Index: kstddirs.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdecore/Attic/kstddirs.cpp,v
retrieving revision 1.124.4.1
diff -u -p -r1.124.4.1 kstddirs.cpp
--- kdelibs/kdecore/kstddirs.cpp	2001/09/21 18:10:47	1.124.4.1
+++ kdelibs/kdecore/kstddirs.cpp	2001/11/27 15:27:55
@@ -620,7 +620,7 @@ QString KStandardDirs::findExe( const QS
 	info.setFile( p );
 
 	if( info.exists() && ( ignore || info.isExecutable() )
-	    && info.isFile() ) {
+           && ( info.isFile() || info.isSymLink() )  ) {
 	    return p;
 	}
     }
