Index: interfaces/kspeech/kspeech.h
===================================================================
--- interfaces/kspeech/kspeech.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ interfaces/kspeech/kspeech.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -371,10 +371,11 @@
  *
  * A Talker Code consists of a series of XML tags and attributes.
  * An example of a full Talker Code with all attributes specified is
- *
+ * \code
  *   <voice lang="en" name="kal" gender="male"/>
  *   <prosody volume="soft" rate="fast"/>
  *   <kttsd synthesizer="Festival" />
+ * \endcode
  *
  * (The @e voice and @e prosody tags are adapted from the W3C Speech Synthesis
  * Markup Language (SSML) and Java Speech Markup Language (JSML).
Index: interfaces/kmediaplayer/kfileaudiopreview/kfileaudiopreview.cpp
===================================================================
--- interfaces/kmediaplayer/kfileaudiopreview/kfileaudiopreview.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ interfaces/kmediaplayer/kfileaudiopreview/kfileaudiopreview.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -21,8 +21,10 @@
 {
 protected:
     virtual QObject *createObject( QObject *parent, const char *name,
-                           const char *, const QStringList & )
+                           const char *className, const QStringList & args)
     {
+        Q_UNUSED(className);
+        Q_UNUSED(args);
         return new KFileAudioPreview( dynamic_cast<QWidget*>( parent ), name );
     }
 };
Index: kmdi/kmdi/mainwindow.h
===================================================================
--- kmdi/kmdi/mainwindow.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kmdi/kmdi/mainwindow.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -107,6 +107,10 @@
      * as toplevel and stay-on-top on the application's main widget.
      * @param pWnd widget for the toolview
      * @param pos docking position
+     * @param pTargetWnd
+     * @param percent
+     * @param tabToolTip
+     * @param tabCaption
      * @return created toolview
      */
     KMDI::ToolViewAccessor *addToolWindow( QWidget* pWnd, KDockWidget::DockPosition pos = KDockWidget::DockNone, QWidget* pTargetWnd = 0L, int percent = 50, const QString& tabToolTip = 0, const QString& tabCaption = 0);
Index: kmdi/kmdi/dockcontainer.h
===================================================================
--- kmdi/kmdi/dockcontainer.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kmdi/kmdi/dockcontainer.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -50,7 +50,7 @@
     /**
      * Add a widget to this container.
      * \param w the KDockWidget object to add
-     * \TODO Remove the extra parameters that we don't use?
+     * \todo Remove the extra parameters that we don't use?
      */
     virtual void insertWidget (KDockWidget *w, QPixmap, const QString &, int &);
 
@@ -63,7 +63,7 @@
     /**
      * Set a tooltip for a widget
      *
-     * \TODO Actually implement it? Right now, it looks just it
+     * \todo Actually implement it? Right now, it looks just it
      * does exactly nothing
      */
     virtual void setToolTip (KDockWidget *, QString &);
Index: kate/part/katecodecompletion.cpp
===================================================================
--- kate/part/katecodecompletion.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kate/part/katecodecompletion.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -56,7 +56,6 @@
   public:
     /**
       @short Create a new CCListBox
-      @param view The KateView, CCListBox is displayed in
     */
     KateCCListBox (QWidget* parent = 0, const char* name = 0, WFlags f = 0):QListBox(parent, name, f)
     {
Index: kate/part/katefactory.h
===================================================================
--- kate/part/katefactory.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kate/part/katefactory.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -75,6 +75,7 @@
      * @param widgetName widget name
      * @param parent QObject parent
      * @param name object name
+     * @param classname class parent
      * @param args additional arguments
      * @return constructed part object
      */
Index: kate/part/katejscript.cpp
===================================================================
--- kate/part/katejscript.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kate/part/katejscript.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -360,7 +360,7 @@
 
 const KJS::ClassInfo KateJSDocument::info = { "KateJSDocument", 0, 0, 0 };
 
-Value KateJSDocumentProtoFunc::call(KJS::ExecState *exec, KJS::Object &thisObj, const KJS::List &args)
+KJS::Value KJS::KateJSDocumentProtoFunc::call(KJS::ExecState *exec, KJS::Object &thisObj, const KJS::List &args)
 {
   KJS_CHECK_THIS( KateJSDocument, thisObj );
 
@@ -523,7 +523,7 @@
 
 const KJS::ClassInfo KateJSView::info = { "KateJSView", 0, &KateJSViewTable, 0 };
 
-Value KateJSViewProtoFunc::call(KJS::ExecState *exec, KJS::Object &thisObj, const KJS::List &args)
+KJS::Value KJS::KateJSViewProtoFunc::call(KJS::ExecState *exec, KJS::Object &thisObj, const KJS::List &args)
 {
   KJS_CHECK_THIS( KateJSView, thisObj );
 
@@ -834,7 +834,7 @@
 
 const KJS::ClassInfo KateJSIndenter::info = { "KateJSIndenter", 0, &KateJSIndenterTable, 0 };
 
-Value KateJSIndenterProtoFunc::call(KJS::ExecState *exec, KJS::Object &thisObj, const KJS::List &args)
+KJS::Value KJS::KateJSIndenterProtoFunc::call(KJS::ExecState *exec, KJS::Object &thisObj, const KJS::List &args)
 {
   KJS_CHECK_THIS( KateJSIndenter, thisObj );
 
Index: kate/part/kateview.h
===================================================================
--- kate/part/kateview.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kate/part/kateview.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -286,8 +286,8 @@
         { return getDoc()->textLine( cursorLine() ); }
     QString currentWord()
         { return m_doc->getWord( m_viewInternal->getCursor() ); }
-    void insertText( const QString& text )
-        { getDoc()->insertText( cursorLine(), cursorColumnReal(), text ); }
+    void insertText( const QString& mark )
+        { getDoc()->insertText( cursorLine(), cursorColumnReal(), mark ); }
     bool canDiscard();
     int tabWidth()                { return m_doc->config()->tabWidth(); }
     void setTabWidth( int w )     { m_doc->config()->setTabWidth(w);  }
@@ -380,6 +380,7 @@
     void find( const QString&, long, bool add=true ); ///< proxy for KateSearch
     void replace();
     void replace( const QString&, const QString &, long ); ///< proxy for KateSearch
+    /** Highly confusing but KateSearch::findAgain() is backwards too. */
     void findAgain( bool back );
     void findAgain()              { findAgain( false );          }
     void findPrev()               { findAgain( true );           }
Index: kate/part/katedocument.h
===================================================================
--- kate/part/katedocument.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kate/part/katedocument.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -264,6 +264,7 @@
      * @p removeLine is true, @p length is ignored (eg not needed).
      * @param line line number
      * @param removeLine if true, force to remove the next line
+     * @param length length of the line
      * @return true on success
      */
     bool editUnWrapLine ( uint line, bool removeLine = true, uint length = 0 );
@@ -557,7 +558,7 @@
 
     bool saveFile ();
 
-    void setReadWrite ( bool rw = true );
+    void setReadWrite ( bool readwrite = true );
 
     void setModified( bool m );
 
Index: kate/part/kateautoindent.h
===================================================================
--- kate/part/kateautoindent.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kate/part/kateautoindent.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -349,7 +349,7 @@
     KateCSmartIndent (KateDocument *doc);
     ~KateCSmartIndent ();
 
-    virtual void processNewline (KateDocCursor &begin, bool needContinue);
+    virtual void processNewline (KateDocCursor &cur, bool needContinue);
     virtual void processChar (QChar c);
 
     virtual void processLine (KateDocCursor &line);
@@ -380,7 +380,7 @@
     KatePythonIndent (KateDocument *doc);
     ~KatePythonIndent ();
 
-    virtual void processNewline (KateDocCursor &begin, bool needContinue);
+    virtual void processNewline (KateDocCursor &cur, bool needContinue);
 
     virtual uint modeNumber () const { return KateDocumentConfig::imPythonStyle; };
 
@@ -402,7 +402,7 @@
     ~KateXmlIndent ();
 
     virtual uint modeNumber () const { return KateDocumentConfig::imXmlStyle; }
-    virtual void processNewline (KateDocCursor &begin, bool needContinue);
+    virtual void processNewline (KateDocCursor &cur, bool needContinue);
     virtual void processChar (QChar c);
     virtual void processLine (KateDocCursor &line);
     virtual bool canProcessLine() const { return true; }
@@ -544,7 +544,7 @@
     KateScriptIndent( KateDocument *doc );
     ~KateScriptIndent();
 
-    virtual void processNewline( KateDocCursor &begin, bool needContinue );
+    virtual void processNewline( KateDocCursor &cur, bool needContinue );
     virtual void processChar( QChar c );
 
     virtual void processLine (KateDocCursor &line);
Index: kate/part/kateviewhelpers.cpp
===================================================================
--- kate/part/kateviewhelpers.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kate/part/kateviewhelpers.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -277,7 +277,7 @@
   public:
     KateCmdLineFlagCompletion() {;}
 
-    QString makeCompletion( const QString & s )
+    QString makeCompletion( const QString & string )
     {
       return QString::null;
     }
Index: kate/part/katefactory.cpp
===================================================================
--- kate/part/katefactory.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kate/part/katefactory.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -52,6 +52,7 @@
      * @param widgetName widget name
      * @param parent QObject parent
      * @param name object name
+     * @param classname class name
      * @param args additional arguments
      * @return constructed part object
      */
Index: kate/part/katesearch.h
===================================================================
--- kate/part/katesearch.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kate/part/katesearch.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -223,7 +223,7 @@
     bool help(class Kate::View *, const QString &, QString &);
     QStringList cmds();
     bool wantsToProcessText( const QString &/*cmdname*/ );
-    void processText( Kate::View *, const QString& );
+    void processText( Kate::View *view, const QString& text );
 
   private:
     /**
Index: kate/part/kateautoindent.cpp
===================================================================
--- kate/part/kateautoindent.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kate/part/kateautoindent.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -714,6 +714,7 @@
   int parenCount = 0;  // Possibly in a multiline for stmt.  Used to skip ';' ...
   bool found = false;
   bool isSpecial = false;
+  bool potentialAnchorSeen = false;
 
   //kdDebug(13030) << "calcIndent begin line:" << begin.line() << " col:" << begin.col() << endl;
 
@@ -733,7 +734,7 @@
       {
         QChar tc = textLine->getChar (pos);
         if ((tc == ';' || tc == ':' || tc == ',') && otherAnchor == -1 && parenCount <= 0)
-          otherAnchor = pos;
+          otherAnchor = pos, potentialAnchorSeen = true;
         else if (tc == ')')
           parenCount++;
         else if (tc == '(')
@@ -742,7 +743,7 @@
           openCount--;
         else if (tc == '{')
         {
-          openCount++;
+          openCount++, potentialAnchorSeen = true;
           if (openCount == 1)
             break;
         }
@@ -826,7 +827,7 @@
   }
 
   // treat beginning of document as anchor position
-  if (cur.line() == 0 && cur.col() == 0)
+  if (cur.line() == 0 && cur.col() == 0 && potentialAnchorSeen)
     found = true;
 
   if (!found)
Index: kab/qconfigDB.h
===================================================================
--- kab/qconfigDB.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kab/qconfigDB.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -199,7 +199,7 @@
   bool save(QTextStream& file, int count); 
   /**
    * Get the value for the key as a string. 
-   * \a key is the key to search for, \value is a reference to the object
+   * \a key is the key to search for, \a value is a reference to the object
    * the value for the key is assigned to.
    */
   bool get(const QCString& key, QCString& value) const;
@@ -226,7 +226,7 @@
   // ---------------
   /**
    * Get the value for the key as a long integer. 
-   * \a key is the key to search for, \value is a reference to the object
+   * \a key is the key to search for, \a value is a reference to the object
    * the value for the key is assigned to.
    */
   bool get(const QCString&, long&) const;
@@ -243,7 +243,7 @@
    * This will probably be not fast, but this methods are not suited to save 
    * large amounts of data. For saving anything else than UNICODE strings,
    * no such conversion is needed.
-   * \a key is the key to search for, \value is a reference to the object
+   * \a key is the key to search for, \a value is a reference to the object
    * the value for the key is assigned to.
    */
   bool get(const QCString&, QString&) const;
@@ -256,7 +256,7 @@
   // ---------------
   /**
    * Get the value for the key as a double. 
-   * \a key is the key to search for, \value is a reference to the object
+   * \a key is the key to search for, \a value is a reference to the object
    * the value for the key is assigned to.
    */
   bool get(const QCString&, double&) const;
@@ -269,7 +269,7 @@
   // ---------------
   /**
    * Get the value for the key as a boolean value. 
-   * \a key is the key to search for, \value is a reference to the object
+   * \a key is the key to search for, \a value is a reference to the object
    * the value for the key is assigned to.
    */
   bool get(const QCString&, bool&) const;
@@ -282,7 +282,7 @@
   // ---------------
   /**
    * Get the value for the key as a list of strings. 
-   * \a key is the key to search for, \value is a reference to the object
+   * \a key is the key to search for, \a value is a reference to the object
    * the value for the key is assigned to.
    */
   bool get(const QCString&, list<QCString>&) const;
@@ -295,7 +295,7 @@
   // --------------
   /**
    * Get the value for the key as a QStrList. 
-   * \a key is the key to search for, \value is a reference to the object
+   * \a key is the key to search for, \a value is a reference to the object
    * the value for the key is assigned to.
    */
   bool get(const QCString&, QStrList&) const;
@@ -310,7 +310,7 @@
    * Get the value for the key as a QStringList. Beware of the difference -
    * a QStringList is a list of QString objects, while QStrList handles
    * char* like objects.
-   * \a key is the key to search for, \value is a reference to the object
+   * \a key is the key to search for, \a value is a reference to the object
    * the value for the key is assigned to.
    */
   bool get(const QCString&, QStringList&) const;
@@ -323,7 +323,7 @@
   // --------------
   /**
    * Get the value for the key as a list of long integers. 
-   * \a key is the key to search for, \value is a reference to the object
+   * \a key is the key to search for, \a value is a reference to the object
    * the value for the key is assigned to.
    */
   bool get(const QCString&, list<long>&) const;
@@ -336,7 +336,7 @@
   // --------------
   /**
    * Get the value for the key as a list of integers. 
-   * \a key is the key to search for, \value is a reference to the object
+   * \a key is the key to search for, \a value is a reference to the object
    * the value for the key is assigned to.
    */
   bool get(const QCString&, list<int>&) const;
@@ -370,7 +370,7 @@
   // --------------
   /**
    * Get the value for the key as a list of doubles. 
-   * \a key is the key to search for, \value is a reference to the object
+   * \a key is the key to search for, \a value is a reference to the object
    * the value for the key is assigned to.
    */
   bool get(const QCString&, list<double>&) const;
@@ -612,7 +612,7 @@
   /**
    * Save the file. 
    * \a header will be the comment in the first line of the file.
-   * If \a force is \tt true, a file opened read-only will be switched
+   * If \a force is \c true, a file opened read-only will be switched
    * to read and write mode and back after saving. 
    * @see ::setFileName
    */
@@ -688,7 +688,7 @@
   static void CleanLockFiles(int);
   /**
    * Lock the current file.
-   * Locking is done by creating a file <filename>.lock. 
+   * Locking is done by creating a file \<filename\> lock. 
    * QConfigDB-objects will reject opening a file for reading and 
    * writing if a lockfile for the  filename exists.
    */
Index: kab/addressbook.h
===================================================================
--- kab/addressbook.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kab/addressbook.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -178,7 +178,7 @@
  *  your mainwindow (or whereever). (The toolbar is not implemented by now). <BR>
  *  Some parts of the AddressBook widget are \e interactive, that means they are
  *  displayed as transparent KURLLabels that react when the user clicks on it.
- *  These interactive parts have to be enabled by calling #setInteractiveMode.
+ *  These interactive parts have to be enabled by calling setInteractiveMode().
  */
 class AddressBook : public QFrame
 {
@@ -266,26 +266,26 @@
    *  specific address of this person is part of an object of the member list
    *  \c addresses referenced in the next paragraph. <BR>
    *  The keys defined directly in the entry sections are: <DL>
-   *  <DD> "title" The title of that person. </DD>
-   *  <DD> "rank" A possible military rank of that person. </DD>
-   *  <DD> "fn" The formatted name. If it is not empty, it replaces the
+   *  <DT>title<DT><DD> The title of that person. </DD>
+   *  <DT>rank<DT><DD>A possible military rank of that person. </DD>
+   *  <DT>fn<DT><DD>The formatted name. If it is not empty, it replaces the
    *       standard combination of the other name fields in the address
    *       display. </DD>
-   *  <DD> "nameprefix" A possible name prefix. </DD>
-   *  <DD> "firstname" The first name. </DD>
-   *  <DD> "middlename" The middle name. </DD>
-   *  <DD> "lastname" The last name. </DD>
-   *  <DD> "birthday" The birthday (a QDate). </DD>
-   *  <DD> "comment" A free form comment. </DD>
-   *  <DD> "talk" The talk addresses (a string list). </DD>
-   *  <DD> "emails" The email addresses (a string list). </DD>
-   *  <DD> "keywords" A list of free-form keywords. </DD>
-   *  <DD> "telephone" A list of telephone numbers in a special format. </DD>
-   *  <DD> "URLs" A list of internet addresses. </DD>
-   *  <DD> "user_1" The first user-declared data field. </DD>
-   *  <DD> "user_2" The second user-declared data field. </DD>
-   *  <DD> "user_3" The third user-declared data field. </DD>
-   *  <DD> "user_4" The fourth user-declared data field. </DD>
+   *  <DT>nameprefix<DT><DD>A possible name prefix. </DD>
+   *  <DT>firstname<DT><DD>The first name. </DD>
+   *  <DT>middlename<DT><DD>The middle name. </DD>
+   *  <DT>lastname<DT><DD>The last name. </DD>
+   *  <DT>birthday<DT><DD>The birthday (a QDate). </DD>
+   *  <DT>comment<DT><DD>A free form comment. </DD>
+   *  <DT>talk<DT><DD>The talk addresses (a string list). </DD>
+   *  <DT>emails<DT><DD>The email addresses (a string list). </DD>
+   *  <DT>keywords<DT><DD>A list of free-form keywords. </DD>
+   *  <DT>telephone<DT><DD>A list of telephone numbers in a special format. </DD>
+   *  <DT>URLs<DT><DD>A list of internet addresses. </DD>
+   *  <DT>user_1<DT><DD>The first user-declared data field. </DD>
+   *  <DT>user_2<DT><DD>The second user-declared data field. </DD>
+   *  <DT>user_3<DT><DD>The third user-declared data field. </DD>
+   *  <DT>user_4<DT><DD>The fourth user-declared data field. </DD>
    *  </DL>
    *  See the next section for a description of the addresses subsections.
    * 
@@ -295,18 +295,18 @@
    *  in the order they are inserted, their keys are the numbers of
    *  inserting converted to a string. <BR>
    *  The keys defined in an address subsection are: <DL>
-   *  <DD> "headline" A headline shown for the address. </DD>
-   *  <DD> "position" The position of the person. </DD>
-   *  <DD> "org" The organization. </DD>
-   *  <DD> "orgunit" The organizational unit. </DD>
-   *  <DD> "orgsubunit" The organizational subunit. </DD>
-   *  <DD> "role" The role of the person. </DD>
-   *  <DD> "deliverylabel" A label for delivering to this address. </DD>
-   *  <DD> "address" The street, house no., flat etc line. </DD>
-   *  <DD> "zip" A zip or postal code. </DD>
-   *  <DD> "town" The town the person lives in in this address. </DD>
-   *  <DD> "country" The country for federal states. </DD>
-   *  <DD> "state" The state for federal states. </DD>
+   *  <DT>headline</DT><DD> A headline shown for the address. </DD>
+   *  <DT>position</DT><DD> The position of the person. </DD>
+   *  <DT>org</DT><DD> The organization. </DD>
+   *  <DT>orgunit</DT><DD> The organizational unit. </DD>
+   *  <DT>orgsubunit</DT><DD> The organizational subunit. </DD>
+   *  <DT>role</DT><DD> The role of the person. </DD>
+   *  <DT>deliverylabel</DT><DD> A label for delivering to this address. </DD>
+   *  <DT>address</DT><DD> The street, house no., flat etc line. </DD>
+   *  <DT>zip</DT><DD> A zip or postal code. </DD>
+   *  <DT>town</DT><DD> The town the person lives in in this address. </DD>
+   *  <DT>country</DT><DD> The country for federal states. </DD>
+   *  <DT>state</DT><DD> The state for federal states. </DD>
    *  </DL>
    *
    *  \par The local configuration section
@@ -315,10 +315,10 @@
    *  with. These settings are called the local configuration. The settings
    *  are stored in the \c config section of the local file. The following
    *  keys are declared in this section: <DL>
-   *  <DD> "user_1" The \e name of the first user-declared field. </DD>
-   *  <DD> "user_2" The \e name of the second user-declared field. </DD>
-   *  <DD> "user_3" The \e name of the third user-declared field. </DD>
-   *  <DD> "user_4" The \e name of the fourth user-declared field. </DD>
+   *  <DT>user_1</DT><DD>The \e name of the first user-declared field. </DD>
+   *  <DT>user_2</DT><DD>The \e name of the second user-declared field. </DD>
+   *  <DT>user_3</DT><DD>The \e name of the third user-declared field. </DD>
+   *  <DT>user_4</DT><DD>The \e name of the fourth user-declared field. </DD>
    *  </DL>
    *  More fields will surely follow.
    **/
Index: kab/kabapi.h
===================================================================
--- kab/kabapi.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kab/kabapi.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -77,6 +77,7 @@
    * without failures. Thus you have to call init before you can
    * use the database.
    * @param parent The QWidget pointer to the parent widget.
+   * @param name The widgets name (used for debugging)
    */
   KabAPI(QWidget* parent=0, const char* name=0);
   /**
@@ -166,7 +167,7 @@
    * for example, nearly similar email addresses. Empty parts of the
    * entry are not considered as criteria.
    * @short This method delivers the closest matches to the given entry.
-   * @param name The name, containing "." for abbreviations.
+   * @param pattern The pattern, containing "." for abbreviations.
    * @param entries Reference to a list of entries where matches are stored.
    * @param max Maximum number of returned entries.
    * @return NoError if an entry is found or NoEntry.
Index: arts/kde/kplayobject.h
===================================================================
--- arts/kde/kplayobject.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ arts/kde/kplayobject.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -37,7 +37,7 @@
 
 	/**
 	  * Sets the internal Arts::PlayObject
-	  * to @playObject
+	  * to @a playObject
 	  */
 	void setObject(Arts::PlayObject playObject);
 	
Index: mimetypes/application/x-mplayer2.desktop
===================================================================
--- mimetypes/application/x-mplayer2.desktop	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ mimetypes/application/x-mplayer2.desktop	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -15,6 +15,7 @@
 Comment[es]=Formator de medios de Microsoft
 Comment[et]=Microsoft Media
 Comment[fa]=قالب میکروسافت ورد
+Comment[fi]=Microsoft Media
 Comment[fr]=Format Microsoft Media
 Comment[hr]=Microsoft Media oblik
 Comment[hu]=Microsoft Media-fájl
Index: kioslave/file/file.h
===================================================================
--- kioslave/file/file.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kioslave/file/file.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -48,10 +48,10 @@
   virtual ~FileProtocol() { }
 
   virtual void get( const KURL& url );
-  virtual void put( const KURL& url, int _mode,
-		    bool _overwrite, bool _resume );
+  virtual void put( const KURL& url, int permissions,
+		    bool overwrite, bool resume );
   virtual void copy( const KURL &src, const KURL &dest,
-                     int mode, bool overwrite );
+                     int permissions, bool overwrite );
   virtual void rename( const KURL &src, const KURL &dest,
                        bool overwrite );
   virtual void symlink( const QString &target, const KURL &dest,
Index: kioslave/http/http.h
===================================================================
--- kioslave/http/http.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kioslave/http/http.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -190,16 +190,16 @@
   virtual void slave_status();
 
   virtual void get( const KURL& url );
-  virtual void put( const KURL& url, int _mode, bool overwrite,
-                    bool _resume );
+  virtual void put( const KURL& url, int permissions, bool overwrite,
+                    bool resume );
 
 //----------------- Re-implemented methods for WebDAV -----------
   virtual void listDir( const KURL& url );
-  virtual void mkdir( const KURL& url, int _permissions );
+  virtual void mkdir( const KURL& url, int permissions );
 
   virtual void rename( const KURL& src, const KURL& dest, bool overwrite );
-  virtual void copy( const KURL& src, const KURL& dest, int _permissions, bool overwrite );
-  virtual void del( const KURL& url, bool _isfile );
+  virtual void copy( const KURL& src, const KURL& dest, int permissions, bool overwrite );
+  virtual void del( const KURL& url, bool isfile );
 
   // ask the host whether it supports WebDAV & cache this info
   bool davHostOk();
Index: kioslave/ftp/ftp.h
===================================================================
--- kioslave/ftp/ftp.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kioslave/ftp/ftp.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -260,7 +260,7 @@
 
   virtual void listDir( const KURL & url );
   virtual void mkdir( const KURL & url, int permissions );
-  virtual void rename( const KURL & src, const KURL & dst, bool overwrite );
+  virtual void rename( const KURL & src, const KURL & dest, bool overwrite );
   virtual void del( const KURL & url, bool isfile );
   virtual void chmod( const KURL & url, int permissions );
 
Index: kioslave/gzip/kgzipfilter.cpp
===================================================================
--- kioslave/gzip/kgzipfilter.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kioslave/gzip/kgzipfilter.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -38,8 +38,12 @@
 public:
     KGzipFilterFactory() : KLibFactory() {}
     ~KGzipFilterFactory(){}
-    QObject *createObject( QObject *, const char *, const char*, const QStringList & )
+    QObject *createObject( QObject *parent, const char *name, const char*className, const QStringList & args )
     {
+        Q_UNUSED(parent);
+        Q_UNUSED(name);
+        Q_UNUSED(className);
+        Q_UNUSED(args);
         return new KGzipFilter;
     }
 };
Index: kio/kio/slavebase.cpp
===================================================================
--- kio/kio/slavebase.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kio/kio/slavebase.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -85,21 +85,22 @@
 
    QStringList groupList() const { return QStringList(); }
 
-   QMap<QString,QString> entryMap(const QString &) const
-      { return QMap<QString,QString>(); }
+   QMap<QString,QString> entryMap(const QString &group) const
+      { Q_UNUSED(group); return QMap<QString,QString>(); }
 
    void reparseConfiguration() { }
 
-   KEntryMap internalEntryMap( const QString &) const { return KEntryMap(); }
+   KEntryMap internalEntryMap( const QString &pGroup) const { Q_UNUSED(pGroup); return KEntryMap(); }
 
    KEntryMap internalEntryMap() const { return KEntryMap(); }
 
-   void putData(const KEntryKey &, const KEntry&, bool) { }
+   void putData(const KEntryKey &_key, const KEntry&_data, bool _checkGroup) 
+   { Q_UNUSED(_key); Q_UNUSED(_data); Q_UNUSED(_checkGroup); }
 
-   KEntry lookupData(const KEntryKey &key) const
+   KEntry lookupData(const KEntryKey &_key) const
    {
      KEntry entry;
-     QString value = slave->metaData(key.c_key);
+     QString value = slave->metaData(_key.c_key);
      if (!value.isNull())
         entry.mValue = value.utf8();
      return entry;
Index: kio/kio/defaultprogress.h
===================================================================
--- kio/kio/defaultprogress.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kio/kio/defaultprogress.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -48,7 +48,7 @@
   /**
    * Creates a new default progress dialog.
    * @param parent the parent of the dialog (or 0 for top-level)
-   * @param the name of the dialog, can be 0
+   * @param name the name of the dialog, can be 0
    * @since 3.1
    */
   DefaultProgress( QWidget* parent, const char* name = 0 );
@@ -62,16 +62,16 @@
                                     unsigned long totalFiles );
 
 public slots:
-  virtual void slotTotalSize( KIO::Job*, KIO::filesize_t size );
-  virtual void slotTotalFiles( KIO::Job*, unsigned long files );
-  virtual void slotTotalDirs( KIO::Job*, unsigned long dirs );
+  virtual void slotTotalSize( KIO::Job *job, KIO::filesize_t size );
+  virtual void slotTotalFiles( KIO::Job *job, unsigned long files );
+  virtual void slotTotalDirs( KIO::Job *job, unsigned long dirs );
 
-  virtual void slotProcessedSize( KIO::Job*, KIO::filesize_t bytes );
-  virtual void slotProcessedFiles( KIO::Job*, unsigned long files );
-  virtual void slotProcessedDirs( KIO::Job*, unsigned long dirs );
+  virtual void slotProcessedSize( KIO::Job *job, KIO::filesize_t bytes );
+  virtual void slotProcessedFiles( KIO::Job *job, unsigned long files );
+  virtual void slotProcessedDirs( KIO::Job *job, unsigned long dirs );
 
-  virtual void slotSpeed( KIO::Job*, unsigned long speed );
-  virtual void slotPercent( KIO::Job*, unsigned long percent );
+  virtual void slotSpeed( KIO::Job *job, unsigned long speed );
+  virtual void slotPercent( KIO::Job *job, unsigned long percent );
   /**
    * Called to set an information message.
    * @param job the KIO::Job
@@ -79,38 +79,38 @@
    */
   virtual void slotInfoMessage( KIO::Job *job, const QString & msg );
 
-  virtual void slotCopying( KIO::Job*, const KURL& src, const KURL& dest );
-  virtual void slotMoving( KIO::Job*, const KURL& src, const KURL& dest );
-  virtual void slotDeleting( KIO::Job*, const KURL& url );
+  virtual void slotCopying( KIO::Job* job, const KURL& src, const KURL& dest );
+  virtual void slotMoving( KIO::Job* job, const KURL& src, const KURL& dest );
+  virtual void slotDeleting( KIO::Job* job, const KURL& url );
   /**
    * Called when the job is transferring.
    * @param job the KIO::Job
    * @param url the url to transfer
    * @since 3.1
    */
-  void slotTransferring( KIO::Job*, const KURL& url );
-  virtual void slotCreatingDir( KIO::Job*, const KURL& dir );
+  void slotTransferring( KIO::Job* job, const KURL& url );
+  virtual void slotCreatingDir( KIO::Job* job, const KURL& dir );
   /**
    * Called when the job is requesting a stat.
    * @param job the KIO::Job
    * @param dir the dir to stat
    * @since 3.1
    */
-  virtual void slotStating( KIO::Job*, const KURL& dir );
+  virtual void slotStating( KIO::Job* job, const KURL& dir );
   /**
    * Called when the job is mounting.
    * @param job the KIO::Job
    * @param dev the device to mount
    * @param point the mount point
    */
-  virtual void slotMounting( KIO::Job*, const QString & dev, const QString & point );
+  virtual void slotMounting( KIO::Job* job, const QString & dev, const QString & point );
   /**
    * Called when the job is unmounting.
    * @param job the KIO::Job
    * @param point the mount point
    */
-  virtual void slotUnmounting( KIO::Job*, const QString & point );
-  virtual void slotCanResume( KIO::Job*, KIO::filesize_t );
+  virtual void slotUnmounting( KIO::Job* job, const QString & point );
+  virtual void slotCanResume( KIO::Job* job, KIO::filesize_t from);
 
   /**
    * Called when the job is cleaned.
Index: kio/kio/slavebase.h
===================================================================
--- kio/kio/slavebase.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kio/kio/slavebase.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -395,7 +395,7 @@
      *
      * @see canResume()
      */
-    virtual void put( const KURL& url, int permissions, bool overwrite, bool /*resume*/ );
+    virtual void put( const KURL& url, int permissions, bool overwrite, bool resume );
 
     /**
      * Finds all details for one file or directory.
Index: kio/kio/kservice.cpp
===================================================================
--- kio/kio/kservice.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kio/kio/kservice.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -447,19 +447,21 @@
 
    QStringList groupList() const { return QStringList(); }
 
-   QMap<QString,QString> entryMap(const QString &) const
-      { return QMap<QString,QString>(); }
+   QMap<QString,QString> entryMap(const QString &group) const
+      { Q_UNUSED(group); return QMap<QString,QString>(); }
 
    void reparseConfiguration() { }
 
-   KEntryMap internalEntryMap( const QString &) const { return KEntryMap(); }
+   KEntryMap internalEntryMap( const QString &pGroup) const 
+   { Q_UNUSED(pGroup); return KEntryMap(); }
 
    KEntryMap internalEntryMap() const { return KEntryMap(); }
 
-   void putData(const KEntryKey &, const KEntry&, bool) { }
+   void putData(const KEntryKey &_key, const KEntry& _data, bool _checkGroup) 
+   { Q_UNUSED(_key); Q_UNUSED(_data); Q_UNUSED(_checkGroup); }
 
-   KEntry lookupData(const KEntryKey &) const
-   { KEntry entry; entry.mValue = value; return entry; }
+   KEntry lookupData(const KEntryKey &_key) const
+   { Q_UNUSED(_key); KEntry entry; entry.mValue = value; return entry; }
 protected:
    QString key;
    QCString value;
Index: kio/kio/kscan.h
===================================================================
--- kio/kio/kscan.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kio/kio/kscan.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -339,7 +339,7 @@
     KOCRDialogFactory( QObject *parent=0, const char *name=0 );
 
     virtual QObject* createObject( QObject* parent = 0, const char* name = 0,
-                                   const char* classname = "QObject",
+                                   const char* className = "QObject",
                                    const QStringList &args = QStringList() );
 
 
Index: kio/kfile/kfiletreeview.h
===================================================================
--- kio/kfile/kfiletreeview.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kio/kfile/kfiletreeview.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -214,9 +214,9 @@
    // The drop event allows to differentiate between move and copy
    void dropped( QWidget*, QDropEvent*, KURL::List&, KURL& );
 
-   void dropped( QDropEvent *, QListViewItem * );
+   void dropped( QDropEvent *e, QListViewItem * after);
    void dropped(KFileTreeView *, QDropEvent *, QListViewItem *);
-   void dropped(QDropEvent *, QListViewItem *, QListViewItem *);
+   void dropped(QDropEvent *e, QListViewItem * parent, QListViewItem * after);
    void dropped(KFileTreeView *, QDropEvent *, QListViewItem *, QListViewItem *);
 
 protected:
Index: kio/kfile/kurlrequester.cpp
===================================================================
--- kio/kfile/kurlrequester.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kio/kfile/kurlrequester.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -306,7 +306,7 @@
     emit urlSelected( d->url() );
 }
 
-void KURLRequester::setMode(unsigned int mode)
+void KURLRequester::setMode(uint mode)
 {
     Q_ASSERT( (mode & KFile::Files) == 0 );
     d->fileDialogMode = mode;
Index: kio/misc/uiserver.h
===================================================================
--- kio/misc/uiserver.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kio/misc/uiserver.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -276,6 +276,7 @@
 
   /**
    * Popup a message box.
+   * @param id The message identifier.
    * @param type type of message box: QuestionYesNo, WarningYesNo, WarningContinueCancel...
    *   This enum is defined in slavebase.h, it currently is:
    *   QuestionYesNo = 1, WarningYesNo = 2, WarningContinueCancel = 3,
Index: kio/kssl/ksslcsessioncache.h
===================================================================
--- kio/kssl/ksslcsessioncache.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kio/kssl/ksslcsessioncache.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -31,14 +31,14 @@
 
     /**
      * Store a SSL session (client side only)
-     * @param url URL the key belongs to. Method, host and port are used
+     * @param kurl URL the key belongs to. Method, host and port are used
      * @param session QString representing session to store
      */
     static void putSessionForURL(const KURL &kurl, const QString &session);
 
     /**
      * Retrieve a SSL session (client side only)
-     * @param url URL the key belongs to
+     * @param kurl URL the key belongs to
      * @return if a key can be found, QString::null otherwise
      */
     static QString getSessionForURL(const KURL &kurl);
Index: kio/kssl/ksmimecrypto.h
===================================================================
--- kio/kssl/ksmimecrypto.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kio/kssl/ksmimecrypto.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -85,7 +85,7 @@
      * Does not check certificates for validity!
      * @param signedText signed message block
      * @param clearText cleartext of signed message
-     * @foundCerts certificates found in this mesasge
+     * @param foundCerts certificates found in this mesasge
      * @return 0 on success
      */
     rc checkOpaqueSignature(const QByteArray &signedText,
@@ -101,7 +101,7 @@
      * @param clearText MIME representation of message to encrypt
      * @param cipherText returned encrypted message
      * @param algorithm encryption algorithm
-     * @recip recipient certificates
+     * @param recip recipient certificates
      * @return 0 on success
      */
     rc encryptMessage(const QCString &clearText,
Index: kabc/plugins/sql/resourcesql.h
===================================================================
--- kabc/plugins/sql/resourcesql.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kabc/plugins/sql/resourcesql.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -43,7 +43,7 @@
   Ticket *requestSaveTicket();
 
   bool load();
-  bool save( Ticket * );
+  bool save( Ticket * ticket );
 
   QString identifier() const;
 
Index: kabc/plugins/ldapkio/resourceldapkio.h
===================================================================
--- kabc/plugins/ldapkio/resourceldapkio.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kabc/plugins/ldapkio/resourceldapkio.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -57,8 +57,8 @@
 
     virtual bool load();
     virtual bool asyncLoad();
-    virtual bool save( Ticket * );
-    virtual bool asyncSave( Ticket * );
+    virtual bool save( Ticket * ticket );
+    virtual bool asyncSave( Ticket * ticket );
 
     virtual void removeAddressee( const Addressee& addr );
 
Index: kabc/sortmode.h
===================================================================
--- kabc/sortmode.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kabc/sortmode.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -66,6 +66,7 @@
       Creates a NameSortMethod with the specified name type.
 
       @param type The name type.
+      @param ascending true for ascending sort, false for descending.
      */
     NameSortMode( NameType type, bool ascending = true );
 
@@ -91,6 +92,7 @@
       Creates a FieldSortMethod with the specified field.
 
       @param field The field.
+      @param ascending true for ascending sort, false for descending.
      */
     FieldSortMode( KABC::Field *field, bool ascending = true );
 
Index: kabc/sound.h
===================================================================
--- kabc/sound.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kabc/sound.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -51,7 +51,7 @@
  *      }
  *      KAudioPlayer::play( tmpFile ); 
  *    }
- *  \code
+ *  \endcode
  *       
  *  Unfortunetly KAudioPlayer::play is ASync, so to delete the temporary file, the best you can really do is set a timer.
  *
Index: kabc/field.h
===================================================================
--- kabc/field.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kabc/field.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -97,7 +97,7 @@
   virtual bool isCustom();
 
   /**
-   * Returns, if the field is equal with @param field.
+   * Returns, if the field is equal with @a field.
    */
   virtual bool equals( Field *field );
 
Index: kcert/kcertpart.h
===================================================================
--- kcert/kcertpart.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kcert/kcertpart.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -73,7 +73,7 @@
 	    const QStringList &args = QStringList() );
   virtual ~KCertPart();
 
-  virtual void setReadWrite(bool rw);
+  virtual void setReadWrite(bool readwrite);
 
   static KAboutData *createAboutData();
 
Index: kdesu/process.h
===================================================================
--- kdesu/process.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdesu/process.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -119,7 +119,7 @@
      */
     int pid() {return m_Pid;};
 
-public /* static */:
+public: /* static */
     /*
     ** This is a collection of static functions that can be
     ** used for process control inside kdesu. I'd suggest 
@@ -127,7 +127,7 @@
     ** nicer Qt based ways to do what you want.
     */
 
-    /*
+    /**
     ** Wait @p ms miliseconds (ie. 1/10th of a second is 100ms),
     ** using @p fd as a filedescriptor to wait on. Returns
     ** select(2)'s result, which is -1 on error, 0 on timeout,
@@ -139,14 +139,14 @@
     static int waitMS(int fd,int ms);
 
 
-    /*
+    /**
     ** Basic check for the existence of @p pid.
     ** Returns true iff @p pid is an extant process,
     ** (one you could kill - see man kill(2) for signal 0).
     */
     static bool checkPid(pid_t pid);
 
-    /*
+    /**
     ** Check process exit status for process @p pid.
     ** On error (no child, no exit), return -1.
     ** If child @p pid has exited, return its exit status,
Index: kspell2/plugins/aspell/kspell_aspellclient.h
===================================================================
--- kspell2/plugins/aspell/kspell_aspellclient.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/plugins/aspell/kspell_aspellclient.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * kspell_aspellclient.h
  *
  * Copyright (C)  2003  Zack Rusin <zack@kde.org>
Index: kspell2/plugins/aspell/kspell_aspellclient.cpp
===================================================================
--- kspell2/plugins/aspell/kspell_aspellclient.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/plugins/aspell/kspell_aspellclient.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * kspell_aspellclient.cpp
  *
  * Copyright (C)  2003  Zack Rusin <zack@kde.org>
Index: kspell2/plugins/ispell/kspell_ispellclient.cpp
===================================================================
--- kspell2/plugins/ispell/kspell_ispellclient.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/plugins/ispell/kspell_ispellclient.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * kspell_aspellclient.cpp
  *
  * Copyright (C)  2003  Zack Rusin <zack@kde.org>
Index: kspell2/plugins/ispell/kspell_ispellclient.h
===================================================================
--- kspell2/plugins/ispell/kspell_ispellclient.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/plugins/ispell/kspell_ispellclient.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * kspell_ispellclient.h
  *
  * Copyright (C)  2003  Zack Rusin <zack@kde.org>
Index: kspell2/settings.cpp
===================================================================
--- kspell2/settings.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/settings.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,5 +1,5 @@
 // -*- Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil; -*-
-/**
+/*
  * settings.cpp
  *
  * Copyright (C)  2003  Zack Rusin <zack@kde.org>
Index: kspell2/threadevents.h
===================================================================
--- kspell2/threadevents.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/threadevents.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * threadevents.h
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: kspell2/dictionary.h
===================================================================
--- kspell2/dictionary.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/dictionary.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,5 +1,5 @@
 // -*- Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil; -*-
-/**
+/*
  * dictionary.h
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: kspell2/backgroundengine.h
===================================================================
--- kspell2/backgroundengine.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/backgroundengine.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * backgroundengine.h
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: kspell2/client.cpp
===================================================================
--- kspell2/client.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/client.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,5 +1,5 @@
 // -*- Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil; -*-
-/**
+/*
  * client.cpp
  *
  * Copyright (C)  2003  Zack Rusin <zack@kde.org>
Index: kspell2/defaultdictionary.h
===================================================================
--- kspell2/defaultdictionary.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/defaultdictionary.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * defaultdictionary.h
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: kspell2/client.h
===================================================================
--- kspell2/client.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/client.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,5 +1,5 @@
 // -*- Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil; -*-
-/**
+/*
  * client.cpp
  *
  * Copyright (C)  2003  Zack Rusin <zack@kde.org>
@@ -28,16 +28,16 @@
 
 #include <kdelibs_export.h>
 
-/**
- * The fact that this class inherits from QObject makes me
- * hugely unhappy. The reason for as of writting is that
- * I don't really feel like writting my own KLibFactory
- * that would load anything else then QObject derivatives.
- */
 namespace KSpell2
 {
     class Dictionary;
 
+    /**
+     * The fact that this class inherits from QObject makes me
+     * hugely unhappy. The reason for as of writting is that
+     * I don't really feel like writting my own KLibFactory
+     * that would load anything else then QObject derivatives.
+     */
     class KDE_EXPORT Client : public QObject
     {
         Q_OBJECT
Index: kspell2/filter.cpp
===================================================================
--- kspell2/filter.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/filter.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,5 +1,5 @@
 // -*- Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil; -*-
-/**
+/*
  * filter.cpp
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: kspell2/filter.h
===================================================================
--- kspell2/filter.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/filter.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,5 +1,5 @@
 // -*- Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil; -*-
-/**
+/*
  * filter.h
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: kspell2/ui/configwidget.h
===================================================================
--- kspell2/ui/configwidget.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/ui/configwidget.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * configwidget.h
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: kspell2/ui/highlighter.cpp
===================================================================
--- kspell2/ui/highlighter.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/ui/highlighter.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * highlighter.cpp
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: kspell2/ui/dialog.cpp
===================================================================
--- kspell2/ui/dialog.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/ui/dialog.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * dialog.cpp
  *
  * Copyright (C)  2003  Zack Rusin <zack@kde.org>
Index: kspell2/ui/highlighter.h
===================================================================
--- kspell2/ui/highlighter.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/ui/highlighter.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * highlighter.h
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: kspell2/ui/dialog.h
===================================================================
--- kspell2/ui/dialog.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/ui/dialog.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,5 +1,5 @@
 // -*- Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil; -*-
-/**
+/*
  * dialog.h
  *
  * Copyright (C)  2003  Zack Rusin <zack@kde.org>
Index: kspell2/ui/configdialog.cpp
===================================================================
--- kspell2/ui/configdialog.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/ui/configdialog.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * configdialog.cpp
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: kspell2/ui/configwidget.cpp
===================================================================
--- kspell2/ui/configwidget.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/ui/configwidget.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * configwidget.cpp
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: kspell2/ui/configdialog.h
===================================================================
--- kspell2/ui/configdialog.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kspell2/ui/configdialog.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * configdialog.h
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: kdecore/kglobalaccel_mac.h
===================================================================
--- kdecore/kglobalaccel_mac.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdecore/kglobalaccel_mac.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -14,16 +14,16 @@
     {}
 
     // reimplemented pure virtuals
-    void setEnabled( bool )
-    {}
-    bool emitSignal( Signal )
-    { return false; }
-    bool connectKey( KAccelAction&, const KKeyServer::Key& )
-    { return false; }
-    bool connectKey( const KKeyServer::Key& )
-    { return false; }
-    bool disconnectKey( KAccelAction&, const KKeyServer::Key& )
-    { return false; }
+    void setEnabled( bool bEnabled )
+    { Q_UNUSED(bEnabled); }
+    bool emitSignal( Signal signal )
+    { Q_UNUSED(signal); return false; }
+    bool connectKey( KAccelAction& action, const KKeyServer::Key& key)
+    { Q_UNUSED(action); Q_UNUSED(key); return false; }
+    bool connectKey( const KKeyServer::Key& key)
+    { Q_UNUSED(key); return false; }
+    bool disconnectKey( KAccelAction&, const KKeyServer::Key& key)
+    { Q_UNUSED(key); return false; }
     bool disconnectKey( const KKeyServer::Key& )
     { return false; }
 };
Index: kdecore/kstandarddirs.cpp
===================================================================
--- kdecore/kstandarddirs.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdecore/kstandarddirs.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -441,7 +441,7 @@
     }
 
 #ifndef NDEBUG
-    if(false && type != "locale")
+    if(false && strcmp(type, "locale"))
       kdDebug() << "KStdDirs::findResDir(): can't find \"" << filename << "\" in type \"" << type << "\"." << endl;
 #endif
 
@@ -690,7 +690,7 @@
 
     /* If the path contains symlinks, get the real name */
     if (realpath( QFile::encodeName(dirname).data(), realpath_buffer) != 0) {
-        // succes, use result from realpath
+        // success, use result from realpath
         int len = strlen(realpath_buffer);
         realpath_buffer[len] = '/';
         realpath_buffer[len+1] = 0;
@@ -708,7 +708,7 @@
 
     /* If the path contains symlinks, get the real name */
     if (realpath( QFile::encodeName(filename).data(), realpath_buffer) != 0) {
-        // succes, use result from realpath
+        // success, use result from realpath
         return QFile::decodeName(realpath_buffer);
     }
 
@@ -1287,6 +1287,7 @@
            kdedirList.append(kdedir);
         }
     }
+
 #ifndef Q_OS_WIN //no default KDEDIR on win32 defined
     kdedirList.append(KDEDIR);
 #endif
Index: kdecore/kglobalaccel_emb.h
===================================================================
--- kdecore/kglobalaccel_emb.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdecore/kglobalaccel_emb.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -9,7 +9,7 @@
 public:
 	KGlobalAccelPrivate();
 
-	virtual void setEnabled( bool );
+	virtual void setEnabled( bool bEnabled );
 
 	virtual bool connectKey( KAccelAction&, KKeySequence );
 	virtual bool disconnectKey( KAccelAction&, KKeySequence );
Index: kdecore/kaccelprivate.h
===================================================================
--- kdecore/kaccelprivate.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdecore/kaccelprivate.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -21,17 +21,17 @@
 
 	KAccelPrivate( KAccel* pParent, QWidget* pWatch );
 
-	virtual void setEnabled( bool );
+	virtual void setEnabled( bool bEnabled );
 
 	bool setEnabled( const QString& sAction, bool bEnable );
 
 	virtual bool removeAction( const QString& sAction );
 
-	virtual bool emitSignal( KAccelBase::Signal );
-	virtual bool connectKey( KAccelAction&, const KKeyServer::Key& );
-	virtual bool connectKey( const KKeyServer::Key& );
-	virtual bool disconnectKey( KAccelAction&, const KKeyServer::Key& );
-	virtual bool disconnectKey( const KKeyServer::Key& );
+	virtual bool emitSignal( KAccelBase::Signal signal );
+	virtual bool connectKey( KAccelAction& action, const KKeyServer::Key& key );
+	virtual bool connectKey( const KKeyServer::Key& key );
+	virtual bool disconnectKey( KAccelAction& action, const KKeyServer::Key& key );
+	virtual bool disconnectKey( const KKeyServer::Key& key );
 
  signals:
 	void menuItemActivated();
Index: kdecore/kglobalaccel_x11.h
===================================================================
--- kdecore/kglobalaccel_x11.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdecore/kglobalaccel_x11.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -38,13 +38,13 @@
 	KGlobalAccelPrivate();
 	virtual ~KGlobalAccelPrivate();
 
-	virtual void setEnabled( bool );
+	virtual void setEnabled( bool bEnabled );
 
-	virtual bool emitSignal( Signal );
-	virtual bool connectKey( KAccelAction&, const KKeyServer::Key& );
-	virtual bool connectKey( const KKeyServer::Key& );
-	virtual bool disconnectKey( KAccelAction&, const KKeyServer::Key& );
-	virtual bool disconnectKey( const KKeyServer::Key& );
+	virtual bool emitSignal( Signal signal );
+	virtual bool connectKey( KAccelAction& action, const KKeyServer::Key& key );
+	virtual bool connectKey( const KKeyServer::Key& key );
+	virtual bool disconnectKey( KAccelAction& action, const KKeyServer::Key& key );
+	virtual bool disconnectKey( const KKeyServer::Key& key );
 
  protected:
 	/**
Index: kdecore/kglobalaccel_win.h
===================================================================
--- kdecore/kglobalaccel_win.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdecore/kglobalaccel_win.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -38,13 +38,13 @@
 	KGlobalAccelPrivate();
 	virtual ~KGlobalAccelPrivate();
 
-	virtual void setEnabled( bool );
+	virtual void setEnabled( bool bEnabled );
 
-	virtual bool emitSignal( Signal );
-	virtual bool connectKey( KAccelAction&, const KKeyServer::Key& );
-	virtual bool connectKey( const KKeyServer::Key& );
-	virtual bool disconnectKey( KAccelAction&, const KKeyServer::Key& );
-	virtual bool disconnectKey( const KKeyServer::Key& );
+	virtual bool emitSignal( Signal signal );
+	virtual bool connectKey( KAccelAction& action, const KKeyServer::Key& key );
+	virtual bool connectKey( const KKeyServer::Key& key );
+	virtual bool disconnectKey( KAccelAction& action, const KKeyServer::Key& key );
+	virtual bool disconnectKey( const KKeyServer::Key& key );
 
  protected:
 
Index: kdecore/kconfigbase.h
===================================================================
--- kdecore/kconfigbase.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdecore/kconfigbase.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -2133,7 +2133,7 @@
   bool groupIsImmutable() const;
 
    // The following functions are reimplemented:
-   virtual void setDirty(bool b);
+   virtual void setDirty(bool _bDirty);
    virtual void putData(const KEntryKey &_key, const KEntry &_data, bool _checkGroup = true);
    virtual KEntry lookupData(const KEntryKey &_key) const;
    virtual void sync();
Index: kdecore/kwin.h
===================================================================
--- kdecore/kwin.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdecore/kwin.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -394,7 +394,7 @@
      * Convenience function to set the current viewport to @p viewport.
      * See NETRootInfo.
      * @param desktop the number of the new desktop
-     * @param desktop the number of the new viewport
+     * @param viewport the position of the new viewport
      * @since 3.5.5
      */
     static void setCurrentDesktopViewport( int desktop, QPoint viewport );
Index: kdecore/kstartupinfo.cpp
===================================================================
--- kdecore/kstartupinfo.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdecore/kstartupinfo.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -146,7 +146,7 @@
         d->wm_module = NULL;
     connect( &d->msgs, SIGNAL( gotMessage( const QString& )), SLOT( got_message( const QString& )));
 #endif
-    d->cleanup = new QTimer( this );
+    d->cleanup = new QTimer( this, "cleanup" );
     connect( d->cleanup, SIGNAL( timeout()), SLOT( startups_cleanup()));
     }
 
Index: kdecore/kstartupinfo.h
===================================================================
--- kdecore/kstartupinfo.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdecore/kstartupinfo.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -175,7 +175,7 @@
         static bool sendStartup( const KStartupInfoId& id, const KStartupInfoData& data );
 
 	/**
-	 * Like sendStartup , uses dpy instead of qt_x11display() for sending the info.
+	 * Like sendStartup , uses dpy instead of qt_xdisplay() for sending the info.
 	 * @param dpy the display of the application. Note that the name field
          * in data is required.
 	 * @param id the id of the application
@@ -198,7 +198,7 @@
         static bool sendChange( const KStartupInfoId& id, const KStartupInfoData& data );
 
 	/**
-	 * Like sendChange , uses dpy instead of qt_x11display() for sending the info.
+	 * Like sendChange , uses dpy instead of qt_xdisplay() for sending the info.
 	 * @param dpy the display of the application.
 	 * @param id the id of the application
 	 * @param data the application's data
@@ -215,7 +215,7 @@
         static bool sendFinish( const KStartupInfoId& id );
 
 	/**
-	 * Like sendFinish , uses dpy instead of qt_x11display() for sending the info.
+	 * Like sendFinish , uses dpy instead of qt_xdisplay() for sending the info.
 	 * @param dpy the display of the application.
 	 * @param id the id of the application
 	 * @return true if successful, false otherwise
@@ -232,7 +232,7 @@
         static bool sendFinish( const KStartupInfoId& id, const KStartupInfoData& data );
 
 	/**
-	 * Like sendFinish , uses dpy instead of qt_x11display() for sending the info.
+	 * Like sendFinish , uses dpy instead of qt_xdisplay() for sending the info.
 	 * @param dpy the display of the application.
 	 * @param id the id of the application
 	 * @param data the application's data
Index: khtml/html/html_formimpl.h
===================================================================
--- khtml/html/html_formimpl.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ khtml/html/html_formimpl.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -289,7 +289,7 @@
 
     virtual bool maintainsState() { return true; }
     virtual QString state();
-    virtual void restoreState(const QString &);
+    virtual void restoreState(const QString &state);
 
     void select();
     void click();
@@ -410,7 +410,7 @@
 
     virtual bool maintainsState() { return true; }
     virtual QString state();
-    virtual void restoreState(const QString &);
+    virtual void restoreState(const QString &state);
 
     virtual NodeImpl *insertBefore ( NodeImpl *newChild, NodeImpl *refChild, int &exceptioncode );
     virtual void      replaceChild ( NodeImpl *newChild, NodeImpl *oldChild, int &exceptioncode );
@@ -547,7 +547,7 @@
 
     virtual bool maintainsState() { return true; }
     virtual QString state();
-    virtual void restoreState(const QString &);
+    virtual void restoreState(const QString &state);
 
     void select (  );
 
Index: khtml/ecma/kjs_events.h
===================================================================
--- khtml/ecma/kjs_events.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ khtml/ecma/kjs_events.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -35,7 +35,9 @@
   public:
     /**
      * @param _listener the function object, that will be called when the event is emitted
+     * @param _compareListenerImp Compare Listener implementation.
      * @param _win Window object, for memory management and caching.
+     * @param _html \c true if it is HTML. 
      * Never create a JSEventListener directly, use Window::getJSEventListener.
      */
     JSEventListener(Object _listener, ObjectImp *_compareListenerImp, const Object &_win, bool _html = false);
Index: khtml/ecma/kjs_events.cpp
===================================================================
--- khtml/ecma/kjs_events.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ khtml/ecma/kjs_events.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -602,7 +602,7 @@
     if ( rend ) {
       int xPos, yPos;
       if ( rend->absolutePosition( xPos, yPos ) ) {
-        kdDebug() << "DOMMouseEvent::getValueProperty rend=" << rend << "  xPos=" << xPos << "  yPos=" << yPos << endl;
+        //kdDebug() << "DOMMouseEvent::getValueProperty rend=" << rend << "  xPos=" << xPos << "  yPos=" << yPos << endl;
         x -= xPos;
         y -= yPos;
       }
Index: khtml/rendering/render_style.cpp
===================================================================
--- khtml/rendering/render_style.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ khtml/rendering/render_style.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -658,7 +658,7 @@
     if (noninherited_flags.f._styleType==NOPSEUDO)
         for (ps = pseudoStyle; ps; ps = ps->pseudoStyle)
             if (ps->noninherited_flags.f._styleType==pid)
-		break;
+                break;
     return ps;
 }
 
Index: khtml/khtml_ext.h
===================================================================
--- khtml/khtml_ext.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ khtml/khtml_ext.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -160,7 +160,7 @@
             const char *slot, QObject *parent, const char *name );
     virtual ~KHTMLZoomFactorAction();
 
-    virtual int plug( QWidget *w, int index );
+    virtual int plug( QWidget *widget, int index );
 
 private slots:
     void slotActivated( int );
Index: khtml/css/css_renderstyledeclarationimpl.h
===================================================================
--- khtml/css/css_renderstyledeclarationimpl.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ khtml/css/css_renderstyledeclarationimpl.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * css_renderstyleimpl.h
  *
  * Copyright (C)  2004  Zack Rusin <zack@kde.org>
Index: khtml/css/css_renderstyledeclarationimpl.cpp
===================================================================
--- khtml/css/css_renderstyledeclarationimpl.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ khtml/css/css_renderstyledeclarationimpl.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -330,12 +330,12 @@
 RenderStyleDeclarationImpl::RenderStyleDeclarationImpl( DOM::NodeImpl *node )
     : CSSStyleDeclarationImpl(0), m_node(node)
 {
-    kdDebug() << "Render Style Declaration created" << endl;
+    //kdDebug() << "Render Style Declaration created" << endl;
 }
 
 RenderStyleDeclarationImpl::~RenderStyleDeclarationImpl()
 {
-    kdDebug() << "Render Style Declaration destroyed" << endl;
+    //kdDebug() << "Render Style Declaration destroyed" << endl;
 }
 
 DOM::DOMString RenderStyleDeclarationImpl::cssText() const
Index: khtml/xml/dom_nodeimpl.cpp
===================================================================
--- khtml/xml/dom_nodeimpl.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ khtml/xml/dom_nodeimpl.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
-/**
+/*
  * This file is part of the DOM implementation for KDE.
  *
  * Copyright (C) 1999 Lars Knoll (knoll@kde.org)
Index: khtml/xml/dom_nodeimpl.h
===================================================================
--- khtml/xml/dom_nodeimpl.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ khtml/xml/dom_nodeimpl.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -447,7 +447,18 @@
      * @param found          When this is set to true, don't print anymore but closing tags.
      * @return An html formatted string for this node and its children between the selectionStart and selectionEnd.
      */
-    virtual DOMString selectionToString(NodeImpl * /*selectionStart*/, NodeImpl * /*selectionEnd*/, int /*startOffset*/, int /*endOffset*/, bool &/*found*/) const { return toString(); }
+    virtual DOMString selectionToString(NodeImpl * selectionStart, 
+                                        NodeImpl * selectionEnd, 
+                                        int startOffset, 
+                                        int endOffset, 
+                                        bool &found) const 
+    { Q_UNUSED(selectionStart);
+      Q_UNUSED(selectionEnd);
+      Q_UNUSED(startOffset);
+      Q_UNUSED(endOffset);
+      Q_UNUSED(found);
+      return toString(); 
+    }
 
 private: // members
     DocumentPtr *document;
Index: khtml/xml/dom_stringimpl.cpp
===================================================================
--- khtml/xml/dom_stringimpl.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ khtml/xml/dom_stringimpl.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -87,7 +87,7 @@
     }
 }
 
-void DOMStringImpl::insert(DOMStringImpl *str, uint pos)
+void DOMStringImpl::insert(DOMStringImpl *str, unsigned int pos)
 {
     if(pos > l)
     {
@@ -119,7 +119,7 @@
     l = len;
 }
 
-void DOMStringImpl::remove(uint pos, int len)
+void DOMStringImpl::remove(unsigned int pos, int len)
 {
   if(pos >= l ) return;
   if(pos+len > l)
@@ -134,7 +134,7 @@
   l = newLen;
 }
 
-DOMStringImpl *DOMStringImpl::split(uint pos)
+DOMStringImpl *DOMStringImpl::split(unsigned int pos)
 {
   if( pos >=l ) return new DOMStringImpl();
 
@@ -144,7 +144,7 @@
   return str;
 }
 
-DOMStringImpl *DOMStringImpl::substring(uint pos, uint len)
+DOMStringImpl *DOMStringImpl::substring(unsigned int pos, unsigned int len)
 {
   if( pos >=l ) return new DOMStringImpl();
   if(pos+len > l)
Index: khtml/xml/dom_docimpl.h
===================================================================
--- khtml/xml/dom_docimpl.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ khtml/xml/dom_docimpl.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -234,12 +234,12 @@
 
     /**
      * This method returns true if all top-level stylesheets have loaded (including
-     * any @imports that they may be loading).
+     * any \@imports that they may be loading).
      */
     bool haveStylesheetsLoaded() { return m_pendingStylesheets <= 0 || m_ignorePendingStylesheets; }
 
     /**
-     * Increments the number of pending sheets.  The <link> elements
+     * Increments the number of pending sheets.  The \<link\> elements
      * invoke this to add themselves to the loading list.
      */
     void addPendingSheet() { m_pendingStylesheets++; }
@@ -254,8 +254,8 @@
      * Called when one or more stylesheets in the document may have been added, removed or changed.
      *
      * Creates a new style selector and assign it to this document. This is done by iterating through all nodes in
-     * document (or those before <BODY> in a HTML document), searching for stylesheets. Stylesheets can be contained in
-     * <LINK>, <STYLE> or <BODY> elements, as well as processing instructions (XML documents only). A list is
+     * document (or those before \<BODY\> in a HTML document), searching for stylesheets. Stylesheets can be contained in
+     * \<LINK\>, \<STYLE\> or \<BODY\> elements, as well as processing instructions (XML documents only). A list is
      * constructed from these which is used to create the a new style selector which collates all of the stylesheets
      * found and is used to calculate the derived styles for all rendering objects.
      *
Index: khtml/Makefile.am
===================================================================
--- khtml/Makefile.am	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ khtml/Makefile.am	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -125,7 +125,8 @@
 testregression_LDADD = libkhtml.la
 
 DOXYGEN_REFERENCES = kdecore kdeui kio kdefx kparts
-DOXYGEN_EXCLUDE =  test*.* html rendering xml misc pics test 
+DOXYGEN_EXCLUDE =  test*.* css ecma html java kmultipart misc pics rendering test xml
+
 include ../admin/Doxyfile.am
 
 .PHONY: parser
Index: kresources/resource.h
===================================================================
--- kresources/resource.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kresources/resource.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -389,9 +389,13 @@
     virtual ConfigWidget *configWidget( QWidget *parent ) = 0;
 
   protected:
-    virtual QObject* createObject( QObject *, const char *, const char *,
-                                   const QStringList & )
+    virtual QObject* createObject( QObject *parent, const char *name, const char *className,
+                                   const QStringList & args)
     {
+      Q_UNUSED(parent);
+      Q_UNUSED(name);
+      Q_UNUSED(className);
+      Q_UNUSED(args);
       return 0;
     }
 };
Index: kjs/operations.h
===================================================================
--- kjs/operations.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kjs/operations.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -55,6 +55,9 @@
   /**
    * Additive operator. Either performs an addition or substraction of v1
    * and v2.
+   * @param exec execution state.
+   * @param v1 First operand.
+   * @param v2 Second operand.
    * @param oper '+' or '-' for an addition or substraction, respectively.
    * @return The result of the operation.
    */
@@ -62,6 +65,9 @@
   /**
    * Multiplicative operator. Either multiplies/divides v1 and v2 or
    * calculates the remainder from an division.
+   * @param exec execution state.
+   * @param v1 First operand.
+   * @param v2 Second operand.
    * @param oper '*', '/' or '%' for a multiplication, division or
    * modulo operation.
    * @return The result of the operation.
Index: kjs/lookup.h
===================================================================
--- kjs/lookup.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kjs/lookup.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -160,10 +160,10 @@
    * unknown property).
    *
    * Template arguments:
-   * @param FuncImp the class which implements this object's functions
-   * @param ThisImp the class of "this". It must implement the getValueProperty(exec,token) method,
-   * for non-function properties.
-   * @param ParentImp the class of the parent, to propagate the lookup.
+   *   - @c FuncImp the class which implements this object's functions
+   *   - @c ThisImp the class of "this". It must implement the 
+   *        getValueProperty(exec,token) method, for non-function properties.
+   *   - @c ParentImp the class of the parent, to propagate the lookup.
    *
    * Method arguments:
    * @param exec execution state, as usual
Index: kdoctools/customization/pt-BR/entities/help-menu.docbook
===================================================================
--- kdoctools/customization/pt-BR/entities/help-menu.docbook	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdoctools/customization/pt-BR/entities/help-menu.docbook	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -28,10 +28,10 @@
 <varlistentry>
 <term><menuchoice>
 <guimenu>Ajuda</guimenu>
-<guimenuitem>Reportar Erro...</guimenuitem>
+<guimenuitem>Relatar Falha...</guimenuitem>
 </menuchoice></term>
-<listitem><para><action>Abre a janela para Reportar Erros</action> onde pode 
-relatar um erro ou <quote>sugerir</quote> uma funcionalidade.</para></listitem>
+<listitem><para><action>Abre a janela para Reportar Falhas</action> onde você pode 
+relatar uma falha ou <quote>sugerir</quote> uma funcionalidade.</para></listitem>
 </varlistentry>
 
 <varlistentry>
@@ -39,7 +39,7 @@
 <guimenu>Ajuda</guimenu>
 <guimenuitem>Sobre o &kappname;</guimenuitem>
 </menuchoice></term>
-<listitem><para><action>Isto irá mostrar a versão e as informações do autor.</action></para></listitem>
+<listitem><para><action>Isto irá mostrar a versão da aplicação e as informações do autor.</action></para></listitem>
 </varlistentry>
 
 <varlistentry>
Index: kdoctools/customization/pt-BR/entities/install-intro.docbook
===================================================================
--- kdoctools/customization/pt-BR/entities/install-intro.docbook	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdoctools/customization/pt-BR/entities/install-intro.docbook	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -1,4 +1,4 @@
 <!-- requires that packagename is defined in the documentation prologue -->
-<para>&kappname; faz parte do projeto &kde; &kde-http;.</para>
+<para>O &kappname; faz parte do projeto &kde; &kde-http;.</para>
 
-<para>&kappname; pode ser encontrado no pacote &package; no site &FTP; principal do projeto &kde; &kde-ftp;.</para>
+<para>O &kappname; pode ser encontrado no pacote &package; no servidor &FTP; principal do projeto &kde; &kde-ftp;.</para>
Index: kdeprint/tools/escputil/escpwidget.cpp
===================================================================
--- kdeprint/tools/escputil/escpwidget.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdeprint/tools/escputil/escpwidget.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -39,8 +39,9 @@
 public:
 	EscpFactory(QObject *parent = 0, const char *name = 0) : KLibFactory(parent, name) {}
 protected:
-	QObject* createObject(QObject *parent = 0, const char *name = 0, const char * /*classname*/ = "QObject", const QStringList& args = QStringList())
+	QObject* createObject(QObject *parent = 0, const char *name = 0, const char * className = "QObject", const QStringList& args = QStringList())
 	{
+               Q_UNUSED(className);
 		KDialogBase	*dlg = new KDialogBase(static_cast<QWidget*>(parent), name, true, i18n("EPSON InkJet Printer Utilities"), KDialogBase::Close);
 		EscpWidget	*w = new EscpWidget(dlg);
 		if (args.count() > 0)
Index: kdeprint/cups/kmcupsmanager.cpp
===================================================================
--- kdeprint/cups/kmcupsmanager.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdeprint/cups/kmcupsmanager.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -920,7 +920,7 @@
 	kdDebug(500) << "Checking for update possible" << endl;
 	delete m_socket;
         m_socket = new KNetwork::KBufferedSocket;
-	m_socket->setTimeout( 1 );
+	m_socket->setTimeout( 1500 );
 	connect( m_socket, SIGNAL( connected(const KResolverEntry&) ), 
                 SLOT( slotConnectionSuccess() ) );
 	connect( m_socket, SIGNAL( gotError( int ) ), SLOT( slotConnectionFailed( int ) ) );
@@ -958,7 +958,7 @@
 
 void KMCupsManager::slotAsyncConnect()
 {
-	kdDebug(500) << "Starting async connect" << endl;
+	kdDebug(500) << "Starting async connect to " << CupsInfos::self()->hostaddr() << endl;
 	//m_socket->startAsyncConnect();
         if (CupsInfos::self()->host().startsWith("/"))
             m_socket->connect( QString(), CupsInfos::self()->host());
Index: kdeprint/lpd/kmlpdfactory.h
===================================================================
--- kdeprint/lpd/kmlpdfactory.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdeprint/lpd/kmlpdfactory.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -29,7 +29,7 @@
 	virtual ~KLpdFactory();
 
 protected:
-	QObject* createObject(QObject *parent = 0, const char *name = 0, const char *classname = "QObject", const QStringList& args = QStringList());
+	QObject* createObject(QObject *parent = 0, const char *name = 0, const char *className = "QObject", const QStringList& args = QStringList());
 };
 
 #endif
Index: kdeprint/tests/richpage.h
===================================================================
--- kdeprint/tests/richpage.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdeprint/tests/richpage.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -12,8 +12,8 @@
 	RichPage(QWidget *parent = 0, const char *name = 0);
 	~RichPage();
 
-	void setOptions(const QMap<QString,QString>& options);
-	void getOptions(QMap<QString,QString>& options, bool incldef = false);
+	void setOptions(const QMap<QString,QString>& opts);
+	void getOptions(QMap<QString,QString>& opts, bool incldef = false);
 
 private:
 	QSpinBox	*margin_;
Index: kdeprint/management/kiconselectaction.h
===================================================================
--- kdeprint/management/kiconselectaction.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdeprint/management/kiconselectaction.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -35,7 +35,7 @@
 
 public slots:
 	virtual void setItems(const QStringList& lst, const QStringList& iconlst);
-	virtual void setCurrentItem(int);
+	virtual void setCurrentItem(int index);
 
 protected:
 	void createPopupMenu();
Index: kdeprint/kpdriverpage.h
===================================================================
--- kdeprint/kpdriverpage.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdeprint/kpdriverpage.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -30,7 +30,7 @@
 	KPDriverPage(KMPrinter *p, DrMain *d = 0, QWidget *parent = 0, const char *name = 0);
 	~KPDriverPage();
 
-	bool isValid(QString&);
+	bool isValid(QString& msg);
 	void setOptions(const QMap<QString,QString>& opts);
 	void getOptions(QMap<QString,QString>& opts, bool incldef = false);
 
Index: kdeui/kactionclasses.h
===================================================================
--- kdeui/kactionclasses.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdeui/kactionclasses.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -940,7 +940,7 @@
         return currentText();
     }
 
-    int plug( QWidget*, int index = -1 );
+    int plug( QWidget*widget, int index = -1 );
 
 public slots:
     void setFont( const QString &family );
Index: kdeui/kactionshortcutlist.h
===================================================================
--- kdeui/kactionshortcutlist.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdeui/kactionshortcutlist.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -16,13 +16,13 @@
 	virtual ~KActionShortcutList();
 
 	virtual uint count() const;
-	virtual QString name( uint i ) const;
-	virtual QString label( uint ) const;
-	virtual QString whatsThis( uint ) const;
-	virtual const KShortcut& shortcut( uint ) const;
-	virtual const KShortcut& shortcutDefault( uint ) const;
-	virtual bool isConfigurable( uint ) const;
-	virtual bool setShortcut( uint, const KShortcut& );
+	virtual QString name( uint index ) const;
+	virtual QString label( uint index ) const;
+	virtual QString whatsThis( uint index ) const;
+	virtual const KShortcut& shortcut( uint index ) const;
+	virtual const KShortcut& shortcutDefault( uint index ) const;
+	virtual bool isConfigurable( uint index ) const;
+	virtual bool setShortcut( uint index, const KShortcut& shortcut );
 
 	virtual const KInstance* instance() const;
 
@@ -54,13 +54,13 @@
 	virtual ~KActionPtrShortcutList();
 
 	virtual uint count() const;
-	virtual QString name( uint i ) const;
-	virtual QString label( uint ) const;
-	virtual QString whatsThis( uint ) const;
-	virtual const KShortcut& shortcut( uint ) const;
-	virtual const KShortcut& shortcutDefault( uint ) const;
-	virtual bool isConfigurable( uint ) const;
-	virtual bool setShortcut( uint, const KShortcut& );
+	virtual QString name( uint index ) const;
+	virtual QString label( uint index ) const;
+	virtual QString whatsThis( uint index ) const;
+	virtual const KShortcut& shortcut( uint index ) const;
+	virtual const KShortcut& shortcutDefault( uint index ) const;
+	virtual bool isConfigurable( uint index ) const;
+	virtual bool setShortcut( uint index, const KShortcut& shortcut);
 
 	virtual QVariant getOther( Other, uint index ) const;
 	virtual bool setOther( Other, uint index, QVariant );
Index: kdeui/kpanelappmenu.h
===================================================================
--- kdeui/kpanelappmenu.h	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdeui/kpanelappmenu.h	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -80,7 +80,7 @@
      * @internal
      */
     bool process(const QCString &fun, const QByteArray &data,
-		 QCString &replyType, QByteArray &reply);
+		 QCString &replyType, QByteArray &replyData);
 
 signals:
     /**
Index: kdeui/kedittoolbar.cpp
===================================================================
--- kdeui/kedittoolbar.cpp	(.../tags/KDE/3.5.5/kdelibs)	(revision 594434)
+++ kdeui/kedittoolbar.cpp	(.../branches/KDE/3.5/kdelibs)	(revision 594434)
@@ -208,7 +208,7 @@
 {
 public:
     /**
-     *
+     * @param instance The instance.
      * @param collection In a non-KParts application, this is the collection passed
      * to the KEditToolbar constructor.
      * In a KParts application we let create a KXMLGUIClient create a dummy one,

Property changes on: .
___________________________________________________________________
Name: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.5/kde-common/admin


