Index: kmconfiggeneral.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdeprint/management/kmconfiggeneral.cpp,v
retrieving revision 1.2
retrieving revision 1.3
diff -u -u -p -r1.2 -r1.3
--- kdelibs/kdeprint/management/kmconfiggeneral.cpp	2001/05/25 08:28:23	1.2
+++ kdelibs/kdeprint/management/kmconfiggeneral.cpp	2001/11/23 16:57:04	1.3
@@ -31,6 +31,7 @@
 #include <knuminput.h>
 #include <kmessagebox.h>
 #include <kcursor.h>
+#include <klineedit.h>
 
 KMConfigGeneral::KMConfigGeneral(QWidget *parent)
 : KMConfigPage(parent,"ConfigTimer")
@@ -47,10 +48,11 @@ KMConfigGeneral::KMConfigGeneral(QWidget
 	QGroupBox	*m_testpagebox = new QGroupBox(0, Qt::Vertical, i18n("Test page"), this);
 	m_defaulttestpage = new QCheckBox(i18n("Use non default test page"), m_testpagebox, "TestPageCheck");
 	m_testpage = new KURLRequester(m_testpagebox,"TestPage");
-	QPushButton	*m_preview = new QPushButton(i18n("Preview..."), m_testpagebox);
+	m_preview = new QPushButton(i18n("Preview..."), m_testpagebox);
 	connect(m_defaulttestpage,SIGNAL(toggled(bool)),m_testpage,SLOT(setEnabled(bool)));
-	connect(m_defaulttestpage,SIGNAL(toggled(bool)),m_preview,SLOT(setEnabled(bool)));
+	connect(m_defaulttestpage,SIGNAL(toggled(bool)),this,SLOT(setEnabledPreviewButton(bool)));
 	connect(m_preview,SIGNAL(clicked()),SLOT(slotTestPagePreview()));
+        connect(m_testpage->lineEdit(),SIGNAL(textChanged ( const QString & )),this,SLOT(testPageChanged(const QString & )));
 	m_testpage->setDisabled(true);
 	m_preview->setDisabled(true);
 	m_defaulttestpage->setCursor(KCursor::handCursor());
@@ -70,6 +72,17 @@ KMConfigGeneral::KMConfigGeneral(QWidget
 	lay2->addLayout(lay3);
 	lay3->addWidget(m_preview);
 	lay3->addStretch(1);
+        m_preview->setEnabled( !m_testpage->lineEdit()->text().isEmpty());
+}
+
+void KMConfigGeneral::testPageChanged(const QString &test )
+{
+    m_preview->setEnabled( !test.isEmpty());
+}
+
+void KMConfigGeneral::setEnabledPreviewButton(bool b)
+{
+    m_preview->setEnabled(!m_testpage->lineEdit()->text().isEmpty()&&b);
 }
 
 void KMConfigGeneral::loadConfig(KConfig *conf)
Index: kmconfiggeneral.h
===================================================================
RCS file: /home/kde/kdelibs/kdeprint/management/kmconfiggeneral.h,v
retrieving revision 1.1
retrieving revision 1.2
diff -u -u -p -r1.1 -r1.2
--- kdelibs/kdeprint/management/kmconfiggeneral.h	2001/03/30 21:48:48	1.1
+++ kdelibs/kdeprint/management/kmconfiggeneral.h	2001/11/23 16:57:05	1.2
@@ -25,7 +25,7 @@
 class KIntNumInput;
 class KURLRequester;
 class QCheckBox;
-
+class QPushButton;
 class KMConfigGeneral : public KMConfigPage
 {
 	Q_OBJECT
@@ -37,11 +37,13 @@ public:
 
 protected slots:
 	void slotTestPagePreview();
-
+        void testPageChanged(const QString & );
+	void setEnabledPreviewButton(bool b);
 private:
 	KIntNumInput	*m_timer;
 	KURLRequester	*m_testpage;
 	QCheckBox	*m_defaulttestpage;
+	QPushButton	*m_preview;
 };
 
 #endif
