--- kdelibs-3.0/kdecore/kiconloader.cpp.than	Wed Mar 27 00:18:08 2002
+++ kdelibs-3.0/kdecore/kiconloader.cpp	Wed Mar 27 00:22:56 2002
@@ -546,9 +546,15 @@
             {
                 // Try "User" icon too. Some apps expect this.
                 pix = loadIcon(name, KIcon::User, size, state, path_store, true);
-                if (!pix.isNull() || canReturnNull)
+                if (!pix.isNull() || canReturnNull) {
+                    if ((group == KIcon::Small) && (pix.width() > 20 || pix.height() > 20)) {
+                        QImage tmp = pix.convertToImage();
+                        tmp = tmp.smoothScale(20, 20);
+                        pix.convertFromImage(tmp);
+                    }
                     return pix;
-
+                }
+		
                 icon = findMatchingIcon(str_unknown, size);
                 if (!icon.isValid())
                 {
