diff -urN kdelibs-3.0.orig/kdecore/kstandarddirs.cpp kdelibs-3.0/kdecore/kstandarddirs.cpp
--- kdelibs-3.0.orig/kdecore/kstandarddirs.cpp	Mon Mar 18 12:42:54 2002
+++ kdelibs-3.0/kdecore/kstandarddirs.cpp	Thu Apr 11 19:13:32 2002
@@ -724,9 +724,9 @@
     if (!strcmp(type, "data"))
 	return "share/apps/";
     if (!strcmp(type, "html"))
-	return "share/doc/HTML/";
+	return "share/doc/kde/HTML/";
     if (!strcmp(type, "icon"))
-	return "share/icons/";
+	return "share/pixmaps/";
     if (!strcmp(type, "config"))
 	return "share/config/";
     if (!strcmp(type, "pixmap"))
@@ -966,6 +966,7 @@
     addResourceDir("cache", dir);
 
     addResourceDir("home", QDir::homeDirPath());
+    addResourceDir("html", "/usr/share/doc/kde/HTML");
 }
 
 void KStandardDirs::checkConfig() const
