--- kdelibs/kio/misc/fileshareset.orig	2003-11-21 19:38:11.000000000 +0100
+++ kdelibs/kio/misc/fileshareset	2003-11-21 19:41:50.000000000 +0100
@@ -80,15 +80,19 @@
     $< = $>;
     @ARGV == 2 && ($cmd eq '--add' || $cmd eq '--remove') or error($usage);
 
-    verify_mntpoint($dir);
-
     if ($cmd eq '--add') {
-	my @errs = map { eval { $_->add($dir) }; $@ } @exports;
-	grep { !$_ } @errs or error("already exported");
-    } else {
-	my @errs = map { eval { $_->remove($dir) }; $@ } @exports;
-	grep { !$_ } @errs or error("not exported");
-    }    
+		verify_mntpoint($dir);
+		my @errs = map { eval { $_->add($dir) }; $@ } @exports;
+		grep { !$_ } @errs or error("already exported");
+    }
+	elsif ($cmd eq '--remove') {
+		verify_mntpoint($dir, 1);
+		my @errs = map { eval { $_->remove($dir) }; $@ } @exports;
+		grep { !$_ } @errs or error("not exported");
+    }
+	else {
+		die("42?");
+	}
     foreach my $export (@exports) {
 	$export->write;
 	$export->update_server;
@@ -101,13 +105,24 @@
 sub own { $uid == 0 || (stat($_[0]))[4] == $uid }
 
 sub verify_mntpoint {
-    local ($_) = @_;
+    $_ = shift;
+	my $mercy = shift; # true if dir doesn't have to exist
     my $ok = 1;
-    $ok &&= m|^/|;
-    $ok &&= !m|/../|;
-    $ok &&= !m|[\0\n\r]|;
-    $ok &&= -d $_;
-    $ok &&= own($_);
+	$ok &&= length $_;
+    $ok &&=  m#^/#;
+    $ok &&= !m#/../#;
+    $ok &&= !m#[\0\n\r]#;
+	if ($ok && $mercy && !-d && m#^/#) {
+		my $tmp = $_;
+		$tmp =~ s#/*$##;
+		1 while $tmp =~ s#/[^/]*$## && length $tmp && !-d $tmp;
+		$ok &&= length $tmp;
+		$ok &&= own($tmp);
+	}
+	else {
+		$ok &&= -d;
+		$ok &&= own($_);
+	}
     $ok or error("invalid mount point");
 }
 
@@ -276,16 +291,16 @@
     my ($s, @l);
     local *F;
     open F, $conf_file;
-    local $_;
     while (<F>) {
-	if (/^\s*\[.*\]/ || eof F) {
-	    #- first line in the category
-	    my ($label) = $s =~ /^\s*\[(.*)\]/;
-	    my ($mntpoint) = $s =~ /^\s*path\s*=\s*(.*)/m;
-	    push @l, { mntpoint => $mntpoint, raw => $s, label => $label };
-	    $s = '';
-	}
-	$s .= $_;
+		if (/^\s*\[.*\]/ || eof F) {
+			#- first line in the category
+			my ($label) = $s =~ /^\s*\[(.*)\]/;
+			my ($mntpoint) = $s =~ /^\s*path\s*=\s*(.*)/m;
+			$s .= $_ if eof F;
+			push @l, { mntpoint => $mntpoint, raw => $s, label => $label };
+			$s = '';
+		}
+		$s .= $_;
     }
     bless \@l, 'smb_exports';
 }
@@ -387,3 +402,5 @@
 	exit 0;
     }
 }
+
+# vim: ts=4 sw=4 noet
