--- kdelibs-2.1.1/kio/http/kcookiejar/kcookiejar.cpp.sec	Wed Mar 21 15:53:03 2001
+++ kdelibs-2.1.1/kio/http/kcookiejar/kcookiejar.cpp	Wed Mar 21 15:53:14 2001
@@ -1025,7 +1025,7 @@
 // On failure 'false' is returned.
 bool KCookieJar::saveCookies(const QString &_filename)
 {
-    KSaveFile saveFile(_filename);
+    KSaveFile saveFile(_filename, 0600);
 
     if (saveFile.status() != 0)
        return false;
