--- kdelibs-R14.0.0/tdecore/kiconloader.cpp~	2015-03-22 11:42:49.000000000 +0200
+++ kdelibs-R14.0.0/tdecore/kiconloader.cpp	2015-03-22 11:42:59.774075564 +0200
@@ -238,6 +238,11 @@
 	if (!d->mpGroups[i].size)
 	    d->mpGroups[i].size = d->mpThemeRoot->theme->defaultSize(i);
     }
+    // Make the app icons themable!!!
+    d->mpDirs->addResourceType("appicon", TDEStandardDirs::kde_default("icon") +
+		TDEIconTheme::current() + appname + "/pics/");
+    d->mpDirs->addResourceType("appicon", TDEStandardDirs::kde_default("icon") +
+		TDEIconTheme::current() + appname + "/toolbar/");
     // Insert application specific themes at the top.
     d->mpDirs->addResourceType("appicon", TDEStandardDirs::kde_default("data") +
 		appname + "/pics/");
