diff -urN -x CVS kdelibs.orig/interfaces/kimproxy/interface/dcopinstantmessenger.desktop kdelibs/interfaces/kimproxy/interface/dcopinstantmessenger.desktop
--- kdelibs.orig/interfaces/kimproxy/interface/dcopinstantmessenger.desktop	Mon Sep 20 15:53:09 2004
+++ kdelibs/interfaces/kimproxy/interface/dcopinstantmessenger.desktop	Mon Oct  4 15:24:48 2004
@@ -32,7 +32,7 @@
 Comment[sk]=Instant Messanger s rozhraním DCOP
 Comment[sl]=Takojšni sporočilnik z vmesnikom DCOP
 Comment[sr]=Брзи гласник са DCOP интерфејсом
-Comment[sr@Latn]=Брзи гласник са DCOP интерфејсом
+Comment[sr@Latn]=Brzi glasnik sa DCOP interfejsom
 Comment[sv]=Direktmeddelanden med ett DCOP-gränssnitt
 Comment[ta]=உடனடி செய்தியாளர் DCOP இடைமுகத்தோடு
 Comment[tg]=Коргири хизматгоҳи AOL пайғоми ахборот
diff -urN -x CVS kdelibs.orig/interfaces/kimproxy/interface/kcm_instantmessenger.desktop kdelibs/interfaces/kimproxy/interface/kcm_instantmessenger.desktop
--- kdelibs.orig/interfaces/kimproxy/interface/kcm_instantmessenger.desktop	Mon Sep 20 15:53:09 2004
+++ kdelibs/interfaces/kimproxy/interface/kcm_instantmessenger.desktop	Mon Oct  4 15:24:48 2004
@@ -24,7 +24,7 @@
 Name[se]=Šleađgadieđut
 Name[sl]=Takojšni sporočilnik
 Name[sr]=Брзи гласник
-Name[sr@Latn]=Брзи гласник
+Name[sr@Latn]=Brzi glasnik
 Name[sv]=Direktmeddelanden
 Name[ta]=உடனடி செய்தியாளர்
 Name[tg]=Пайғоми ахборот
@@ -61,7 +61,7 @@
 Comment[sk]=Instant messenger umožňuje rozhovor medzi rôznymi osobami a skupinami.
 Comment[sl]=Takojšni sporočilnik omogoča dvosmeren klepet med posamezniki in skupinami.
 Comment[sr]=Брзи гласник вам омогућава двосмерно ћаскање између индивидуа и група.
-Comment[sr@Latn]=Брзи гласник вам омогућава двосмерно ћаскање између индивидуа и група.
+Comment[sr@Latn]=Brzi glasnik vam omogućava dvosmerno ćaskanje između individua i grupa.
 Comment[sv]=Direktmeddelanden möjliggör tvåvägschatt mellan individer och grupper.
 Comment[ta]=உடனடி தகவல் உங்களை இரண்டு வழியில் தனிநபர் மற்றும் குழுக்கள் இடையே அனுமதிக்கும்.
 Comment[tg]=Пайғоми ахборот ба шумо иҷозат медиҳад, ки ду роҳигуфтугӯиро байни шахсҳо ва гурӯҳҳоро ба истифода баред.
diff -urN -x CVS kdelibs.orig/interfaces/ktexteditor/ktexteditorplugin.desktop kdelibs/interfaces/ktexteditor/ktexteditorplugin.desktop
--- kdelibs.orig/interfaces/ktexteditor/ktexteditorplugin.desktop	Tue Aug 24 08:28:52 2004
+++ kdelibs/interfaces/ktexteditor/ktexteditorplugin.desktop	Mon Oct  4 15:24:48 2004
@@ -55,7 +55,7 @@
 Comment[sl]=Vstavek KTextEditor
 Comment[sq]=Shtojca KTextEditor
 Comment[sr]=KTextEditor прикључак
-Comment[sr@Latn]=KTextEditor dodatak
+Comment[sr@Latn]=KTextEditor priključak
 Comment[ss]=I-plugini ye KTextEditor 
 Comment[sv]=Insticksprogram för texteditor
 Comment[ta]=கேஉரைதொகுப்பி சொருகுப்பொருள்
diff -urN -x CVS kdelibs.orig/kabc/ldif.cpp kdelibs/kabc/ldif.cpp
--- kdelibs.orig/kabc/ldif.cpp	Tue Oct  5 18:50:42 2004
+++ kdelibs/kabc/ldif.cpp	Tue Oct 12 22:29:16 2004
@@ -25,8 +25,6 @@
 
 using namespace KABC;
 
-static const char delimiter[ 3 ] = { '\n', '\n', '\n' };
-
 LDIF::LDIF()
 {
   startParsing();
@@ -90,9 +88,11 @@
   QCString ret;
   QByteArray tmp;
   uint valuelen = value.length();
-  tmp.setRawData( value, valuelen );
+  const char *data = value.data();
+
+  tmp.setRawData( data, valuelen );
   ret = assembleLine( fieldname, tmp, linelen, url );
-  tmp.resetRawData( value, valuelen );
+  tmp.resetRawData( data, valuelen );
   return ret;
 
 }
@@ -108,6 +108,7 @@
   int position;
   QByteArray tmp;
   int linelen;
+  const char *data;
 
 //  kdDebug(5700) << "splitLine line: " << QString::fromUtf8(line) << endl;
 
@@ -118,9 +119,10 @@
     QCString str;
     str = line.stripWhiteSpace();
     linelen = str.length();
-    tmp.setRawData( str.data(), linelen );
+    data = str.data();
+    tmp.setRawData( data, linelen );
     value = tmp.copy();
-    tmp.resetRawData( str.data(), linelen );
+    tmp.resetRawData( data, linelen );
 //    kdDebug(5700) << "value : " << value[0] << endl;
     return false;
   }
@@ -135,9 +137,10 @@
       value.resize( 0 );
       return false;
     }
-    tmp.setRawData( &line.data()[ position + 3 ], linelen - position - 3 );
+    data = &line.data()[ position + 3 ];
+    tmp.setRawData( data, linelen - position - 3 );
     KCodecs::base64Decode( tmp, value );
-    tmp.resetRawData( &line.data()[ position + 3 ], linelen - position - 3 );
+    tmp.resetRawData( data, linelen - position - 3 );
     return false;
   }
 
@@ -149,9 +152,10 @@
       value.resize( 0 );
       return false;
     }
-    tmp.setRawData( &line.data()[ position + 3 ], linelen - position - 3 );
+    data = &line.data()[ position + 3];
+    tmp.setRawData( data, linelen - position - 3 );
     value = tmp.copy();
-    tmp.resetRawData( &line.data()[ position + 3 ], linelen - position - 3 );
+    tmp.resetRawData( data, linelen - position - 3 );
     return true;
   }
 
@@ -160,9 +164,10 @@
     value.resize( 0 );
     return false;
   }
-  tmp.setRawData( &line.data()[ position + 2 ], linelen - position - 2 );
+  data = &line.data()[ position + 2 ];
+  tmp.setRawData( data, linelen - position - 2 );
   value = tmp.copy();
-  tmp.resetRawData( &line.data()[ position + 2 ], linelen - position - 2 );
+  tmp.resetRawData( data, linelen - position - 2 );
   return false;
 }
 
@@ -338,7 +343,11 @@
 
 void LDIF::endLDIF()
 {
-  mLdif.resetRawData( delimiter, 3 );  
+  QByteArray tmp( 3 );
+  tmp[ 0 ] = '\n';
+  tmp[ 1 ] = '\n';
+  tmp[ 2 ] = '\n';
+  mLdif = tmp;
   mPos = 0;
 }
 
diff -urN -x CVS kdelibs.orig/kate/data/bash.xml kdelibs/kate/data/bash.xml
--- kdelibs.orig/kate/data/bash.xml	Wed Sep 29 10:02:40 2004
+++ kdelibs/kate/data/bash.xml	Wed Oct 13 08:40:49 2004
@@ -3,10 +3,11 @@
 [
 	<!ENTITY funcname "[A-Za-z_:][A-Za-z0-9_:#&#37;@-]*">
 	<!ENTITY varname  "[A-Za-z_][A-Za-z0-9_]*">
+	<!ENTITY word     "[^|&amp;;()&lt;&gt;\s]+">	<!-- see man bash -->
 	<!ENTITY eos      "(?=($|\s))">			<!-- eol or space following -->
 	<!ENTITY noword   "(?![\w$+-])">		<!-- no word, $, + or - following -->
 ]>
-<language name="Bash" version="2.02" kateversion="2.3" section="Scripts" extensions="*.sh;*.bash;*.ebuild;*.eclass" mimetype="application/x-shellscript" casesensitive="1" author="Wilbert Berendsen (wilbert@kde.nl)" license="LGPL">
+<language name="Bash" version="2.04" kateversion="2.3" section="Scripts" extensions="*.sh;*.bash;*.ebuild;*.eclass" mimetype="application/x-shellscript" casesensitive="1" author="Wilbert Berendsen (wilbert@kde.nl)" license="LGPL">
 
 <!-- (c) 2004 by Wilbert Berendsen, wilbert@kde.nl
     Released under the LGPL, part of kdelibs/kate -->
@@ -514,8 +515,6 @@
         <keyword attribute="Command" context="#stay" String="unixcommands" />
 	<!-- handle commands that have variable names as argument -->
 	<keyword attribute="Builtin" context="VarName" String="builtins_var" />
-	<!-- handle &, &&, | and || -->
-	<RegExpr attribute="Control" context="#stay" String="([|&amp;])\1?" />
         <!-- handle here-string -->
 	<StringDetect attribute="Redirection" context="#stay" String="&lt;&lt;&lt;" />
 	<!-- handle here document -->
@@ -524,6 +523,8 @@
         <RegExpr attribute="Redirection" context="ProcessSubst" String="[&lt;&gt;]\(" />
         <!-- handle redirection -->
         <RegExpr attribute="Redirection" context="#stay" String="([0-9]*(&gt;{1,2}|&lt;)(&amp;[0-9]+-?)?|&amp;&gt;|&gt;&amp;|[0-9]*&lt;&gt;)" />
+	<!-- handle &, &&, | and || -->
+	<RegExpr attribute="Control" context="#stay" String="([|&amp;])\1?" />
 	<!-- mark function definitions without function keyword -->
         <RegExpr attribute="Function" context="#stay" String="&funcname;\s*\(\)" />
       </context>
@@ -759,14 +760,14 @@
 
       <!-- HereDoc consumes Here-documents. It is called at the beginning of the "<<" construct. -->
       <context attribute="Normal Text" lineEndContext="#stay" name="HereDoc">
-	<RegExpr attribute="Redirection" context="HereDocQ"   String="(&lt;&lt;\s*&quot;(&varname;)&quot;)" lookAhead="true" />
-	<RegExpr attribute="Redirection" context="HereDocQ"   String="(&lt;&lt;\s*'(&varname;)')" lookAhead="true" />
-	<RegExpr attribute="Redirection" context="HereDocQ"   String="(&lt;&lt;\s*\\(&varname;))" lookAhead="true" />
-	<RegExpr attribute="Redirection" context="HereDocNQ"  String="(&lt;&lt;\s*(&varname;))" lookAhead="true" />
-	<RegExpr attribute="Redirection" context="HereDocIQ"  String="(&lt;&lt;-\s*&quot;(&varname;)&quot;)" lookAhead="true" />
-	<RegExpr attribute="Redirection" context="HereDocIQ"  String="(&lt;&lt;-\s*'(&varname;)')" lookAhead="true" />
-	<RegExpr attribute="Redirection" context="HereDocIQ"  String="(&lt;&lt;-\s*\\(&varname;))" lookAhead="true" />
-	<RegExpr attribute="Redirection" context="HereDocINQ" String="(&lt;&lt;-\s*(&varname;))" lookAhead="true" />
+	<RegExpr attribute="Redirection" context="HereDocQ"   String="(&lt;&lt;\s*&quot;(&word;)&quot;)" lookAhead="true" />
+	<RegExpr attribute="Redirection" context="HereDocQ"   String="(&lt;&lt;\s*'(&word;)')" lookAhead="true" />
+	<RegExpr attribute="Redirection" context="HereDocQ"   String="(&lt;&lt;\s*\\(&word;))" lookAhead="true" />
+	<RegExpr attribute="Redirection" context="HereDocNQ"  String="(&lt;&lt;\s*(&word;))" lookAhead="true" />
+	<RegExpr attribute="Redirection" context="HereDocIQ"  String="(&lt;&lt;-\s*&quot;(&word;)&quot;)" lookAhead="true" />
+	<RegExpr attribute="Redirection" context="HereDocIQ"  String="(&lt;&lt;-\s*'(&word;)')" lookAhead="true" />
+	<RegExpr attribute="Redirection" context="HereDocIQ"  String="(&lt;&lt;-\s*\\(&word;))" lookAhead="true" />
+	<RegExpr attribute="Redirection" context="HereDocINQ" String="(&lt;&lt;-\s*(&word;))" lookAhead="true" />
 	<StringDetect attribute="Redirection" context="#pop"  String="&lt;&lt;" /><!-- always met -->
       </context>
 
diff -urN -x CVS kdelibs.orig/kate/part/kateautoindent.cpp kdelibs/kate/part/kateautoindent.cpp
--- kdelibs.orig/kate/part/kateautoindent.cpp	Tue Sep 14 10:23:13 2004
+++ kdelibs/kate/part/kateautoindent.cpp	Tue Oct 12 22:15:58 2004
@@ -495,7 +495,7 @@
 void KateCSmartIndent::processChar(QChar c)
 {
   static const QString triggers("}{)/:;#n");
-  if (triggers.find(c, true) == -1)
+  if (triggers.find(c) < 0)
     return;
 
   KateView *view = doc->activeView();
diff -urN -x CVS kdelibs.orig/kate/part/katebuffer.cpp kdelibs/kate/part/katebuffer.cpp
--- kdelibs.orig/kate/part/katebuffer.cpp	Sat May 22 17:21:31 2004
+++ kdelibs/kate/part/katebuffer.cpp	Wed Oct  6 12:36:16 2004
@@ -88,18 +88,18 @@
       , m_decoder (codec->makeDecoder())
       , m_position (0)
       , m_lastLineStart (0)
-      , m_eof (false) // default to not eof     
+      , m_eof (false) // default to not eof
       , lastWasEndOfLine (true) // at start of file, we had a virtual newline
       , lastWasR (false) // we have not found a \r as last char
       , m_eol (-1) // no eol type detected atm
     {
     }
-    
+
     ~KateFileLoader ()
     {
       delete m_decoder;
     }
-    
+
     /**
      * open file, read first chunk of data, detect eol
      */
@@ -108,12 +108,12 @@
       if (m_file.open (IO_ReadOnly))
       {
         int c = m_file.readBlock (m_buffer.data(), m_buffer.size());
-            
+
         if (c > 0)
           m_text = m_decoder->toUnicode (m_buffer, c);
-          
-        m_eol = m_file.atEnd();
-        
+
+        m_eof = m_file.atEnd();
+
         for (uint i=0; i < m_text.length(); i++)
         {
           if (m_text[i] == '\n')
@@ -135,23 +135,23 @@
             }
           }
         }
-      
+
         return true;
       }
-      
+
       return false;
     }
-    
+
     // no new lines around ?
     inline bool eof () const { return m_eof && !lastWasEndOfLine && (m_lastLineStart == m_text.length()); }
-    
+
     // eol mode ? autodetected on open(), -1 for no eol found in the first block!
     inline int eol () const { return m_eol; }
 
     // read a line, return per reference, only valid until the next readLine call
     // or until this object goes to trash !!!
     QConstString readLine ()
-    {  
+    {
       while (m_position <= m_text.length())
       {
         if (m_position == m_text.length())
@@ -160,48 +160,48 @@
           if (!m_eof)
           {
             int c = m_file.readBlock (m_buffer.data(), m_buffer.size());
-            
+
             if (c > 0)
               m_text = m_text.mid (m_lastLineStart, m_position-m_lastLineStart)
                        + m_decoder->toUnicode (m_buffer, c);
             else
               m_text = m_text.mid (m_lastLineStart, m_position-m_lastLineStart);
-            
+
             // is file completly read ?
             m_eof = m_file.atEnd();
-            
+
             // recalc current pos and last pos
             m_position -= m_lastLineStart;
             m_lastLineStart = 0;
           }
-          
+
           // oh oh, end of file, escape !
           if (m_eof && (m_position == m_text.length()))
           {
             lastWasEndOfLine = false;
-          
+
             QConstString line = QConstString (m_text.unicode()+m_lastLineStart, m_position-m_lastLineStart);
             m_lastLineStart = m_position;
-            
+
             return line;
           }
         }
-        
+
         if (m_text[m_position] == '\n')
         {
           lastWasEndOfLine = true;
-        
+
           if (lastWasR)
           {
             m_lastLineStart++;
             lastWasR = false;
           }
           else
-          {         
+          {
             QConstString line = QConstString (m_text.unicode()+m_lastLineStart, m_position-m_lastLineStart);
             m_lastLineStart = m_position+1;
             m_position++;
-                  
+
             return line;
           }
         }
@@ -209,11 +209,11 @@
         {
           lastWasEndOfLine = true;
           lastWasR = true;
-          
+
           QConstString line = QConstString (m_text.unicode()+m_lastLineStart, m_position-m_lastLineStart);
           m_lastLineStart = m_position+1;
           m_position++;
-          
+
           return line;
         }
         else
@@ -221,13 +221,13 @@
           lastWasEndOfLine = false;
           lastWasR = false;
         }
-        
+
         m_position++;
       }
-      
+
       return QConstString (m_text.unicode(), 0);
-    } 
-    
+    }
+
   private:
     QFile m_file;
     QByteArray m_buffer;
@@ -415,11 +415,11 @@
     clear();
     return false; // Error
   }
-  
+
   // set eol mode, if a eol char was found in the first 256kb block!
   if (file.eol() != -1)
     m_doc->config()->setEol (file.eol());
-  
+
   // flush current content
   clear ();
 
@@ -463,7 +463,7 @@
     // fix region tree
     m_regionTree.fixRoot (m_lines);
   }
-  
+
   // if we have no hl or the "None" hl activated, whole file is correct highlighted
   // after loading, which wonder ;)
   if (!m_highlight || m_highlight->noHighlighting())
@@ -471,7 +471,7 @@
     m_lineHighlighted = m_lines;
     m_lineHighlightedMax = m_lines;
   }
-  
+
   return !m_loadingBorked;
 }
 
@@ -648,7 +648,7 @@
       {
         // remember this block as last found !
         m_lastFoundBlock = m_lastInSyncBlock;
-      
+
         if (index)
           (*index) = m_lastFoundBlock;
 
@@ -700,7 +700,7 @@
   // last sync block adjust
   if (m_lastInSyncBlock > index)
     m_lastInSyncBlock = index;
-    
+
   // last found
   if (m_lastInSyncBlock < m_lastFoundBlock)
     m_lastFoundBlock = m_lastInSyncBlock;
@@ -755,7 +755,7 @@
     if (m_lastInSyncBlock > index)
       m_lastInSyncBlock = index;
   }
-  
+
   // last found
   if (m_lastInSyncBlock < m_lastFoundBlock)
     m_lastFoundBlock = m_lastInSyncBlock;
diff -urN -x CVS kdelibs.orig/kate/part/katecmds.h kdelibs/kate/part/katecmds.h
--- kdelibs.orig/kate/part/katecmds.h	Wed May 12 18:36:13 2004
+++ kdelibs/kate/part/katecmds.h	Mon Oct  4 23:30:48 2004
@@ -44,9 +44,9 @@
      * @return success
      */
     bool exec( class Kate::View *view, const QString &cmd, QString &errorMsg );
-  
+
     bool help( class Kate::View *, const QString &, QString & ) {return false;};
-  
+
     /**
      * supported commands as prefixes
      * @return prefix list
@@ -82,8 +82,7 @@
      * supported commands as prefixes
      * @return prefix list
      */
-    QStringList cmds () { return QStringList("s"); };
-
+    QStringList cmds () { QStringList l; l << "s"<<"%s" /*<<"$s"*/; return l; };
   private:
     static QString sedMagic(QString textLine, const QString &find, const QString &replace, bool noCase, bool repeat);
 };
diff -urN -x CVS kdelibs.orig/kate/part/katedocument.cpp kdelibs/kate/part/katedocument.cpp
--- kdelibs.orig/kate/part/katedocument.cpp	Tue Sep 28 09:34:27 2004
+++ kdelibs/kate/part/katedocument.cpp	Fri Oct 15 12:54:32 2004
@@ -3276,7 +3276,7 @@
   if (!(config()->configFlags() & KateDocument::cfKeepSelection))
     clearSelection ();
 
-  setSelection (cursor.line(), 0, cursor.line()/*+1, 0*/, m_buffer->plainLine(cursor.line())->length() );
+  setSelection (cursor.line(), 0, cursor.line()+1, 0 );
 }
 
 void KateDocument::selectLength( const KateTextCursor& cursor, int length )
@@ -4342,28 +4342,51 @@
   {
     m_isasking = 1;
 
-    int exitval = ( v && v->hasFocus() ? 0 : -1 );
-
-    switch ( KMessageBox::warningYesNoCancel( widget(),
-                reasonedMOHString() + "\n\n" + i18n("What do you want to do?"),
-                i18n("File Was Modified on Disk"),
-                i18n("&Reload File"), i18n("&Ignore Changes")) )
+    if ( m_modOnHdReason == 3 ) // deleted
     {
-      case KMessageBox::Yes: // "reload file"
-        m_modOnHd = false; // trick reloadFile() to not ask again
-        emit modifiedOnDisc( this, false, 0 );
-        reloadFile();
-        m_isasking = 0;
-        break;
-
-      case KMessageBox::No:  // "ignore changes"
-        m_modOnHd = false;
-        emit modifiedOnDisc( this, false, 0 );
-        m_isasking = 0;
-        break;
+      switch ( KMessageBox::warningYesNoCancel( widget(),
+               reasonedMOHString() + "\n\n" + i18n("What do you want to do?"),
+               i18n("File Was Deleted on Disk"),
+               i18n("&Save As..."), i18n("&Ignore Changes")) )
+      {
+        case KMessageBox::Yes: // "save file"
+          m_modOnHd = false; // trick save() to not ask again
+          emit modifiedOnDisc( this, false, 0 );
+          saveAs();
+          m_isasking = 0;
+          break;
+
+          case KMessageBox::No:  // "ignore changes"
+            m_modOnHd = false;
+            emit modifiedOnDisc( this, false, 0 );
+            m_isasking = 0;
+            break;
 
-      default:               // cancel: ignore next focus event
-        m_isasking = -1;
+            default:               // cancel: ignore next focus event
+              m_isasking = -1;
+      }
+    } else {
+      switch ( KMessageBox::warningYesNoCancel( widget(),
+               reasonedMOHString() + "\n\n" + i18n("What do you want to do?"),
+               i18n("File Was Modified on Disk"),
+               i18n("&Reload File"), i18n("&Ignore Changes")) )
+      {
+        case KMessageBox::Yes: // "reload file"
+          m_modOnHd = false; // trick reloadFile() to not ask again
+          emit modifiedOnDisc( this, false, 0 );
+          reloadFile();
+          m_isasking = 0;
+          break;
+
+          case KMessageBox::No:  // "ignore changes"
+            m_modOnHd = false;
+            emit modifiedOnDisc( this, false, 0 );
+            m_isasking = 0;
+            break;
+
+            default:               // cancel: ignore next focus event
+              m_isasking = -1;
+      }
     }
   }
 }
diff -urN -x CVS kdelibs.orig/kate/part/katehighlight.cpp kdelibs/kate/part/katehighlight.cpp
--- kdelibs.orig/kate/part/katehighlight.cpp	Mon Sep 13 11:57:17 2004
+++ kdelibs/kate/part/katehighlight.cpp	Sun Oct 10 22:03:32 2004
@@ -1989,6 +1989,9 @@
 
 QString KateHighlighting::getCommentString( int which, int attrib ) const
 {
+  if ( noHl )
+    return which == 3 ? stdDeliminator : "";
+
   int k = hlKeyForAttrib( attrib );
   const QStringList& lst = m_additionalData[k];
   return lst.isEmpty() ? QString::null : lst[which];
diff -urN -x CVS kdelibs.orig/kate/part/katerenderer.cpp kdelibs/kate/part/katerenderer.cpp
--- kdelibs.orig/kate/part/katerenderer.cpp	Thu Jun 10 09:33:05 2004
+++ kdelibs/kate/part/katerenderer.cpp	Sun Oct 10 09:15:51 2004
@@ -354,8 +354,7 @@
 
       // Determine current syntax highlighting attribute
       // A bit legacy but doesn't need to change
-      KateAttribute* curAt = (!noAttribs && (*a) >= atLen) ? &at[0] : &at[*a];
-
+      KateAttribute* curAt = (noAttribs || ((*a) >= atLen)) ? &at[0] : &at[*a];
       // X position calculation. Incorrect for fonts with non-zero leftBearing() and rightBearing() results.
       // TODO: make internal charWidth() function, use QFontMetrics::charWidth().
       xPosAfter += curAt->width(*fs, curChar, m_tabWidth);
@@ -501,6 +500,12 @@
           // Here's where the money is...
           paint.drawText(oldXPos-xStart, y, textLine->string(), oldCol, curCol+1-oldCol);
 
+          // Draw preedit's underline
+          if (isIMEdit) {
+            QRect r( oldXPos - xStart, 0, xPosAfter - oldXPos, fs->fontHeight );
+            paint.drawLine( r.bottomLeft(), r.bottomRight() );
+          }
+
           // Put pen color back
           if (isIMSel) paint.restore();
 
diff -urN -x CVS kdelibs.orig/kate/part/kateviewinternal.cpp kdelibs/kate/part/kateviewinternal.cpp
--- kdelibs.orig/kate/part/kateviewinternal.cpp	Thu Aug 26 13:06:16 2004
+++ kdelibs/kate/part/kateviewinternal.cpp	Sun Oct 10 09:13:08 2004
@@ -85,6 +85,7 @@
   , m_imPreeditStartLine(0)
   , m_imPreeditStart(0)
   , m_imPreeditLength(0)
+  , m_imPreeditSelStart(0)
 {
   setMinimumSize (0,0);
 
@@ -891,6 +892,32 @@
   return QPoint(x, y);
 }
 
+
+void KateViewInternal::updateMicroFocusHint()
+{
+  int line = 0;
+  if (m_imPreeditStartLine)
+    line = m_imPreeditStartLine;
+  else
+    line = displayViewLine(displayCursor, true);
+
+  if (line == -1)
+      return;
+  
+  uint y = line * m_view->renderer()->fontHeight();
+  
+  // Cursor placement code is changed for Asian input method that
+  // shows candidate window. This behavior is same as Qt/E 2.3.7
+  // which supports Asian input methods. Asian input methods need
+  // start point of IM selection text to place candidate window as
+  // adjacent to the selection text.
+  uint preeditStrLen = m_view->renderer()->textWidth(textLine(line), cursor.col()) - m_view->renderer()->textWidth(textLine(line), m_imPreeditSelStart);
+  uint x = cXPos - m_startX - lineRanges[line].startX + lineRanges[line].xOffset() - preeditStrLen;
+  
+  setMicroFocusHint( x, y, 0, m_view->renderer()->fontHeight() );  
+}
+
+
 void KateViewInternal::doReturn()
 {
   KateTextCursor c = cursor;
@@ -2062,8 +2089,7 @@
   tagLine(oldDisplayCursor);
   tagLine(displayCursor);
 
-  QPoint cursorP = cursorCoordinates();
-  setMicroFocusHint( cursorP.x(), cursorP.y(), 0, m_view->renderer()->fontHeight() );
+  updateMicroFocusHint();
 
   if (m_cursorTimer.isActive ())
   {
@@ -2969,6 +2995,7 @@
   m_imPreeditStartLine = cursor.line();
   m_imPreeditStart = cursor.col();
   m_imPreeditLength = 0;
+  m_imPreeditSelStart = m_imPreeditStart;
 
   m_doc->setIMSelectionValue( m_imPreeditStartLine, m_imPreeditStart, 0, 0, 0, true );
 }
@@ -2993,7 +3020,9 @@
 
   updateView( true );
   updateCursor( cursor, true );
+  
   m_imPreeditLength = e->text().length();
+  m_imPreeditSelStart = m_imPreeditStart + e->cursorPos();
 }
 
 void KateViewInternal::imEndEvent( QIMEvent *e )
@@ -3023,6 +3052,7 @@
 
   m_imPreeditStart = 0;
   m_imPreeditLength = 0;
+  m_imPreeditSelStart = 0;
 }
 
 //
diff -urN -x CVS kdelibs.orig/kate/part/kateviewinternal.h kdelibs/kate/part/kateviewinternal.h
--- kdelibs.orig/kate/part/kateviewinternal.h	Sun May 23 19:00:07 2004
+++ kdelibs/kate/part/kateviewinternal.h	Sun Oct 10 09:13:08 2004
@@ -260,6 +260,8 @@
 
     void paintCursor();
 
+    void updateMicroFocusHint();
+
     void placeCursor( const QPoint& p, bool keepSelection = false, bool updateSelection = true );
     bool isTargetSelected( const QPoint& p );
 
@@ -428,6 +430,7 @@
    int m_imPreeditStartLine;
    int m_imPreeditStart;
    int m_imPreeditLength;
+   int m_imPreeditSelStart;
 };
 
 #endif
diff -urN -x CVS kdelibs.orig/kate/plugins/autobookmarker/ktexteditor_autobookmarker.desktop kdelibs/kate/plugins/autobookmarker/ktexteditor_autobookmarker.desktop
--- kdelibs.orig/kate/plugins/autobookmarker/ktexteditor_autobookmarker.desktop	Sun Sep 12 15:55:49 2004
+++ kdelibs/kate/plugins/autobookmarker/ktexteditor_autobookmarker.desktop	Mon Oct  4 15:24:51 2004
@@ -30,7 +30,7 @@
 Name[se]=KTextEditor:a automáhtalaš girjemerkejeaddji
 Name[sk]=Automatické záložky pre KTextEditor
 Name[sr]=KTextEditor-ов аутомаркер
-Name[sr@Latn]=KTextEditor-ов аутомаркер
+Name[sr@Latn]=KTextEditor-ov automarker
 Name[sv]=Automatiska bokmärken för Ktexteditor
 Name[ta]=கேஉரைதொகுப்பாளர் தானியக்க புத்தக உருவாக்கி
 Name[tg]=Китоби худқайдкуни KTextEditor
@@ -67,7 +67,7 @@
 Comment[ru]=Установка закладок в текстовом документе по шаблону
 Comment[sk]=Nastavuje záložky pri načítaní dokumentu na riadky zodpovedajúce vzorke
 Comment[sr]=Поставља маркере на линијама које се уклапају у узорак када се учитавају документи
-Comment[sr@Latn]=Поставља маркере на линијама које се уклапају у узорак када се учитавају документи
+Comment[sr@Latn]=Postavlja markere na linijama koje se uklapaju u uzorak kada se učitavaju dokumenti
 Comment[sv]=Lägg till bokmärken på rader som motsvarar ett mönster när dokument laddas
 Comment[ta]= ஆவணங்கள் பதிவாகும் போது அமைப்பு புத்தகக்குறிப்பு கோடுகள் பொருத்தும் மாதிரி
 Comment[tg]=Вақте ки санад пурбор шудааст, монданиҳоро дар хати мутобиқат низ сабт намоед
diff -urN -x CVS kdelibs.orig/kate/plugins/insertfile/ktexteditor_insertfile.desktop kdelibs/kate/plugins/insertfile/ktexteditor_insertfile.desktop
--- kdelibs.orig/kate/plugins/insertfile/ktexteditor_insertfile.desktop	Sun Sep 12 15:55:49 2004
+++ kdelibs/kate/plugins/insertfile/ktexteditor_insertfile.desktop	Mon Oct  4 15:24:51 2004
@@ -50,7 +50,7 @@
 Name[sl]=Vstavek KTextEditor za vstavljanje datotek
 Name[sq]=KTextEditor Shtojca: Fute skedën
 Name[sr]=KTextEditor прикључак за уметање фајла
-Name[sr@Latn]=KTextEditor dodatak za umetanje fajla.
+Name[sr@Latn]=KTextEditor priključak za umetanje fajla
 Name[ss]=I-KTextEditor yingenisa i-plugin yelifayela
 Name[sv]=Ktexteditor-insticksprogram för att infoga filer
 Name[ta]=கேஉரைதொகுப்பாளர் கோப்பு உள்ளீட்டுச் சொருகுப்பொருள்
@@ -96,7 +96,7 @@
 Comment[sk]=Vloží ľubovoľný súbor na pozíciu kurzora
 Comment[sl]=Vstavi katerokoli berljivo datoteko na položaju kazalca
 Comment[sr]=Убаците било који читљиви фајл на положају курсора
-Comment[sr@Latn]=Убаците било који читљиви фајл на положају курсора
+Comment[sr@Latn]=Ubacite bilo koji čitljivi fajl na položaju kursora
 Comment[sv]=Infoga vilken läsbar fil som helst vid markörens plats
 Comment[ta]=சுட்டும் இடத்தில் படிக்கக்கூடிய கோப்பினை உள்ளிடு
 Comment[tg]=Ягон файли хонда мешударо ба ҳолати курсор низ гузоред
diff -urN -x CVS kdelibs.orig/kate/plugins/isearch/ktexteditor_isearch.desktop kdelibs/kate/plugins/isearch/ktexteditor_isearch.desktop
--- kdelibs.orig/kate/plugins/isearch/ktexteditor_isearch.desktop	Mon Sep 20 15:53:12 2004
+++ kdelibs/kate/plugins/isearch/ktexteditor_isearch.desktop	Mon Oct  4 15:24:51 2004
@@ -45,7 +45,7 @@
 Name[sl]=Vstavek KTextEditor za postopno iskanje
 Name[sq]=KTextEditor Shtojca: Kërkim përmes Numrimit
 Name[sr]=KTextEditor прикључак за инкрементално претраживање
-Name[sr@Latn]=KTextEditor dodatak za inkrementalno pretraživanje.
+Name[sr@Latn]=KTextEditor priključak za inkrementalno pretraživanje
 Name[ss]=I-plugin yekusesha lokungetekako kwe KTextEditor 
 Name[sv]=Ktexteditor-insticksprogram för inkrementell sökning
 Name[ta]=கேஉரை தொகுப்பாளர் படிப்படியான தேடற் சொருகுப்பொருள்
@@ -91,7 +91,7 @@
 Comment[sk]=Známe aj ako "Hľadanie podľa napísaného textu"
 Comment[sl]=Znano tudi kot »Iskanje med tipkanjem«
 Comment[sr]=Познато и као „претрага док куцате“
-Comment[sr@Latn]=Познато и као „претрага док куцате“
+Comment[sr@Latn]=Poznato i kao „pretraga dok kucate“
 Comment[sv]=Också känt som "Sök medan du skriver"
 Comment[ta]="தேடு என்று தட்டச்சிட்டவுடன்"  என்றும் அழைக்கப்படும்
 Comment[tg]=Ҳамчун "Вақти чоп кардан ба ҷустуҷ шавед" шинохта мешавад
diff -urN -x CVS kdelibs.orig/kate/plugins/kdatatool/ktexteditor_kdatatool.desktop kdelibs/kate/plugins/kdatatool/ktexteditor_kdatatool.desktop
--- kdelibs.orig/kate/plugins/kdatatool/ktexteditor_kdatatool.desktop	Sun Sep 12 15:55:49 2004
+++ kdelibs/kate/plugins/kdatatool/ktexteditor_kdatatool.desktop	Mon Oct  4 15:24:51 2004
@@ -47,7 +47,7 @@
 Name[sl]=Vstavek KTextEditor KDataTool
 Name[sq]=KTextEditor Shtojca: KDataTool
 Name[sr]=KTextEditor KDataTool прикључак
-Name[sr@Latn]=KTextEditor KDataTool dodatak
+Name[sr@Latn]=KTextEditor KDataTool priključak
 Name[ss]=I-plugin KDataTool ku KTextEditor
 Name[sv]=Ktexteditor-insticksprogram med dataverktyg
 Name[ta]=கேஉரைதொகுப்பாளர் கேதகவல் கருவி சொருகுப்பொருள்
diff -urN -x CVS kdelibs.orig/kate/plugins/wordcompletion/ktexteditor_docwordcompletion.desktop kdelibs/kate/plugins/wordcompletion/ktexteditor_docwordcompletion.desktop
--- kdelibs.orig/kate/plugins/wordcompletion/ktexteditor_docwordcompletion.desktop	Sun Sep 12 15:55:50 2004
+++ kdelibs/kate/plugins/wordcompletion/ktexteditor_docwordcompletion.desktop	Mon Oct  4 15:24:52 2004
@@ -30,7 +30,7 @@
 Name[ru]=Модуль автодополнения слов KTextEditor
 Name[sk]=Modul pre doplnenie slov pre KTextEditor
 Name[sr]=KTextEditor прикључак за довршаванје речи
-Name[sr@Latn]=KTextEditor прикључак за довршаванје речи
+Name[sr@Latn]=KTextEditor priključak za dovršavanje reči
 Name[sv]=Ktexteditor-insticksprogram för att komplettera ord
 Name[ta]=கேஉரைதொகுப்பாளர் கோப்பு உள்ளீட்டுச் சொருகுப்பொருள்
 Name[tg]=Мутассалкунандаи файли вуруди KTextEditor
@@ -64,7 +64,7 @@
 Comment[ro]=Propune completarea cuvintelor din document dintr-o listă popup sau direcţională
 Comment[sk]=Dopĺňanie slov v dokumente priame alebo pomocou dialógu
 Comment[sr]=Директна или искачућа допуна од речи у документу
-Comment[sr@Latn]=Директна или искачућа допуна од речи у документу
+Comment[sr@Latn]=Direktna ili iskačuća dopuna od reči u dokumentu
 Comment[sv]=Komplettering av ord i dokumentet baserad på ordlista eller dialogruta
 Comment[ta]=ஆவணத்திலுள்ள வார்த்தைகளிலிருந்து திசைகள் அல்லது மேல்விரி-சார்ந்த நிறைவு 
 Comment[tg]=Моқабл ё мобаъди ҷамъи аз калимаҳо дар санадот
diff -urN -x CVS kdelibs.orig/kcert/kcertpart.desktop kdelibs/kcert/kcertpart.desktop
--- kdelibs.orig/kcert/kcertpart.desktop	Tue Aug 31 09:21:11 2004
+++ kdelibs/kcert/kcertpart.desktop	Mon Oct  4 15:24:53 2004
@@ -51,7 +51,7 @@
 Comment[sl]=Vgradljivi osebni upravljalnik certifikatov
 Comment[sq]=Menagjer për diplomë personale
 Comment[sr]=Уградиви менаџер личних сертификата
-Comment[sr@Latn]=Ugradivi menadžer ličnih sertifikata.
+Comment[sr@Latn]=Ugradivi menadžer ličnih sertifikata
 Comment[ss]=Siphatsi sesithifikethi samuntfu sicu lesinamatselekako
 Comment[sv]=Inbäddningsbar personlig certifikatshanterare
 Comment[ta]=உட்பொதிந்த சொந்தச் சான்றிதழ் மேலாளர்
diff -urN -x CVS kdelibs.orig/kde.pot kdelibs/kde.pot
--- kdelibs.orig/kde.pot	Sun Oct  3 09:41:12 2004
+++ kdelibs/kde.pot	Tue Oct  5 09:42:20 2004
@@ -6,7 +6,7 @@
 msgid ""
 msgstr ""
 "Project-Id-Version: PACKAGE VERSION\n"
-"POT-Creation-Date: 2004-10-03 09:40+0200\n"
+"POT-Creation-Date: 2004-10-05 09:42+0200\n"
 "PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
 "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
 "Language-Team: LANGUAGE <LL@li.org>\n"
diff -urN -x CVS kdelibs.orig/kdecore/README.kiosk kdelibs/kdecore/README.kiosk
--- kdelibs.orig/kdecore/README.kiosk	Fri Jul  9 17:36:42 2004
+++ kdelibs/kdecore/README.kiosk	Wed Oct  6 13:55:04 2004
@@ -558,6 +558,26 @@
 data_konsole=false
 
 
+Control module restrictions
+===========================
+
+It is possible to restrict access to particular control modules.
+Although it is possible to remove control modules from the Control
+Center by editing the menu structure, such modules will then still
+be available to applications. A better way is to use the control
+module restrictions offered by KIOSK:
+
+[KDE Control Module Restrictions][$i]
+<menu-id>=false
+
+Some example menu-ids are:
+
+kde-display.desktop
+kde-proxy.desktop
+kde-screensaver.desktop
+
+See also kcmshell --list for a list of all the base names.
+
 Expansion of environment variables in KDE config files.
 =======================================================
 
diff -urN -x CVS kdelibs.orig/kdecore/eventsrc kdelibs/kdecore/eventsrc
--- kdelibs.orig/kdecore/eventsrc	Sat Oct  2 16:30:04 2004
+++ kdelibs/kdecore/eventsrc	Tue Oct  5 15:20:09 2004
@@ -171,7 +171,7 @@
 Comment[sk]=Bol dosiahnutý koniec zoznamu nájdených položiek
 Comment[sl]=To je konec seznama zadetkov
 Comment[sr]=Достигнут је крај листе подударања
-Comment[sr@Latn]=Достигнут је крај листе подударања
+Comment[sr@Latn]=Dostignut je kraj liste podudaranja
 Comment[sv]=Slutet på listan med träffar har nåtts
 Comment[ta]=பொருத்தப் பட்டியலின் முடிவை அடைந்தாயிற்று.
 Comment[tg]=Охири рӯйхати наздикоянда расидед
@@ -279,7 +279,7 @@
 Comment[sk]=Žiadna zhoda nebola nájdená
 Comment[sl]=Ni najti ujemajočih se zadetkov
 Comment[sr]=Подударна допуна није пронађена
-Comment[sr@Latn]=Подударна допуна није пронађена
+Comment[sr@Latn]=Podudarna dopuna nije pronađena
 Comment[sv]=Ingen matchande komplettering hittades
 Comment[ta]=பொருத்தமான நிறைவு ஏதுமில்லை.
 Comment[tg]=Анҷомдиҳии наздикоянда накофтан
@@ -387,7 +387,7 @@
 Comment[sk]=Existuje viac ako jedna zhoda
 Comment[sl]=Mogoč je več kakor en zadetek
 Comment[sr]=Постоји више од једног могућег подударања
-Comment[sr@Latn]=Постоји више од једног могућег подударања
+Comment[sr@Latn]=Postoji više od jednog mogućeg podudaranja
 Comment[sv]=Det finns mer än en möjlig träff
 Comment[ta]=ஒன்றுக்கு மேற்பட்ட பொருத்தங்கள் உள்ளன.
 Comment[tg]=Ин ҷо калонтар нисбат ба як дута эҳтимолӣ
@@ -525,7 +525,7 @@
 Comment[sl]=Izbrane datoteke ni bilo moč odpreti za branje ali pisanje.
 Comment[sq]=Skedari i zgjedhur nuk mund të hapet për lexim apo editim
 Comment[sr]=Изабрани фајл је немогуће отворити ради читања или писања
-Comment[sr@Latn]=Izabrani fajl je nemoguće otvoriti radi čitanja ili pisanja.
+Comment[sr@Latn]=Izabrani fajl je nemoguće otvoriti radi čitanja ili pisanja
 Comment[ss]=lifayela lelikhetsiwe akukhonakali kutsi livulwe kutsi lifundvwe kumbe libhalwe
 Comment[sv]=Den valda filen kan inte öppnas för läsning eller skrivning
 Comment[ta]=தேர்வு செய்த கோப்பை வாசிக்கவோ எழுதவோ திறக்க முடியாது
@@ -645,7 +645,7 @@
 Comment[sk]=Vyskytla sa vážna chyba, ktorá spôsobila ukončenie programu
 Comment[sl]=Program se je končal zaradi resne napake
 Comment[sr]=Десила се озбиљна грешка, која је изазвала да се програм затвори
-Comment[sr@Latn]=Десила се озбиљна грешка, која је изазвала да се програм затвори
+Comment[sr@Latn]=Desila se ozbiljna greška, koja je izazvala da se program zatvori
 Comment[sv]=Ett allvarligt fel uppstod vilket fick programmet att avslutas
 Comment[ta]=நிரலை வெளியேறச் செய்கிற அளவில் ஒரு பெரிய தவறுண்டு.
 Comment[tg]=Хатои ҷиддӣ иҷозат дода шудан ба нақшаи баромадан таъсир кардан
@@ -756,7 +756,7 @@
 Comment[sk]=V programe sa udialo niečo neočakávané
 Comment[sl]=V programu se je zgodilo nekaj izrednega
 Comment[sr]=Нешто посебно се десило у програму
-Comment[sr@Latn]=Нешто посебно се десило у програму
+Comment[sr@Latn]=Nešto posebno se desilo u programu
 Comment[sv]=Någonting speciellt inträffade i programmet
 Comment[ta]=நிரலில் ஏதோ விசேடம் நிகழ்ந்துள்ளது
 Comment[tg]=Чизе махсус ин нақша рӯй додан
@@ -870,7 +870,7 @@
 Comment[sk]=V programe sa vyskytla chyba, ktorá môže spôsobiť problémy
 Comment[sl]=V programu je nastala napaka, ki lahko povzroči težave
 Comment[sr]=Десила се грешка у програму која може изазвати проблеме
-Comment[sr@Latn]=Десила се грешка у програму која може изазвати проблеме
+Comment[sr@Latn]=Desila se greška u programu koja može izazvati probleme
 Comment[sv]=Det uppstod ett fel i programmet vilket kan orsaka problem
 Comment[ta]=பிரச்னைகளை ஏற்படுத்தும் அளவிற்கு நிரலில் ஒரு தவறு இருந்தது.
 Comment[tg]=Хатои иҷозат дода шудан дар нақша, кадом масъалаҳо таъсир кардан метавонед
@@ -996,7 +996,7 @@
 Comment[sl]=Nastala je tako resna napaka, da se bo končal program.
 Comment[sq]=Një gabim serioz ka ndodhur, së paku do të shkaktoj mbylljen e programit
 Comment[sr]=Десила се веома озбиљна грешка, која ће у најмању руку проузроковати затварање програма
-Comment[sr@Latn]=Desila se veoma ozbiljna greška, koja će u najmanju ruku prouzrokovati zatvaranje programa.
+Comment[sr@Latn]=Desila se veoma ozbiljna greška, koja će u najmanju ruku prouzrokovati zatvaranje programa
 Comment[sv]=Ett väldigt allvarligt fel uppstod som åtminstone fick programmet att avslutas
 Comment[ta]=நிரலை வெளியேறச் செய்யும் அளவில் ஒரு பெரிய தவறு ஏற்பட்டது
 Comment[tg]=Хатои хеле ҷиддӣ воқеъ шудан, дар охир нақшаи баромадан таъсир кардан
@@ -1104,7 +1104,7 @@
 Comment[sk]=KDE sa spúšťa
 Comment[sl]=KDE se zaganja
 Comment[sr]=KDE се покреће
-Comment[sr@Latn]=KDE се покреће
+Comment[sr@Latn]=KDE se pokreće
 Comment[sv]=KDE startas
 Comment[ta]=கேடிஇ ஆரம்பிக்கிறது
 Comment[tg]=KDE боркарда шуда истодааст
@@ -1212,7 +1212,7 @@
 Comment[sk]=KDE sa ukončuje
 Comment[sl]=KDE se končuje
 Comment[sr]=KDE се завршава
-Comment[sr@Latn]=KDE се завршава
+Comment[sr@Latn]=KDE se završava
 Comment[sv]=KDE avslutas
 Comment[ta]=கேடிஇ வெளிச்செல்கிறது
 Comment[tg]=KDE бароварда шуда истодааст
@@ -1718,6 +1718,7 @@
 Name[be]=Пытаньне
 Name[bg]=Въпрос
 Name[bn]=প্রশ্ন
+Name[br]=Goulenn
 Name[bs]=Pitanje
 Name[ca]=Pregunta
 Name[cs]=Dotaz
diff -urN -x CVS kdelibs.orig/kdecore/kdesktopfile.cpp kdelibs/kdecore/kdesktopfile.cpp
--- kdelibs.orig/kdecore/kdesktopfile.cpp	Wed Jul 21 15:32:05 2004
+++ kdelibs/kdecore/kdesktopfile.cpp	Mon Oct  4 12:14:06 2004
@@ -19,7 +19,7 @@
   Boston, MA 02111-1307, USA.
 */
 
-// $Id$
+// $Id$
 
 #include <stdlib.h>
 #include <unistd.h>
@@ -27,6 +27,7 @@
 #include <qfile.h>
 #include <qtextstream.h>
 
+#include <kdebug.h>
 #include "kurl.h"
 #include "kconfigbackend.h"
 #include "kapplication.h"
@@ -138,6 +139,8 @@
      return true;
   if (dirs->relativeLocation("data", path).startsWith("kdesktop/Desktop"))
      return true;
+     
+  kdWarning() << "Access to '" << path << "' denied because of 'run_desktop_files' restriction." << endl;
   return false;
 }
 
diff -urN -x CVS kdelibs.orig/kdecore/kicontheme.cpp kdelibs/kdecore/kicontheme.cpp
--- kdelibs.orig/kdecore/kicontheme.cpp	Tue Sep 28 05:36:02 2004
+++ kdelibs/kdecore/kicontheme.cpp	Sun Oct  3 16:28:58 2004
@@ -1,6 +1,6 @@
 /* vi: ts=8 sts=4 sw=4
  *
- * $Id$
+ * $Id$
  *
  * This file is part of the KDE project, module kdecore.
  * Copyright (C) 2000 Geert Jansen <jansen@kde.org>
diff -urN -x CVS kdelibs.orig/kdecore/kprocess.cpp kdelibs/kdecore/kprocess.cpp
--- kdelibs.orig/kdecore/kprocess.cpp	Mon Jan 19 13:00:07 2004
+++ kdelibs/kdecore/kprocess.cpp	Wed Oct 13 22:07:23 2004
@@ -1,6 +1,6 @@
 /*
 
-   $Id$
+   $Id$
 
    This file is part of the KDE libraries
    Copyright (C) 1997 Christian Czezatke (e9025461@student.tuwien.ac.at)
@@ -739,7 +739,25 @@
 void KProcess::setUseShell(bool useShell, const char *shell)
 {
   d->useShell = useShell;
-  d->shell = (shell && *shell) ? shell : "/bin/sh";
+  if (shell && *shell)
+    d->shell = shell;
+  else
+// #ifdef NON_FREE // ... as they ship non-POSIX /bin/sh
+#if !defined(__linux__) && !defined(__FreeBSD__) && !defined(__NetBSD__) && !defined(__OpenBSD__) && !defined(__GNU__)
+  // Solaris POSIX ...
+  if (!access( "/usr/xpg4/bin/sh", X_OK ))
+    d->shell = "/usr/xpg4/bin/sh";
+  else
+  // ... which links here anyway
+  if (!access( "/bin/ksh", X_OK ))
+    d->shell = "/bin/ksh";
+  else
+  // dunno, maybe superfluous?
+  if (!access( "/usr/ucb/sh", X_OK ))
+    d->shell = "/usr/ucb/sh";
+  else
+#endif
+    d->shell = "/bin/sh";
 }
 
 void KProcess::setUsePty(Communication usePty, bool addUtmp)
diff -urN -x CVS kdelibs.orig/kdecore/kstartupinfo.cpp kdelibs/kdecore/kstartupinfo.cpp
--- kdelibs.orig/kdecore/kstartupinfo.cpp	Thu Jul 15 16:07:27 2004
+++ kdelibs/kdecore/kstartupinfo.cpp	Tue Oct 12 16:58:19 2004
@@ -1,6 +1,6 @@
 /****************************************************************************
 
- $Id$
+ $Id$
 
  Copyright (C) 2001-2003 Lubos Lunak        <l.lunak@kde.org>
 
@@ -595,28 +595,6 @@
         return NoMatch;
     if( d->startups.count() == 0 )
         return NoMatch; // no startups
-#if defined Q_WS_X11 && ! defined K_WS_QTONLY
-// SELI ???
-    NETWinInfo info( qt_xdisplay(),  w_P, qt_xrootwin(),
-        NET::WMWindowType | NET::WMPid | NET::WMState );
-    // ignore NET::Tool and other special window types
-    NET::WindowType type = info.windowType( NET::NormalMask | NET::DesktopMask
-        | NET::DockMask | NET::ToolbarMask | NET::MenuMask | NET::DialogMask
-        | NET::OverrideMask | NET::TopMenuMask | NET::UtilityMask | NET::SplashMask );
-    if( type != NET::Normal
-        && type != NET::Override
-        && type != NET::Unknown
-        && type != NET::Dialog
-        && type != NET::Utility )
-//        && type != NET::Dock ) why did I put this here?
-	return NoMatch;
-    // lets see if this is a transient
-    Window transient_for;
-    if( XGetTransientForHint( qt_xdisplay(), static_cast< Window >( w_P ), &transient_for )
-        && static_cast< WId >( transient_for ) != qt_xrootwin()
-        && transient_for != None )
-	return NoMatch;
-#endif
     // Strategy:
     //
     // Is this a compliant app ?
@@ -636,6 +614,8 @@
         return find_id( id, id_O, data_O ) ? Match : NoMatch;
         }
 #if defined Q_WS_X11 && ! defined K_WS_QTONLY
+    NETWinInfo info( qt_xdisplay(),  w_P, qt_xrootwin(),
+        NET::WMWindowType | NET::WMPid | NET::WMState );
     pid_t pid = info.pid();
     if( pid > 0 )
         {
@@ -655,6 +635,23 @@
         if( find_wclass( res_name, res_class, id_O, data_O ))
             return Match;
         }
+    // ignore NET::Tool and other special window types, if they can't be matched
+    NET::WindowType type = info.windowType( NET::NormalMask | NET::DesktopMask
+        | NET::DockMask | NET::ToolbarMask | NET::MenuMask | NET::DialogMask
+        | NET::OverrideMask | NET::TopMenuMask | NET::UtilityMask | NET::SplashMask );
+    if( type != NET::Normal
+        && type != NET::Override
+        && type != NET::Unknown
+        && type != NET::Dialog
+        && type != NET::Utility )
+//        && type != NET::Dock ) why did I put this here?
+	return NoMatch;
+    // lets see if this is a transient
+    Window transient_for;
+    if( XGetTransientForHint( qt_xdisplay(), static_cast< Window >( w_P ), &transient_for )
+        && static_cast< WId >( transient_for ) != qt_xrootwin()
+        && transient_for != None )
+	return NoMatch;
 #endif
     kdDebug( 172 ) << "check_startup:cantdetect" << endl;
     return CantDetect;
diff -urN -x CVS kdelibs.orig/kdecore/malloc/README kdelibs/kdecore/malloc/README
--- kdelibs.orig/kdecore/malloc/README	Wed Feb 26 18:10:54 2003
+++ kdelibs/kdecore/malloc/README	Mon Oct 11 14:50:50 2004
@@ -4,11 +4,11 @@
 Basically, it's here because for now it has better performance than both glibc-2.2.x and
 FreeBSD's libc.
 
- There's a new configure switch, --enable-fast-malloc. By default it's turned on, disabling
+ There's a new configure switch, --enable-fast-malloc. By default it's turned off, disabling
 the system libc one and using this one. Using --enable-fast-malloc=full enables this
 malloc unconditionally, aiming for the maximum performance. By using only --enable-fast-malloc,
-it's possible to select both malloc implementations at runtime. When $KDE_MALLOC is set to 1,
-this malloc is used, the system libc one is used otherwise.
+it's possible to select both malloc implementations at runtime. When $KDE_MALLOC is set to 0,
+the system libc malloc is used, otherwise this malloc is used.
 
  For now, the requirements are :
  - x86 CPU (because of the spinlock implementation in assembler), it should be easy to add
diff -urN -x CVS kdelibs.orig/kdecore/netwm.cpp kdelibs/kdecore/netwm.cpp
--- kdelibs.orig/kdecore/netwm.cpp	Wed Jul 28 17:49:07 2004
+++ kdelibs/kdecore/netwm.cpp	Fri Oct  8 15:25:55 2004
@@ -494,7 +494,7 @@
 	sz = p->icons[j].size.width * p->icons[j].size.height;
 	s = sz * sizeof(long);
 
-	if ( i + s - 1 > bufsize ) {
+	if ( i + s - 1 > bufsize || sz == 0 || sz > 1024 * 1024 ) {
 	    break;
 	}
 
diff -urN -x CVS kdelibs.orig/kdecore/network/kresolvermanager.cpp kdelibs/kdecore/network/kresolvermanager.cpp
--- kdelibs.orig/kdecore/network/kresolvermanager.cpp	Mon Aug 30 02:36:29 2004
+++ kdelibs/kdecore/network/kresolvermanager.cpp	Sat Oct  9 02:10:16 2004
@@ -203,7 +203,7 @@
 // a thread will try maxThreadRetries to get data, waiting at most
 // maxThreadWaitTime milliseconds between each attempt. After that, it'll
 // exit
-static const int maxThreadWaitTime = 20000; // 20 seconds
+static const int maxThreadWaitTime = ULONG_MAX; // wait forever
 static const int maxThreads = 5;
 
 static pid_t pid;		// FIXME -- disable when everything is ok
diff -urN -x CVS kdelibs.orig/kded/kdedmodule.desktop kdelibs/kded/kdedmodule.desktop
--- kdelibs.orig/kded/kdedmodule.desktop	Tue Aug 24 08:29:07 2004
+++ kdelibs/kded/kdedmodule.desktop	Tue Oct  5 15:20:12 2004
@@ -9,6 +9,7 @@
 Comment[be]=Модуль KDED
 Comment[bg]=Модул KDED
 Comment[bn]=KDED মডিইল
+Comment[br]=Mollad KDED
 Comment[bs]=KDED modul
 Comment[ca]=Mòdul per a KDED
 Comment[cs]=KDED modul
diff -urN -x CVS kdelibs.orig/kdeprint/filters/poster.desktop kdelibs/kdeprint/filters/poster.desktop
--- kdelibs.orig/kdeprint/filters/poster.desktop	Tue Aug 31 09:21:17 2004
+++ kdelibs/kdeprint/filters/poster.desktop	Mon Oct  4 15:24:59 2004
@@ -107,7 +107,7 @@
 Description[sl]=Orodje za tiskanje velikih posterjev na majhnih listih papirja. Za uporabo tega ukaza mora biti izvedljiva datoteka <b>poster</b> v vaši poti <tt>PATH</tt>. Izvirna koda za to orodje je na voljo na <a href="http://printing.kde.org/downloads">spletni strani KDEPrint</a>. <p><b><u>OPOZORILO:</u></b> Paket, ki je na strani KDEPrint je spremenjena različica originalne, ki je na voljo na vsakem zrcalnem strežniku arhivov CTAN, vendar pa ta <b>ne</b> dela s KDE. Uporabiti <b>morate</b> paket s <a href="http://printing.kde.org/downloads">spletne strani KDEPrint</a>.
 Description[sq]=Vegël për shtypjen edhe ndarjen e pllakatave në letra të vogla. Për ta përdorur këtë komandë, <b>pllakata</b> ekzekutuese duhet të jetë  e arrijshme në <tt>PATH</tt> e juaj. Kodi burimor për këtë vegël mund të gjindet në <a href="http://printing.kde.org/downloads">Web Faqën KDEPrint</a>. <p><b><u>VËREJTJE:</u></b> Pako e gjetur në web faqën e KDEPrint është verzion i ndryshuar nga origjinali, ky verzion mund të gjindet në çdo web faqe arkivore të CTAN pasqyrave, por verzioni origjinal <b>nuk punon</b> me sistemin KDE. Ju <b>duheni</b> ta përdorni pakon që gjendet në këtë web faqe: <a href="http://printing.kde.org/downloads">Web Faqja KDEPrint</a>.
 Description[sr]=Алатка за штампање великих постера на више малих папира. Да би сте користили ову наредбу, извршни фајл <b>poster</b> мора бити приступачан кроз ваш <tt>PATH(путању)</tt>. Изворни код овог алата можете наћи на <a href="http=://printing.kde.org/downloads">KDEPrint веб сајту</a>. <p><b><u>Упозорење:</u></b> Пакет који се налази на KDE Print веб сајту је измењена верзија оригинала који се може наћи на било ком CTAN огледалу, али оригинални пакет <b>неће</b> радити са KDE-ом. <b>Морате</b> користити пакет са <a href="http://printing.kde.org/downloads">KDEPrint веб сајта</a>.
-Description[sr@Latn]=Alatka za štampanje velikih postera na više malih papira. Da bi ste koristili ovu naredbu, <b>poster</b> izvršni fajl mora biti pristupačan kroz vaš <tt>PATH(putanju)</tt>. Izvorni kod ovog alata možete naći na <a href=" http=://printing.kde.org/downloads">KDEPrint veb sajtu</a>. <p><b><u>Upozorenje:</u></b> Paket koji se nalazi na KDE Print veb sajtu je izmenjena verzija originala koji se može naći na bilo kom CTAN ogledalu, ali originalni paket <b>neće</b> raditi sa KDE-om. <b>Morate</b> koristiti paket sa <a href="http://printing.kde.org/downloads">KDEPrint veb sajta</a>.
+Description[sr@Latn]=Alatka za štampanje velikih postera na više malih papira. Da bi ste koristili ovu naredbu, izvršni fajl <b>poster</b> mora biti pristupačan kroz vaš <tt>PATH(putanju)</tt>. Izvorni kod ovog alata možete naći na <a href="http=://printing.kde.org/downloads">KDEPrint veb sajtu</a>. <p><b><u>Upozorenje:</u></b> Paket koji se nalazi na KDE Print veb sajtu je izmenjena verzija originala koji se može naći na bilo kom CTAN ogledalu, ali originalni paket <b>neće</b> raditi sa KDE-om. <b>Morate</b> koristiti paket sa <a href="http://printing.kde.org/downloads">KDEPrint veb sajta</a>.
 Description[sv]=Verktyg för att skriva ut stora affischer på flera små blad. För att använda kommandot måste det körbara programmet <b>poster</b> vara tillgängligt i <tt>PATH</tt>. Källkod för detta verktyg kan finnas på <a href="http://printing.kde.org/downloads">KDEPrints webbplats</a>. <p><b><u>Varning:</u></b> Paketet som finns på KDEPrints webbplats är en ändrad version av originalet som finns på alla CTAN-arkivplatser. Men originalpaketet fungerar <b>inte</b> med KDE, man <b>måste</b> använda paketet som finns på <a href="http://printing.kde.org/downloads">KDEPrints webbplats</a>.
 Description[ta]=பலவித சிறிய தாள்களில் பெரிய போஸ்டர்களை அச்சிப்பதற்கான பயன்பாடு. இந்த கட்டளையை பயன்படுத்த <b>போஸ்டர் </b>நிறுவல் அணுகக்கூடியதாக <tt>உங்கள் பாதை இருக்க வேண்டும்</tt>. இந்த பயன்பாட்டுக்கான மூல குறியீடு கேடிஇஅச்சு வலைதளத்தில் <a href="http://printing.kde.org/downloads">இருக்கும்</a>. <p><b><u>எச்சரிக்கை:</u></b>கேடிஇ அச்சு வலைத்தளத்தில் காணப்படும் தொகுப்பு மாற்றப்பட்டுள்ளது. இது எல்லா CTAN காப்பத்திலும் காணப்படும்.ஆனால் அசல் தொகுப்பு கேடிஇயுடன் இயங்காது.நீங்கள் <b></b> <a href="http://printing.kde.org/downloads">கேடிஇஅச்சு வலைதளத்தில்</a> உள்ள தொகுப்பைத்தான் பயன்படுத்த வேண்டும்</a>.
 Description[tg]=Барномаҳои барои чопи сатроҳои бузург барои сафҳоти коғаз купак барои истифода аз ин дастури<b>poster</b> бояд дар<tt>PATH-и</tt> шумо кобали дастурӣ бошад . Коди марҷа ин обзор метавонад дар<a href="http://printing.kde.org/downloads"> KDEPrint сйти Вэб ёфта шавад.</a>. <p><b><u>Огоҳӣ:</u></b> Пакете, ки нусхаи таъғиршуда буд, дар Вэби KDEPrint ёфта шуд , ва он дар ягон CTAN archive mirror мумкин аст ки ёбад ,лекин нусхаи аслиаш <b>дар KDE</b> кор намекунад. Шумо<b>бояд</b> пакети ёфташударо истифода кунед дар<a href="http://printing.kde.org/downloads">KDEPrint web site</a>.
diff -urN -x CVS kdelibs.orig/kdeprint/kdeprintd.desktop kdelibs/kdeprint/kdeprintd.desktop
--- kdelibs.orig/kdeprint/kdeprintd.desktop	Mon Sep 20 15:53:16 2004
+++ kdelibs/kdeprint/kdeprintd.desktop	Mon Oct  4 15:24:58 2004
@@ -103,7 +103,7 @@
 Comment[sk]=Démon tlače pre KDE
 Comment[sl]=Tiskalniški strežnik za KDE
 Comment[sr]=KDE-ов демон за штампање
-Comment[sr@Latn]=KDE-ов демон за штампање
+Comment[sr@Latn]=KDE-ov demon za štampanje
 Comment[sv]=Skrivardemon för KDE
 Comment[ta]=கேடிஇக்கான ஒரு அச்ச் டெமான்
 Comment[tg]=Демони Чопкардан барои KDE
diff -urN -x CVS kdelibs.orig/kdeprint/specials.desktop kdelibs/kdeprint/specials.desktop
--- kdelibs.orig/kdeprint/specials.desktop	Sun Sep 12 15:55:55 2004
+++ kdelibs/kdeprint/specials.desktop	Mon Oct  4 15:24:58 2004
@@ -806,7 +806,7 @@
 Name[sk]=Poslať súbor PDF
 Name[sl]=Pošlji datoteko PDF po pošti
 Name[sr]=Слање PDF фајла поштом
-Name[sr@Latn]=Слање PDF фајла поштом
+Name[sr@Latn]=Slanje PDF fajla poštom
 Name[sv]=Skicka PDF-fil
 Name[ta]=PDF கோப்பை அஞ்சல் அனுப்பு
 Name[tg]=Файли почтавии PDF
@@ -881,7 +881,7 @@
 Description[sl]=Uporabite kdsendfax za faksiranje trenutnega dokumenta
 Description[sq]=Përdor ksendfax për ta dërguar dokumentin aktual me faks
 Description[sr]=Користите ksendfax да пошаљте текући документ на факс
-Description[sr@Latn]=Koristi ksendfax da pošalje tekući dokument na faks.
+Description[sr@Latn]=Koristite ksendfax da pošaljte tekući dokument na faks
 Description[ss]=Sebentisa i-ksendfax kutfumela ngefax kwelidokhumente lanyalo
 Description[sv]=Använd Ksendfax för att faxa aktuellt dokument
 Description[ta]=கேஃபாக்ஸ்அனுப்பு மூலம் நடப்பு ஆவணத்தை தொலைநகலிடு
diff -urN -x CVS kdelibs.orig/kdeui/kdetrayproxy/kdetrayproxy.desktop kdelibs/kdeui/kdetrayproxy/kdetrayproxy.desktop
--- kdelibs.orig/kdeui/kdetrayproxy/kdetrayproxy.desktop	Mon Sep 20 15:53:18 2004
+++ kdelibs/kdeui/kdetrayproxy/kdetrayproxy.desktop	Mon Oct  4 15:25:01 2004
@@ -26,7 +26,7 @@
 Name[ro]=Modul proxy pentru tava de sistem KDE
 Name[sk]=Modul pre proxy lišty KDE
 Name[sr]=Прокси модул KDE-ове касете
-Name[sr@Latn]=Прокси модул KDE-ове касете
+Name[sr@Latn]=Proksi modul KDE-ove kasete
 Name[sv]=KDE-proxymodul för systembricka
 Name[ta]=KDE தட்டு மாற்று பதிலாள் கூறு
 Name[tg]=Новаи KDE-и модули прокси
@@ -59,7 +59,7 @@
 Comment[se]=Moduvla mii dagaha ahte KDE:a vuogádatgárcoprográmmat doibmet KWin haga.
 Comment[sk]=Proxy pre aplikácie z lišty KDE, aby nevyžadovali KWin
 Comment[sr]=Прокси који омогућава KDE програмима за системску касету да раде без KWin-а
-Comment[sr@Latn]=Прокси који омогућава KDE програмима за системску касету да раде без KWin-а
+Comment[sr@Latn]=Proksi koji omogućava KDE programima za sistemsku kasetu da rade bez KWin-a
 Comment[sv]=Proxymodul som gör det möjligt för KDE:s program i systembrickan att fungera utan Kwin
 Comment[ta]=மாற்று பதிலாள் KDE அமைப்பு தட்டு பயன்பாட்டை KWin இல்லாமல் வேலை செய்ய செயல்படுத்துகிறது
 Comment[tg]=Proxy барномаҳои KDE systray-ро барои корнамоӣ, бидуни KWin фаъол месозад
diff -urN -x CVS kdelibs.orig/kdeui/kedittoolbar.cpp kdelibs/kdeui/kedittoolbar.cpp
--- kdelibs.orig/kdeui/kedittoolbar.cpp	Sun Jul 11 20:04:59 2004
+++ kdelibs/kdeui/kedittoolbar.cpp	Fri Oct  8 11:41:37 2004
@@ -47,8 +47,8 @@
 #include "kpushbutton.h"
 #include <kprocio.h>
 
-#define LINESEPARATORSTRING i18n("--- line separator ---")
-#define SEPARATORSTRING i18n("--- separator ---")
+#define LINESEPARATORSTRING I18N_NOOP("--- line separator ---")
+#define SEPARATORSTRING I18N_NOOP("--- separator ---")
 
 static void dump_xml(const QDomDocument& doc)
 {
@@ -105,9 +105,9 @@
   virtual QString key(int column, bool) const
   {
     QString s = text( column );
-    if ( s == LINESEPARATORSTRING )
+    if ( s == i18n(LINESEPARATORSTRING) )
       return "0";
-    if ( s == SEPARATORSTRING )
+    if ( s == i18n(SEPARATORSTRING) )
       return "1";
     return "2" + s;
   }
@@ -883,9 +883,9 @@
       ToolbarItem *act = new ToolbarItem(m_activeList, tagSeparator, sep_name.arg(sep_num++), QString::null);
       bool isLineSep = ( it.attribute(attrLineSeparator, "true").lower() == QString::fromLatin1("true") );
       if(isLineSep)
-        act->setText(1, LINESEPARATORSTRING);
+        act->setText(1, i18n(LINESEPARATORSTRING));
       else
-        act->setText(1, SEPARATORSTRING);
+        act->setText(1, i18n(SEPARATORSTRING));
       it.setAttribute( attrName, act->internalName() );
       continue;
     }
@@ -954,9 +954,9 @@
 
   // finally, add default separators to the inactive list
   ToolbarItem *act = new ToolbarItem(m_inactiveList, tagSeparator, sep_name.arg(sep_num++), QString::null);
-  act->setText(1, LINESEPARATORSTRING);
+  act->setText(1, i18n(LINESEPARATORSTRING));
   act = new ToolbarItem(m_inactiveList, tagSeparator, sep_name.arg(sep_num++), QString::null);
-  act->setText(1, SEPARATORSTRING);
+  act->setText(1, i18n(SEPARATORSTRING));
 }
 
 KActionCollection *KEditToolbarWidget::actionCollection() const
@@ -1103,9 +1103,9 @@
 
   QDomElement new_item;
   // let's handle the separator specially
-  if (item->text(1) == LINESEPARATORSTRING) {
+  if (item->text(1) == i18n(LINESEPARATORSTRING)) {
     new_item = domDocument().createElement(tagSeparator);
-  } else if (item->text(1) == SEPARATORSTRING) {
+  } else if (item->text(1) == i18n(SEPARATORSTRING)) {
     new_item = domDocument().createElement(tagSeparator);
     new_item.setAttribute(attrLineSeparator, "false");
   } else
diff -urN -x CVS kdelibs.orig/kdeui/klineedit.cpp kdelibs/kdeui/klineedit.cpp
--- kdelibs.orig/kdeui/klineedit.cpp	Sun Sep 19 16:17:43 2004
+++ kdelibs/kdeui/klineedit.cpp	Wed Oct  6 22:42:17 2004
@@ -848,17 +848,13 @@
 
     QPopupMenu *popup = QLineEdit::createPopupMenu();
 
-    if ( isReadOnly() )
-      popup->changeItem( popup->idAt(0), SmallIconSet("editcopy"), popup->text( popup->idAt(0) ) );
-    else {
-      int id = popup->idAt(0);
-      popup->changeItem( id - IdUndo, SmallIconSet("undo"), popup->text( id - IdUndo) );
-      popup->changeItem( id - IdRedo, SmallIconSet("redo"), popup->text( id - IdRedo) );
-      popup->changeItem( id - IdCut, SmallIconSet("editcut"), popup->text( id - IdCut) );
-      popup->changeItem( id - IdCopy, SmallIconSet("editcopy"), popup->text( id - IdCopy) );
-      popup->changeItem( id - IdPaste, SmallIconSet("editpaste"), popup->text( id - IdPaste) );
-      popup->changeItem( id - IdClear, SmallIconSet("editclear"), popup->text( id - IdClear) );
-    }
+    int id = popup->idAt(0);
+    popup->changeItem( id - IdUndo, SmallIconSet("undo"), popup->text( id - IdUndo) );
+    popup->changeItem( id - IdRedo, SmallIconSet("redo"), popup->text( id - IdRedo) );
+    popup->changeItem( id - IdCut, SmallIconSet("editcut"), popup->text( id - IdCut) );
+    popup->changeItem( id - IdCopy, SmallIconSet("editcopy"), popup->text( id - IdCopy) );
+    popup->changeItem( id - IdPaste, SmallIconSet("editpaste"), popup->text( id - IdPaste) );
+    popup->changeItem( id - IdClear, SmallIconSet("editclear"), popup->text( id - IdClear) );
       
     // If a completion object is present and the input
     // widget is not read-only, show the Text Completion
diff -urN -x CVS kdelibs.orig/kdeui/kmainwindow.cpp kdelibs/kdeui/kmainwindow.cpp
--- kdelibs.orig/kdeui/kmainwindow.cpp	Mon Aug 30 02:04:47 2004
+++ kdelibs/kdeui/kmainwindow.cpp	Wed Oct  6 15:34:08 2004
@@ -1192,6 +1192,19 @@
     return size;
 }
 
+#if KDE_IS_VERSION( 3, 9, 0 )
+#ifdef __GNUC__
+#warning Remove, should be in Qt
+#endif
+#endif
+void KMainWindow::setIcon( const QPixmap& p )
+{
+    QMainWindow::setIcon( p );
+    // Qt3 doesn't support _NET_WM_ICON, but KApplication::setTopWidget(), which
+    // is used by KMainWindow, sets it
+    KWin::setIcons( winId(), p, QPixmap());
+}
+
 // why do we support old gcc versions? using KXMLGUIBuilder::finalizeGUI;
 // DF: because they compile KDE much faster :)
 void KMainWindow::finalizeGUI( KXMLGUIClient *client )
diff -urN -x CVS kdelibs.orig/kdeui/kmainwindow.h kdelibs/kdeui/kmainwindow.h
--- kdelibs.orig/kdeui/kmainwindow.h	Sat Oct  2 13:38:19 2004
+++ kdelibs/kdeui/kmainwindow.h	Wed Oct  6 14:50:58 2004
@@ -637,6 +637,12 @@
     // KDE4 to be removed
     QSize sizeForCentralWidgetSize(QSize size) KDE_DEPRECATED;
 
+    /**
+     * @internal
+     */
+    // KDE4 remove
+    virtual void setIcon( const QPixmap & );
+
 public slots:
     /**
      * Show a standard configure toolbar dialog.
diff -urN -x CVS kdelibs.orig/kdeui/ktabbar.cpp kdelibs/kdeui/ktabbar.cpp
--- kdelibs.orig/kdeui/ktabbar.cpp	Sun Sep 12 21:59:21 2004
+++ kdelibs/kdeui/ktabbar.cpp	Tue Oct  5 20:29:33 2004
@@ -19,6 +19,7 @@
 */
 
 #include <qapplication.h>
+#include <qcursor.h>
 #include <qpainter.h>
 #include <qstyle.h>
 #include <qtimer.h>
@@ -220,7 +221,9 @@
 
 void KTabBar::activateDragSwitchTab()
 {
-    setCurrentTab( mDragSwitchTab );
+    QTab *tab = selectTab( mapFromGlobal( QCursor::pos() ) );
+    if ( tab!= 0L && mDragSwitchTab == tab )
+      setCurrentTab( mDragSwitchTab );
     mDragSwitchTab = 0;
 }
 
diff -urN -x CVS kdelibs.orig/khtml/ChangeLog kdelibs/khtml/ChangeLog
--- kdelibs.orig/khtml/ChangeLog	Thu Oct  7 14:35:07 2004
+++ kdelibs/khtml/ChangeLog	Mon Oct 11 20:05:03 2004
@@ -1,3 +1,13 @@
+2004-10-11  David Faure  <faure@kde.org>
+
+        * xml/dom_nodeimpl.cpp (dispatchGenericEvent): Fixed stopPropagation
+        when called from the target node itself (#90750).
+
+2004-10-08  David Faure  <faure@kde.org>
+
+        * ecma/kjs_html.cpp (KJS::HTMLCollection::tryGet): 
+        return Undefined instead of Null for out-of-range indices.
+
 2004-10-07  David Faure  <faure@kde.org>
 
         * xml/dom_nodeimpl.cpp (dispatchWindowEvent): Emit load event for frames
diff -urN -x CVS kdelibs.orig/khtml/ecma/kjs_html.cpp kdelibs/khtml/ecma/kjs_html.cpp
--- kdelibs.orig/khtml/ecma/kjs_html.cpp	Thu Oct  7 14:34:20 2004
+++ kdelibs/khtml/ecma/kjs_html.cpp	Mon Oct 11 12:40:23 2004
@@ -3139,8 +3139,11 @@
   bool ok;
   unsigned int u = propertyName.toULong(&ok);
   if (ok) {
-    DOM::Node node = collection.item(u);
-    return getDOMNode(exec,node);
+    if ( u < collection.length() ) {
+      DOM::Node node = collection.item(u);
+      return getDOMNode(exec,node);
+    } else
+      return Undefined();
   }
   else
     return getNamedItems(exec,propertyName);
@@ -3480,15 +3483,16 @@
 void Image::putValueProperty(ExecState *exec, int token, const Value& value, int)
 {
   switch (token) {
-  case Src: {
-    String str = value.toString(exec);
-    src = str.value();
+  case Src:
+    src = value.toString(exec);
     if ( img ) img->deref(this);
     img = static_cast<DOM::DocumentImpl*>( doc.handle() )->docLoader()->requestImage( src.string() );
 // ### img = doc ? doc->docLoader()->requestImage( src.string() ) : 0;
-    if ( img ) img->ref(this);
+    if ( img ) {
+      img->ref(this);
+      src = img->url();
+    }
     break;
-  }
   case OnLoad:
     if ( m_onLoadListener )
         m_onLoadListener->deref();
diff -urN -x CVS kdelibs.orig/khtml/rendering/render_form.cpp kdelibs/khtml/rendering/render_form.cpp
--- kdelibs.orig/khtml/rendering/render_form.cpp	Sun Oct  3 01:40:29 2004
+++ kdelibs/khtml/rendering/render_form.cpp	Fri Oct  8 19:16:26 2004
@@ -20,7 +20,7 @@
  * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  * Boston, MA 02111-1307, USA.
  *
- * $Id$
+ * $Id$
  */
 
 #include <kcompletionbox.h>
@@ -934,15 +934,13 @@
             else if (listItems[listIndex]->id() == ID_OPTION) {
                 HTMLOptionElementImpl* optElem = static_cast<HTMLOptionElementImpl*>(listItems[listIndex]);
                 QString text = optElem->text().string();
-
-                // Prefer label if set
-                DOMString label = optElem->getAttribute(ATTR_LABEL);
-                if (!label.isEmpty())
-                    text = label.string();
-
-                if (optElem->parentNode()->id() == ID_OPTGROUP)
+                if (optElem->parentNode()->id() == ID_OPTGROUP) {
+                    // Prefer label if set
+                    DOMString label = optElem->getAttribute(ATTR_LABEL);
+                    if (!label.isEmpty())
+                        text = label.string();
                     text = QString::fromLatin1("    ")+text;
-
+                }
                 if(m_useListBox) {
                     KListBox *l = static_cast<KListBox*>(m_widget);
                     l->insertItem(text, listIndex);
diff -urN -x CVS kdelibs.orig/khtml/xml/dom_nodeimpl.cpp kdelibs/khtml/xml/dom_nodeimpl.cpp
--- kdelibs.orig/khtml/xml/dom_nodeimpl.cpp	Thu Oct  7 14:35:07 2004
+++ kdelibs/khtml/xml/dom_nodeimpl.cpp	Mon Oct 11 20:05:04 2004
@@ -428,19 +428,19 @@
 
     // dispatch to the actual target node
     it.toLast();
+    NodeImpl* propagationSentinel = 0;
     if (!evt->propagationStopped()) {
         evt->setEventPhase(Event::AT_TARGET);
         evt->setCurrentTarget(it.current());
         it.current()->handleLocalEvents(evt, true);
         if (!evt->propagationStopped())
             it.current()->handleLocalEvents(evt,false);
+        else
+            propagationSentinel = it.current();
     }
     --it;
 
     if (evt->bubbles()) {
-        evt->stopPropagation(false);
-        NodeImpl* propagationSentinel = 0;
-
         evt->setEventPhase(Event::BUBBLING_PHASE);
         for (; it.current() && !evt->propagationStopped(); --it) {
             if (evt->propagationStopped()) propagationSentinel = it.current();
diff -urN -x CVS kdelibs.orig/kimgio/dds.kimgio kdelibs/kimgio/dds.kimgio
--- kdelibs.orig/kimgio/dds.kimgio	Sun Sep 12 15:56:01 2004
+++ kdelibs/kimgio/dds.kimgio	Mon Oct  4 15:25:05 2004
@@ -28,7 +28,7 @@
 Name[sk]=Povrch Direct Draw
 Name[sl]=Površina Direct Draw
 Name[sr]=Површина за директно цртање
-Name[sr@Latn]=Површина за директно цртање
+Name[sr@Latn]=Površina za direktno crtanje
 Name[sv]=Direktrityta
 Name[ta]=நேரடியாக வரையும் சூழல்
 Name[tg]=Кашидани сатҳи мустақим
diff -urN -x CVS kdelibs.orig/kimgio/exr.kimgio kdelibs/kimgio/exr.kimgio
--- kdelibs.orig/kimgio/exr.kimgio	Sun Sep 12 15:56:01 2004
+++ kdelibs/kimgio/exr.kimgio	Mon Oct  4 15:25:05 2004
@@ -28,7 +28,7 @@
 Name[se]=ILM EXR-govva Kimgio
 Name[sk]=Obrázok ILM EXR KImgio
 Name[sr]=ILM EXR Kimgio слика
-Name[sr@Latn]=ILM EXR Kimgio слика
+Name[sr@Latn]=ILM EXR Kimgio slika
 Name[sv]=ILM EXR-bild Kimgio
 Name[ta]=ILM EXR பிம்ப கிம்ஜியோ
 Name[tg]=Тасвири ILM EXR Kimgio
diff -urN -x CVS kdelibs.orig/kimgio/rgb.kimgio kdelibs/kimgio/rgb.kimgio
--- kdelibs.orig/kimgio/rgb.kimgio	Thu Sep 30 16:03:30 2004
+++ kdelibs/kimgio/rgb.kimgio	Mon Oct  4 15:25:05 2004
@@ -38,7 +38,7 @@
 Name[sk]=SGI obrázok (RGB)
 Name[sl]=Slika SGI (RGB)
 Name[sr]=SGI слика (RGB)
-Name[sr@Latn]=SGI слика (RGB)
+Name[sr@Latn]=SGI slika (RGB)
 Name[sv]=SGI-bild (RGB)
 Name[ta]=GIF பிம்பம்
 Name[tg]=Тасвири SGI (RGB)
diff -urN -x CVS kdelibs.orig/kioslave/http/kcookiejar/kcookiejar.desktop kdelibs/kioslave/http/kcookiejar/kcookiejar.desktop
--- kdelibs.orig/kioslave/http/kcookiejar/kcookiejar.desktop	Mon Sep 20 15:53:24 2004
+++ kdelibs/kioslave/http/kcookiejar/kcookiejar.desktop	Mon Oct  4 15:25:10 2004
@@ -89,7 +89,7 @@
 Comment[sk]=Sleduje všetky cookie v systéme
 Comment[sl]=Opazuje vse piškotke v sistemu
 Comment[sr]=Води евиденцију о свим колачићима на систему
-Comment[sr@Latn]=Води евиденцију о свим колачићима на систему
+Comment[sr@Latn]=Vodi evidenciju o svim kolačićima na sistemu
 Comment[sv]=Håller ordning på alla kakor i systemet
 Comment[ta]=கணினியின் எல்லா தற்காலிக நினைவகங்களையும் கண்காணிக்கிறது
 Comment[tg]=Гузаргоҳи ҳамаша Cookies дар система муҳофизат кунед
diff -urN -x CVS kdelibs.orig/kresources/kresources.desktop kdelibs/kresources/kresources.desktop
--- kdelibs.orig/kresources/kresources.desktop	Sun Sep 12 15:56:08 2004
+++ kdelibs/kresources/kresources.desktop	Mon Oct  4 15:25:12 2004
@@ -98,7 +98,7 @@
 Comment[sk]=Nastavenie zdrojov KDE
 Comment[sl]=Nastavi vire KDE
 Comment[sr]=Подешавање KDE-ових ресурса
-Comment[sr@Latn]=Подешавање KDE-ових ресурса
+Comment[sr@Latn]=Podešavanje KDE-ovih resursa
 Comment[sv]=Anpassa KDE-resurser
 Comment[ta]=KDE மூலங்களை கட்டமை
 Comment[tg]=Танзими манбаъи истифодашуда тавассути китоби адрес
@@ -130,7 +130,7 @@
 Keywords[ro]=resurse,konnector,contact,calendar,notiţe,imap
 Keywords[sk]=zdroje,zdroj pre konnector,zdroj kontaktov,zdroj kalendára,zdroj poznámok,imap
 Keywords[sr]=resources,konnector resource,contact resource,calendar resource,notes resource,imap,ресурси,ресурс,повезивач,контакт ресурс, календарски ресурс
-Keywords[sr@Latn]=resources,konnector resource,contact resource,calendar resource,notes resource,imap,ресурси,ресурс,повезивач,контакт ресурс, календарски ресурс
+Keywords[sr@Latn]=resources,konnector resource,contact resource,calendar resource,notes resource,imap,resursi,resurs,povezivač,kontakt resurs, kalendarski resurs
 Keywords[sv]=resurser,konnector-resurs,kontaktresurs,kalenderresurs,anteckningsresurs,IMAP
 Keywords[ta]=மூலங்கள்,இணைப்பு மூலம்,தொடர்பு மூலம்,நாள்காட்டி மூலம்,குறிப்புகள் மூலம்e,imap
 Keywords[uk]=ресурси,ресурс для konnector,ресурс контактів,ресурс календаря,ресурс приміток,imap
diff -urN -x CVS kdelibs.orig/kresources/kresources_plugin.desktop kdelibs/kresources/kresources_plugin.desktop
--- kdelibs.orig/kresources/kresources_plugin.desktop	Tue Aug 24 08:29:33 2004
+++ kdelibs/kresources/kresources_plugin.desktop	Mon Oct  4 15:25:12 2004
@@ -45,7 +45,7 @@
 Comment[sl]=Vstavek za ogrodje KResource
 Comment[sq]=Shtojca KResource Framework
 Comment[sr]=Прикључак KResource Framework-а
-Comment[sr@Latn]=Dodatak KResource Framework-a
+Comment[sr@Latn]=Priključak KResource Framework-a
 Comment[sv]=Insticksprogram för resursramverk
 Comment[ta]=கேவழிமுறை கட்டமைப்பு சொருகுபொருள்
 Comment[tg]=Мутассалкунандаи қолабгар барои KResource
diff -urN -x CVS kdelibs.orig/ksmartcard/kardimpl/kcardgemsafe.desktop kdelibs/ksmartcard/kardimpl/kcardgemsafe.desktop
--- kdelibs.orig/ksmartcard/kardimpl/kcardgemsafe.desktop	Mon Sep 20 15:53:26 2004
+++ kdelibs/ksmartcard/kardimpl/kcardgemsafe.desktop	Mon Oct  4 15:25:14 2004
@@ -38,7 +38,7 @@
 Name[sk]=Implementácia karty GemSafe
 Name[sl]=Implementacija kartice GemSafe
 Name[sr]=Имплементација GemSafe картице
-Name[sr@Latn]=Имплементација GemSafe картице
+Name[sr@Latn]=Implementacija GemSafe kartice
 Name[sv]=GemSafe-kortimplementation
 Name[ta]=GemSafe அட்டை அமலாக்கம்
 Name[tg]=Пиёдасозии корти GemSafe
diff -urN -x CVS kdelibs.orig/ksmartcard/kardimpl/kcardgsm.desktop kdelibs/ksmartcard/kardimpl/kcardgsm.desktop
--- kdelibs.orig/ksmartcard/kardimpl/kcardgsm.desktop	Mon Sep 20 15:53:26 2004
+++ kdelibs/ksmartcard/kardimpl/kcardgsm.desktop	Mon Oct  4 15:25:14 2004
@@ -38,7 +38,7 @@
 Name[sk]=Implementácia karty GSM
 Name[sl]=Implementacija kartice GSM
 Name[sr]=Имплементација GSM картице
-Name[sr@Latn]=Имплементација GSM картице
+Name[sr@Latn]=Implementacija GSM kartice
 Name[sv]=GSM-kortimplementation
 Name[ta]=GSM அட்டை அமலாக்கம்
 Name[tg]=Пиёдасозии корти GSM
diff -urN -x CVS kdelibs.orig/kspell2/kspellclient.desktop kdelibs/kspell2/kspellclient.desktop
--- kdelibs.orig/kspell2/kspellclient.desktop	Sun Sep 12 15:56:12 2004
+++ kdelibs/kspell2/kspellclient.desktop	Mon Oct  4 15:25:16 2004
@@ -34,7 +34,7 @@
 Comment[sk]=Klient KSpell
 Comment[sl]=Odjemnik KSpell
 Comment[sr]=KSpell-ов клијент
-Comment[sr@Latn]=KSpell-ов клијент
+Comment[sr@Latn]=KSpell-ov klijent
 Comment[sv]=Kspell-klient
 Comment[ta]=ஓட்டு எழுத்தாக்கம்
 Comment[tg]=Клиенти KSpell
diff -urN -x CVS kdelibs.orig/kwallet/backend/kwalletbackend.cc kdelibs/kwallet/backend/kwalletbackend.cc
--- kdelibs.orig/kwallet/backend/kwalletbackend.cc	Mon Jul 12 00:31:59 2004
+++ kdelibs/kwallet/backend/kwalletbackend.cc	Mon Oct  4 11:29:43 2004
@@ -338,6 +338,7 @@
 		MD5Digest ba;
 		QMap<MD5Digest,QValueList<MD5Digest> >::iterator it;
 		Q_UINT32 fsz;
+		if (hds.atEnd()) return -43;
 		hds.readRawBytes(reinterpret_cast<char *>(d), 16);
 		hds >> fsz;
 		ba.duplicate(reinterpret_cast<char *>(d), 16);
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.mozilla.xul+xml.desktop kdelibs/mimetypes/application/vnd.mozilla.xul+xml.desktop
--- kdelibs.orig/mimetypes/application/vnd.mozilla.xul+xml.desktop	Mon Sep 20 15:53:29 2004
+++ kdelibs/mimetypes/application/vnd.mozilla.xul+xml.desktop	Mon Oct  4 15:25:18 2004
@@ -35,7 +35,7 @@
 Comment[sk]=Súbor Mozilla XUL
 Comment[sl]=Datoteka Mozilla XUL
 Comment[sr]=Mozilla XUL фајл
-Comment[sr@Latn]=Mozilla XUL фајл
+Comment[sr@Latn]=Mozilla XUL fajl
 Comment[sv]=Mozilla XUL-fil
 Comment[ta]=மோசில்லா XUL ஆவணம்
 Comment[tg]=Файли Mozilla XUL 
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.ms-asf.desktop kdelibs/mimetypes/application/vnd.ms-asf.desktop
--- kdelibs.orig/mimetypes/application/vnd.ms-asf.desktop	Tue Aug 24 08:29:48 2004
+++ kdelibs/mimetypes/application/vnd.ms-asf.desktop	Tue Oct  5 15:20:16 2004
@@ -8,6 +8,7 @@
 Comment[az]=ASF Mediyası
 Comment[bg]=Медия ASF
 Comment[bn]=এ-এস-এফ মিডিয়া
+Comment[br]=Media ASF
 Comment[bs]=ASF medija
 Comment[ca]=Medi ASF
 Comment[cs]=ASF média
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.stardivision.chart.desktop kdelibs/mimetypes/application/vnd.stardivision.chart.desktop
--- kdelibs.orig/mimetypes/application/vnd.stardivision.chart.desktop	Thu Sep 30 16:03:33 2004
+++ kdelibs/mimetypes/application/vnd.stardivision.chart.desktop	Mon Oct  4 15:25:18 2004
@@ -33,7 +33,7 @@
 Comment[sk]=Graf StarOffice
 Comment[sl]=Preglednica StarOffice
 Comment[sr]=StarOffice-ов графикон
-Comment[sr@Latn]=StarOffice-ов графикон
+Comment[sr@Latn]=StarOffice-ov grafikon
 Comment[sv]=StarOffice diagram
 Comment[ta]=ஸ்டார்ஆபீஸ் வரைபடம்
 Comment[uk]=Діаграма StarOffice
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.stardivision.draw.desktop kdelibs/mimetypes/application/vnd.stardivision.draw.desktop
--- kdelibs.orig/mimetypes/application/vnd.stardivision.draw.desktop	Thu Sep 30 16:03:33 2004
+++ kdelibs/mimetypes/application/vnd.stardivision.draw.desktop	Mon Oct  4 15:25:18 2004
@@ -36,7 +36,7 @@
 Comment[sk]=Kresba StarOffice
 Comment[sl]=Risanje StarOffice
 Comment[sr]=StarOffice-ов цртеж
-Comment[sr@Latn]=StarOffice-ов цртеж
+Comment[sr@Latn]=StarOffice-ov crtež
 Comment[sv]=StarOffice teckning
 Comment[ta]=மைக்ரோசாப்ட் ஆபிஸ் சித்திரம்
 Comment[tg]=Расмкашӣ Microsoft Office
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.stardivision.math.desktop kdelibs/mimetypes/application/vnd.stardivision.math.desktop
--- kdelibs.orig/mimetypes/application/vnd.stardivision.math.desktop	Thu Sep 30 16:03:33 2004
+++ kdelibs/mimetypes/application/vnd.stardivision.math.desktop	Mon Oct  4 15:25:18 2004
@@ -36,7 +36,7 @@
 Comment[sk]=Dokument StarOffice Math
 Comment[sl]=Dokument StarOffice Math
 Comment[sr]=StarOffice-ов математички документ
-Comment[sr@Latn]=StarOffice-ов математички документ
+Comment[sr@Latn]=StarOffice-ov matematički dokument
 Comment[sv]=StarOffice matematikdokument
 Comment[ta]=ஸ்டார்ஆபீஸ் ரைட்டர் ஆவணம்
 Comment[tg]=Санади математикии StarOffice
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.stardivision.writer-global.desktop kdelibs/mimetypes/application/vnd.stardivision.writer-global.desktop
--- kdelibs.orig/mimetypes/application/vnd.stardivision.writer-global.desktop	Thu Sep 30 16:03:33 2004
+++ kdelibs/mimetypes/application/vnd.stardivision.writer-global.desktop	Mon Oct  4 15:25:18 2004
@@ -36,7 +36,7 @@
 Comment[sk]=Hlavný dokument StarOffice Writer
 Comment[sl]=Glavni dokument StarOffice Writer
 Comment[sr]=StarOffice Writer-ов главни документ
-Comment[sr@Latn]=StarOffice Writer-ов главни документ
+Comment[sr@Latn]=StarOffice Writer-ov glavni dokument
 Comment[sv]=StarOffice Writer-huvuddokument
 Comment[ta]=ஸ்டார்ஆபீஸ் ரைட்டர் மூல ஆவணம்
 Comment[tg]=Санади устоди нависаи StarOffice
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.sun.xml.calc.template.desktop kdelibs/mimetypes/application/vnd.sun.xml.calc.template.desktop
--- kdelibs.orig/mimetypes/application/vnd.sun.xml.calc.template.desktop	Thu Sep 30 16:03:33 2004
+++ kdelibs/mimetypes/application/vnd.sun.xml.calc.template.desktop	Mon Oct  4 15:25:18 2004
@@ -33,7 +33,7 @@
 Comment[sk]=Šablóna tabuľky OpenOffice.org
 Comment[sl]=Predloga preglednice OpenOffice.org
 Comment[sr]=OpenOffice.org-ов шаблон табеле
-Comment[sr@Latn]=OpenOffice.org-ов шаблон табеле
+Comment[sr@Latn]=OpenOffice.org-ov šablon tabele
 Comment[sv]=OpenOffice.org kalkylarksmall
 Comment[ta]=ழ-ஓப்பன்ஆபீஸ் விரிதாள்
 Comment[tg]=Ҷадвали OpenOffice.org
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.sun.xml.draw.template.desktop kdelibs/mimetypes/application/vnd.sun.xml.draw.template.desktop
--- kdelibs.orig/mimetypes/application/vnd.sun.xml.draw.template.desktop	Thu Sep 30 16:03:33 2004
+++ kdelibs/mimetypes/application/vnd.sun.xml.draw.template.desktop	Mon Oct  4 15:25:18 2004
@@ -33,7 +33,7 @@
 Comment[sk]=Šablóna kresby OpenOffice.org
 Comment[sl]=Predloga risanja OpenOffice.org
 Comment[sr]=OpenOffice.org-ов шаблон цртежа
-Comment[sr@Latn]=OpenOffice.org-ов шаблон цртежа
+Comment[sr@Latn]=OpenOffice.org-ov šablon crteža
 Comment[sv]=OpenOffice.org teckningsmall
 Comment[ta]=ழ-ஓப்பன்ஆபீஸ் வரைபடம்
 Comment[tg]=Нақши OpenOffice.org
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.sun.xml.impress.template.desktop kdelibs/mimetypes/application/vnd.sun.xml.impress.template.desktop
--- kdelibs.orig/mimetypes/application/vnd.sun.xml.impress.template.desktop	Thu Sep 30 16:03:33 2004
+++ kdelibs/mimetypes/application/vnd.sun.xml.impress.template.desktop	Mon Oct  4 15:25:18 2004
@@ -33,7 +33,7 @@
 Comment[sk]=Šablóna prezentácie OpenOffice.org
 Comment[sl]=Predloga predstavitve OpenOffice.org
 Comment[sr]=OpenOffice.org-ов шаблон презентације
-Comment[sr@Latn]=OpenOffice.org-ов шаблон презентације
+Comment[sr@Latn]=OpenOffice.org-ov šablon prezentacije
 Comment[sv]=OpenOffice.org presentationsmall
 Comment[ta]=ழ-ஓப்பன்ஆபீஸ் அளிக்கை
 Comment[tg]=Тақдимоти OpenOffice.org
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.sun.xml.writer.master.desktop kdelibs/mimetypes/application/vnd.sun.xml.writer.master.desktop
--- kdelibs.orig/mimetypes/application/vnd.sun.xml.writer.master.desktop	Thu Sep 30 16:03:33 2004
+++ kdelibs/mimetypes/application/vnd.sun.xml.writer.master.desktop	Mon Oct  4 15:25:18 2004
@@ -31,7 +31,7 @@
 Comment[sk]=Hlavný textový dokument OpenOffice.org
 Comment[sl]=Glavni besedilni dokument OpenOffice.org
 Comment[sr]=OpenOffice.org-ов главни текстуални документ
-Comment[sr@Latn]=OpenOffice.org-ов главни текстуални документ
+Comment[sr@Latn]=OpenOffice.org-ov glavni tekstualni dokument
 Comment[sv]=OpenOffice.org huvudtextdokument
 Comment[ta]=OpenOffice.org ஆசிரியர் உரை ஆவணம்
 Comment[tg]=Устоди санади OpenOffice.org
diff -urN -x CVS kdelibs.orig/mimetypes/application/vnd.sun.xml.writer.template.desktop kdelibs/mimetypes/application/vnd.sun.xml.writer.template.desktop
--- kdelibs.orig/mimetypes/application/vnd.sun.xml.writer.template.desktop	Thu Sep 30 16:03:33 2004
+++ kdelibs/mimetypes/application/vnd.sun.xml.writer.template.desktop	Mon Oct  4 15:25:18 2004
@@ -33,7 +33,7 @@
 Comment[sk]=Šablóna textového dokumentu OpenOffice.org
 Comment[sl]=Predloga besedilnega dokumenta OpenOffice.org
 Comment[sr]=OpenOffice.org-ов шаблон текстуалног документа
-Comment[sr@Latn]=OpenOffice.org-ов шаблон текстуалног документа
+Comment[sr@Latn]=OpenOffice.org-ov šablon tekstualnog dokumenta
 Comment[sv]=OpenOffice.org textdokumentmall
 Comment[ta]=ழ-ஓப்பன்ஆபீஸ் உரை ஆவணம்
 Comment[tg]=Санади OpenOffice.org
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-bz2dvi.desktop kdelibs/mimetypes/application/x-bz2dvi.desktop
--- kdelibs.orig/mimetypes/application/x-bz2dvi.desktop	Mon Sep 20 15:53:29 2004
+++ kdelibs/mimetypes/application/x-bz2dvi.desktop	Mon Oct  4 15:25:18 2004
@@ -29,7 +29,7 @@
 Comment[sk]=Komprimovaný súbor DVI pomocou bzip2
 Comment[sl]=Datoteka DVI, stisnjena s bzip2
 Comment[sr]=Bzip2 компресовани DVI фајл
-Comment[sr@Latn]=Bzip2 компресовани DVI фајл
+Comment[sr@Latn]=Bzip2 kompresovani DVI fajl
 Comment[sv]=Bzip2-komprimerad DVI-fil
 Comment[ta]=Bzip2 அழுத்திய DVI கோப்பு
 Comment[tg]=Bzip2 Файли DVI фушурдашуда
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-executable-script.desktop kdelibs/mimetypes/application/x-executable-script.desktop
--- kdelibs.orig/mimetypes/application/x-executable-script.desktop	Mon Sep 20 15:53:29 2004
+++ kdelibs/mimetypes/application/x-executable-script.desktop	Mon Oct  4 15:25:18 2004
@@ -29,7 +29,7 @@
 Comment[sk]=Skript (možno spustiteľný)
 Comment[sl]=Skript (verjetno izvedljiv)
 Comment[sr]=Скрипта (можда извршна)
-Comment[sr@Latn]=Скрипта (можда извршна)
+Comment[sr@Latn]=Skripta (možda izvršna)
 Comment[sv]=Skript (möjligen körbart)
 Comment[ta]=சிறுநிரல் (இயக்க முடிந்த)
 Comment[tg]=Скрипт (мумкин аст, ки ба кор бурда бошад)
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-gzdvi.desktop kdelibs/mimetypes/application/x-gzdvi.desktop
--- kdelibs.orig/mimetypes/application/x-gzdvi.desktop	Mon Sep 20 15:53:29 2004
+++ kdelibs/mimetypes/application/x-gzdvi.desktop	Mon Oct  4 15:25:18 2004
@@ -29,7 +29,7 @@
 Comment[sk]=Komprimovaný súbor DVI pomocou gzip
 Comment[sl]=Datoteka DVI, stisnjena s gzip-om
 Comment[sr]=Gzip компресовани DVI фајл
-Comment[sr@Latn]=Gzip компресовани DVI фајл
+Comment[sr@Latn]=Gzip kompresovani DVI fajl
 Comment[sv]=Gzip-komprimerad DVI-fil
 Comment[ta]=Gzip அழுத்திய கோப்பு
 Comment[tg]=Gzip Файли DVI фушурдашуда
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-ica.desktop kdelibs/mimetypes/application/x-ica.desktop
--- kdelibs.orig/mimetypes/application/x-ica.desktop	Mon Sep 20 15:53:29 2004
+++ kdelibs/mimetypes/application/x-ica.desktop	Mon Oct  4 15:25:18 2004
@@ -32,7 +32,7 @@
 Comment[se]=Heivet Citrix ICA-klientta
 Comment[sk]=Konfigurácia klienta Citrix ICA
 Comment[sr]=Подешавање Citrix-овог ICA клијента
-Comment[sr@Latn]=Подешавање Citrix-овог ICA клијента
+Comment[sr@Latn]=Podešavanje Citrix-ovog ICA klijenta
 Comment[sv]=Citrix ICA-klientanpassning
 Comment[ta]=Citrix ICA உறுப்பினம் வடிவமைப்பு
 Comment[tg]=Танзимоти коргири Citrix ICA
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-iso.desktop kdelibs/mimetypes/application/x-iso.desktop
--- kdelibs.orig/mimetypes/application/x-iso.desktop	Mon Sep 20 15:53:29 2004
+++ kdelibs/mimetypes/application/x-iso.desktop	Mon Oct  4 15:25:18 2004
@@ -33,7 +33,7 @@
 Comment[sk]=Obraz disku ISO9660
 Comment[sl]=Podoba plošče ISO9660
 Comment[sr]=Фајл ISO9660 слике
-Comment[sr@Latn]=Фајл ISO9660 слике
+Comment[sr@Latn]=Fajl ISO9660 slike
 Comment[sv]=ISO9660-avbildsfil
 Comment[ta]=ISO9660 பிம்பக் கோப்பு
 Comment[tg]=Файли тасвири ISO9660
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-javascript.desktop kdelibs/mimetypes/application/x-javascript.desktop
--- kdelibs.orig/mimetypes/application/x-javascript.desktop	Sun Sep 12 15:56:15 2004
+++ kdelibs/mimetypes/application/x-javascript.desktop	Mon Oct  4 15:25:18 2004
@@ -35,7 +35,7 @@
 Comment[sk]=Zdrojový kód JavaScript
 Comment[sl]=Datoteka JavaScript
 Comment[sr]=JavaScript фајл
-Comment[sr@Latn]=JavaScript фајл
+Comment[sr@Latn]=JavaScript fajl
 Comment[sv]=Javaskript-fil
 Comment[ta]=ஜாவா மூலக் கோப்பு
 Comment[tg]=Файли барномавии Java
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-msmetafile.desktop kdelibs/mimetypes/application/x-msmetafile.desktop
--- kdelibs.orig/mimetypes/application/x-msmetafile.desktop	Mon Sep 20 15:53:29 2004
+++ kdelibs/mimetypes/application/x-msmetafile.desktop	Mon Oct  4 15:25:18 2004
@@ -34,7 +34,7 @@
 Comment[se]=Microsoft-metafiila
 Comment[sk]=Metafile Microsoft
 Comment[sr]=Microsoft-ов метафајл
-Comment[sr@Latn]=Microsoft-ов метафајл
+Comment[sr@Latn]=Microsoft-ov metafajl
 Comment[sv]=Microsoft-metafil
 Comment[ta]=மைக்ரோசாஃட் மெடாகோப்பு
 Comment[tg]=Метафайли QPicture
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-pak.desktop kdelibs/mimetypes/application/x-pak.desktop
--- kdelibs.orig/mimetypes/application/x-pak.desktop	Mon Sep 20 15:53:29 2004
+++ kdelibs/mimetypes/application/x-pak.desktop	Mon Oct  4 15:25:18 2004
@@ -34,7 +34,7 @@
 Comment[sk]=Archív PAK
 Comment[sl]=Datotečni arhiv PAK
 Comment[sr]=Фајл PAK архиве
-Comment[sr@Latn]=Фајл PAK архиве
+Comment[sr@Latn]=Fajl PAK arhive
 Comment[sv]=PAK-filarkiv
 Comment[ta]=PAK கோப்பு காப்பகம்
 Comment[tg]=Файли архивии PAK
diff -urN -x CVS kdelibs.orig/mimetypes/application/x-vnd.kde.kplato.desktop kdelibs/mimetypes/application/x-vnd.kde.kplato.desktop
--- kdelibs.orig/mimetypes/application/x-vnd.kde.kplato.desktop	Sun Sep 12 15:56:15 2004
+++ kdelibs/mimetypes/application/x-vnd.kde.kplato.desktop	Mon Oct  4 15:25:18 2004
@@ -31,7 +31,7 @@
 Comment[se]=KPlato-prošeaktajođiheapmedokumeanta
 Comment[sk]=KPlato dokument pre správu projektov
 Comment[sr]=KPlato-ов документ пројекта управљања
-Comment[sr@Latn]=KPlato-ов документ пројекта управљања
+Comment[sr@Latn]=KPlato-ov dokument projekta upravljanja
 Comment[sv]=Kplato projekthanteringsdokument
 Comment[ta]=கேப்ளாட்டோ திட்ட மேலாண்மை ஆவணம்
 Comment[tg]=Сохти идоракунии санади KPlato
diff -urN -x CVS kdelibs.orig/mimetypes/audio/mpeg.desktop kdelibs/mimetypes/audio/mpeg.desktop
--- kdelibs.orig/mimetypes/audio/mpeg.desktop	Sun Sep 12 15:56:17 2004
+++ kdelibs/mimetypes/audio/mpeg.desktop	Mon Oct  4 15:25:19 2004
@@ -40,7 +40,7 @@
 Comment[sk]=Zvuk MPEG
 Comment[sl]=Zvok MPEG
 Comment[sr]=MPEG аудио
-Comment[sr@Latn]=MPEG аудио
+Comment[sr@Latn]=MPEG audio
 Comment[sv]=MPEG-ljud
 Comment[ta]=MPEG கேட்பொலி
 Comment[th]=แฟ้มเสียง MPEG
diff -urN -x CVS kdelibs.orig/mimetypes/audio/prs.sid.desktop kdelibs/mimetypes/audio/prs.sid.desktop
--- kdelibs.orig/mimetypes/audio/prs.sid.desktop	Mon Sep 20 15:53:29 2004
+++ kdelibs/mimetypes/audio/prs.sid.desktop	Mon Oct  4 15:25:19 2004
@@ -37,7 +37,7 @@
 Comment[sk]=Zvuk C64 SID
 Comment[sl]=Glasba C64 SID
 Comment[sr]=C64 SID музика
-Comment[sr@Latn]=C64 SID музика
+Comment[sr@Latn]=C64 SID muzika
 Comment[sv]=C64 SID-musik
 Comment[ta]=குறுந்தட்டு ஒலி
 Comment[tg]=Мусиқии C64 SID
diff -urN -x CVS kdelibs.orig/mimetypes/audio/x-mp2.desktop kdelibs/mimetypes/audio/x-mp2.desktop
--- kdelibs.orig/mimetypes/audio/x-mp2.desktop	Sun Sep 12 15:56:17 2004
+++ kdelibs/mimetypes/audio/x-mp2.desktop	Mon Oct  4 15:25:19 2004
@@ -37,7 +37,7 @@
 Comment[sk]=Zvuk MPEG layer 2
 Comment[sl]=Zvok oblike MPEG layer 2
 Comment[sr]=MPEG ниво 2 аудио
-Comment[sr@Latn]=MPEG ниво 2 аудио
+Comment[sr@Latn]=MPEG nivo 2 audio
 Comment[sv]=MPEG lager 2-ljud
 Comment[ta]=MPEG அடுக்கு 3  கேட்பொலி
 Comment[tg]=Файли садои лояи 2 барои MPEG
diff -urN -x CVS kdelibs.orig/mimetypes/audio/x-ms-wma.desktop kdelibs/mimetypes/audio/x-ms-wma.desktop
--- kdelibs.orig/mimetypes/audio/x-ms-wma.desktop	Sun Sep 12 15:56:17 2004
+++ kdelibs/mimetypes/audio/x-ms-wma.desktop	Mon Oct  4 15:25:19 2004
@@ -35,7 +35,7 @@
 Comment[se]=Windows Media-jietna
 Comment[sl]=Avdio Windows Media
 Comment[sr]=Windows медија аудио
-Comment[sr@Latn]=Windows медија аудио
+Comment[sr@Latn]=Windows medija audio
 Comment[sv]=Windows-medialjud
 Comment[ta]=சாளரங்களின் ஊடக கேட்பொலி
 Comment[tg]=Аудиофайли Windows Media
diff -urN -x CVS kdelibs.orig/mimetypes/audio/x-musepack.desktop kdelibs/mimetypes/audio/x-musepack.desktop
--- kdelibs.orig/mimetypes/audio/x-musepack.desktop	Sun Sep 12 15:56:17 2004
+++ kdelibs/mimetypes/audio/x-musepack.desktop	Mon Oct  4 15:25:19 2004
@@ -35,7 +35,7 @@
 Comment[sk]=Zvuk Musepack
 Comment[sl]=Zvok Musepack
 Comment[sr]=Musepack аудио
-Comment[sr@Latn]=Musepack аудио
+Comment[sr@Latn]=Musepack audio
 Comment[sv]=Musepack-ljud
 Comment[ta]= மியுஸ்பேக் ஒலி
 Comment[tg]=Файли садоии Musepack
diff -urN -x CVS kdelibs.orig/mimetypes/image/x-dds.desktop kdelibs/mimetypes/image/x-dds.desktop
--- kdelibs.orig/mimetypes/image/x-dds.desktop	Sun Sep 12 15:56:17 2004
+++ kdelibs/mimetypes/image/x-dds.desktop	Mon Oct  4 15:25:20 2004
@@ -30,7 +30,7 @@
 Comment[sk]=Povrch DirectDraw
 Comment[sl]=Površina DirectDraw
 Comment[sr]=Површина за директно цртање
-Comment[sr@Latn]=Површина за директно цртање
+Comment[sr@Latn]=Površina za direktno crtanje
 Comment[sv]=Direktrityta
 Comment[ta]=நேரடி வரையும் சூழல்
 Comment[tg]=Кашидани сатҳи мустақим
diff -urN -x CVS kdelibs.orig/mimetypes/image/x-exr.desktop kdelibs/mimetypes/image/x-exr.desktop
--- kdelibs.orig/mimetypes/image/x-exr.desktop	Sun Sep 12 15:56:17 2004
+++ kdelibs/mimetypes/image/x-exr.desktop	Mon Oct  4 15:25:20 2004
@@ -39,7 +39,7 @@
 Comment[sk]=ILM EXR obrázok
 Comment[sl]=Slika ILM EXR
 Comment[sr]=ILM EXR слика
-Comment[sr@Latn]=ILM EXR слика
+Comment[sr@Latn]=ILM EXR slika
 Comment[sv]=ILM EXR-bild
 Comment[ta]=ILM EXR பிம்பம்
 Comment[tg]=Тасвири ILM EXR
diff -urN -x CVS kdelibs.orig/mimetypes/image/x-rgb.desktop kdelibs/mimetypes/image/x-rgb.desktop
--- kdelibs.orig/mimetypes/image/x-rgb.desktop	Sun Sep 12 15:56:17 2004
+++ kdelibs/mimetypes/image/x-rgb.desktop	Mon Oct  4 15:25:20 2004
@@ -39,7 +39,7 @@
 Comment[sk]=Obrázok SGI (RGB)
 Comment[sl]=Slika SGI (RGB)
 Comment[sr]=SGI слика (RGB)
-Comment[sr@Latn]=SGI слика (RGB)
+Comment[sr@Latn]=SGI slika (RGB)
 Comment[sv]=SGI-bild (RGB)
 Comment[ta]=GIF பிம்பம்
 Comment[tg]=Тасвири SGI (RGB)
diff -urN -x CVS kdelibs.orig/mimetypes/text/x-hex.desktop kdelibs/mimetypes/text/x-hex.desktop
--- kdelibs.orig/mimetypes/text/x-hex.desktop	Mon Sep 20 15:53:29 2004
+++ kdelibs/mimetypes/text/x-hex.desktop	Mon Oct  4 15:25:20 2004
@@ -31,7 +31,7 @@
 Comment[se]=Intel®-heksadesimála objeaktafiila
 Comment[sk]=Intel® hexadecimalný objektový súbor
 Comment[sr]=Intel® хексадецимални формат објектног фајла
-Comment[sr@Latn]=Intel® хексадецимални формат објектног фајла
+Comment[sr@Latn]=Intel® heksadecimalni format objektnog fajla
 Comment[sv]=Intel® hexadecimalt objektfilformat
 Comment[ta]=Qt மெடா பொருள் கோப்பு
 Comment[tg]=Андозаи Файли Объекти Intel® Hexadecimal
diff -urN -x CVS kdelibs.orig/mimetypes/text/x-katefilelist.desktop kdelibs/mimetypes/text/x-katefilelist.desktop
--- kdelibs.orig/mimetypes/text/x-katefilelist.desktop	Tue Aug 31 09:21:47 2004
+++ kdelibs/mimetypes/text/x-katefilelist.desktop	Mon Oct  4 15:25:20 2004
@@ -48,7 +48,7 @@
 Comment[sl]=Seznam datotek (za vstavek seznama datotek Kate)
 Comment[sq]=Lista e Skedarëve (për Shtojcën Kate File List Loader)
 Comment[sr]=Листа фајлова (за Kate-ов прикључак за учитавање листе фајлова)
-Comment[sr@Latn]=Lista fajlova (za Kate-ov dodatak za učitavanje liste fajlova)
+Comment[sr@Latn]=Lista fajlova (za Kate-ov priključak za učitavanje liste fajlova)
 Comment[sv]=Fillista för (insticksprogram för Kates fillistladdare)
 Comment[ta]=கோப்புப் பட்டியல் (கேட் கோப்புப் பட்டியல் ஏற்றச் செருகுப்பொருளுக்கு)
 Comment[tg]=Рӯйхати Файл ( барои Файли Kate Модули Пурборкуни)
diff -urN -x CVS kdelibs.orig/mimetypes/video/x-ms-asf.desktop kdelibs/mimetypes/video/x-ms-asf.desktop
--- kdelibs.orig/mimetypes/video/x-ms-asf.desktop	Tue Aug 24 08:30:00 2004
+++ kdelibs/mimetypes/video/x-ms-asf.desktop	Tue Oct  5 15:20:16 2004
@@ -10,6 +10,7 @@
 Comment[az]=ASF Mediyası
 Comment[bg]=Медия ASF
 Comment[bn]=এ-এস-এফ মিডিয়া
+Comment[br]=Media ASF
 Comment[bs]=ASF medija
 Comment[ca]=Medi ASF
 Comment[cs]=ASF média
diff -urN -x CVS kdelibs.orig/pics/crystalsvg/index.theme kdelibs/pics/crystalsvg/index.theme
--- kdelibs.orig/pics/crystalsvg/index.theme	Sun Sep 12 15:56:20 2004
+++ kdelibs/pics/crystalsvg/index.theme	Mon Oct  4 15:25:22 2004
@@ -49,7 +49,7 @@
 Comment[sk]=Téma ikon od Everaldo.com Design Studio
 Comment[sl]=Ikonska tema od oblikovalskega sutida Everaldo.com
 Comment[sr]=Тема икона Everaldo.com дизајнерског студија
-Comment[sr@Latn]=Тема икона Everaldo.com дизајнерског студија
+Comment[sr@Latn]=Tema ikona Everaldo.com dizajnerskog studija
 Comment[sv]=Ikontema av Everaldo.com designstudio
 Comment[ta]=எவரால்டோ.காம் வடிவமைப்பு படப்பிடிப்பகத்தின் சின்னம்
 Comment[tg]=Мавзӯъҳои нишонаҳо аз Эвералдо мебошад (www.everaldo.com)
