--- kdelibs-3.5.10/kio/kio/kdirwatch.cpp.org	2008-11-06 18:36:38.663669249 +0100
+++ kdelibs-3.5.10/kio/kio/kdirwatch.cpp	2008-11-06 18:37:21.566582248 +0100
@@ -68,38 +68,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <sys/syscall.h>
-#include <linux/types.h>
-// Linux kernel headers are documented to not compile
-#define _S390_BITOPS_H
-#include <linux/inotify.h>
-
-static inline int inotify_init (void)
-{
-  return syscall (__NR_inotify_init);
-}
-
-static inline int inotify_add_watch (int fd, const char *name, __u32 mask)
-{
-  return syscall (__NR_inotify_add_watch, fd, name, mask);
-}
-
-static inline int inotify_rm_watch (int fd, __u32 wd)
-{
-  return syscall (__NR_inotify_rm_watch, fd, wd);
-}
-
-#ifndef  IN_ONLYDIR
-#define  IN_ONLYDIR 0x01000000 
-#endif
-
-#ifndef IN_DONT_FOLLOW
-#define IN_DONT_FOLLOW 0x02000000
-#endif
-
-#ifndef IN_MOVE_SELF
-#define IN_MOVE_SELF 0x00000800
-#endif
-
+#include <sys/inotify.h>
 #endif
 
 #include <sys/utsname.h>
