--- kdelibs-3.1.3/kdecore/kglobalsettings.cpp.orig	Wed Jul 30 23:06:45 2003
+++ kdelibs-3.1.3/kdecore/kglobalsettings.cpp	Wed Jul 30 23:07:44 2003
@@ -280,8 +280,8 @@
     if (_generalFont)
         return *_generalFont;
 
-    _generalFont = new QFont("helvetica", 12);
-    _generalFont->setPixelSize(12);
+    _generalFont = new QFont("helvetica", 11);
+    _generalFont->setPixelSize(11);
     _generalFont->setStyleHint(QFont::SansSerif);
 
     KConfig *c = KGlobal::config();
@@ -312,7 +312,7 @@
     if(_toolBarFont)
         return *_toolBarFont;
 
-    _toolBarFont = new QFont("helvetica", 10);
+    _toolBarFont = new QFont("helvetica", 11);
     _toolBarFont->setPixelSize(10);
     _toolBarFont->setStyleHint(QFont::SansSerif);
 
@@ -328,7 +328,7 @@
     if(_menuFont)
         return *_menuFont;
 
-    _menuFont = new QFont("helvetica", 12);
+    _menuFont = new QFont("helvetica", 11);
     _menuFont->setPixelSize(12);
     _menuFont->setStyleHint(QFont::SansSerif);
 
@@ -344,7 +344,7 @@
     if(_windowTitleFont)
         return *_windowTitleFont;
 
-    _windowTitleFont = new QFont("helvetica", 12, QFont::Bold);
+    _windowTitleFont = new QFont("helvetica", 11, QFont::Bold);
     _windowTitleFont->setPixelSize(12);
     _windowTitleFont->setStyleHint(QFont::SansSerif);
 
