diff -urN kdelibs-2.2.1.orig/kdecore/kstddirs.cpp kdelibs-2.2.1/kdecore/kstddirs.cpp
--- kdelibs-2.2.1.orig/kdecore/kstddirs.cpp	Wed Jan 24 23:29:38 2001
+++ kdelibs-2.2.1/kdecore/kstddirs.cpp	Thu Nov  1 19:50:21 2001
@@ -692,9 +692,9 @@
     if (!strcmp(type, "data"))
 	return "share/apps/";
     if (!strcmp(type, "html"))
-	return "share/doc/HTML/";
+	return "share/doc/kde/HTML/";
     if (!strcmp(type, "icon"))
-	return "share/icons/";
+	return "share/pixmaps/";
     if (!strcmp(type, "config"))
 	return "share/config/";
     if (!strcmp(type, "pixmap"))
@@ -833,6 +833,7 @@
     QString dir = QString("%1tmp-%2/").arg(localKdeDir).arg(hostname);
     addResourceDir("tmp", dir);
 
+    addResourceDir("html", "/usr/share/doc/kde/HTML");
 }
 
 void KStandardDirs::checkConfig() const
