Index: dcopserver.cpp
===================================================================
RCS file: /home/kde/kdelibs/dcop/dcopserver.cpp,v
retrieving revision 1.160
retrieving revision 1.160.2.1
diff -u -r1.160 -r1.160.2.1
--- kdelibs/dcop/dcopserver.cpp~	14 Oct 2003 11:57:50 -0000	1.160
+++ kdelibs/dcop/dcopserver.cpp	3 Feb 2004 11:49:29 -0000	1.160.2.1
@@ -931,6 +931,7 @@
     return 1;
 }
 
+static int pipeOfDeath[2];
 
 static void sighandler(int sig)
 {
@@ -939,8 +940,7 @@
 	return;
     }
 
-    qApp->quit();
-    //exit(0);
+    write(pipeOfDeath[1], "x", 1);
 }
 
 DCOPServer::DCOPServer(bool _suicide)
@@ -1629,6 +1629,8 @@
 	    return 0; // get rid of controlling terminal
     }
 
+    pipe(pipeOfDeath);
+
     signal(SIGHUP, sighandler);
     signal(SIGTERM, sighandler);
     signal(SIGPIPE, SIG_IGN);
@@ -1636,7 +1638,10 @@
     putenv(strdup("SESSION_MANAGER="));
 
     QApplication a( argc, argv, false );
-
+    
+    QSocketNotifier DEATH(pipeOfDeath[0], QSocketNotifier::Read, 0, 0);
+    a.connect(&DEATH, SIGNAL(activated(int)), SLOT(quit()));
+    
     IceSetIOErrorHandler (IoErrorHandler );
     DCOPServer *server = new DCOPServer(suicide); // this sets the_server
 
