cvs server: Diffing .
Index: kpfilterpage.cpp
===================================================================
RCS file: /home/kde/kdelibs/kdeprint/kpfilterpage.cpp,v
retrieving revision 1.2
diff -u -p -r1.2 kpfilterpage.cpp
--- kdelibs/kdeprint/kpfilterpage.cpp	2001/05/30 11:50:30	1.2
+++ kdelibs/kdeprint/kpfilterpage.cpp	2002/02/14 08:53:49
@@ -80,12 +80,8 @@ KPFilterPage::KPFilterPage(QWidget *pare
 	l2->addSpacing(5);
 	l2->addWidget(m_configure);
 	l2->addStretch(1);
+	updateButton();
 
-	m_remove->setEnabled(false);
-	m_up->setEnabled(false);
-	m_down->setEnabled(false);
-	m_configure->setEnabled(false);
-
 	resize(100,50);
 }
 
@@ -93,6 +89,16 @@ KPFilterPage::~KPFilterPage()
 {
 }
 
+void KPFilterPage::updateButton()
+{
+  QListViewItem	*item = m_view->currentItem();
+  bool state=(item!=0);
+  m_remove->setEnabled(state);
+  m_up->setEnabled((state && item->itemAbove() != 0));
+  m_down->setEnabled((state && item->itemBelow() != 0));
+  m_configure->setEnabled(state);
+}
+
 void KPFilterPage::slotAddClicked()
 {
 	if (m_filters.count() == 0)
@@ -117,6 +123,7 @@ void KPFilterPage::slotAddClicked()
 		m_activefilters.insert(filter->idName(),filter);
 		new QListViewItem(m_view,choice,filter->idName());
 	}
+	updateButton();
 }
 
 void KPFilterPage::slotRemoveClicked()
@@ -126,6 +133,7 @@ void KPFilterPage::slotRemoveClicked()
 		QString	idname = m_view->selectedItem()->text(1);
 		delete m_view->selectedItem();
 		m_activefilters.remove(idname);
+		updateButton();
 	}
 }
 
Index: kpfilterpage.h
===================================================================
RCS file: /home/kde/kdelibs/kdeprint/kpfilterpage.h,v
retrieving revision 1.1
diff -u -p -r1.1 kpfilterpage.h
--- kdelibs/kdeprint/kpfilterpage.h	2001/05/03 15:12:10	1.1
+++ kdelibs/kdeprint/kpfilterpage.h	2002/02/14 08:53:49
@@ -49,7 +49,7 @@ protected slots:
 
 protected:
 	KPrintFilter* currentFilter();
-
+	void updateButton();
 private:
 	KListView		*m_view;
 	QStringList		m_filters;	// <idname,description> pairs
cvs server: Diffing cups
cvs server: Diffing cups/cupsdconf
cvs server: Diffing ext
cvs server: Diffing filters
cvs server: Diffing lpd
cvs server: Diffing lpdunix
cvs server: Diffing management
cvs server: Diffing pics
cvs server: Diffing rlpr
cvs server: Diffing slave
cvs server: Diffing slave/mimetypes
cvs server: Diffing slave/pics
cvs server: Diffing slave/templates
cvs server: Diffing tests
