Index: libkscreensaver/kscreensaver.cpp
===================================================================
--- libkscreensaver/kscreensaver.cpp	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ libkscreensaver/kscreensaver.cpp	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -65,6 +65,7 @@
     if ( w == 0 ) w = 600;
     if ( h == 0 ) h = 420;
     resize( w, h );
+    KApplication::sendPostedEvents();
     show();
 }
 
@@ -76,10 +77,12 @@
 
 void KScreenSaver::embed( QWidget *w )
 {
+    KApplication::sendPostedEvents();
 #ifdef Q_WS_X11 //FIXME
     XReparentWindow(qt_xdisplay(), w->winId(), winId(), 0, 0);
 #endif
-    w->resize( width(), height() );
+    w->setGeometry( 0, 0, width(), height() );
+    KApplication::sendPostedEvents();
 }
 
 bool KScreenSaver::eventFilter( QObject *o, QEvent *e )
Index: kate/part/katesearch.cpp
===================================================================
--- kate/part/katesearch.cpp	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ kate/part/katesearch.cpp	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -130,6 +130,7 @@
   searchFlags.replace = false;
   searchFlags.finished = false;
   searchFlags.regExp = KateViewConfig::global()->searchFlags() & KFindDialog::RegularExpression;
+  searchFlags.useBackRefs = KateViewConfig::global()->searchFlags() & KReplaceDialog::BackReference;
 
   if ( searchFlags.selected )
   {
@@ -225,6 +226,7 @@
   searchFlags.replace = false;
   searchFlags.finished = false;
   searchFlags.regExp = KateViewConfig::global()->searchFlags() & KFindDialog::RegularExpression;
+  searchFlags.useBackRefs = KateViewConfig::global()->searchFlags() & KReplaceDialog::BackReference;
 
   searchFlags.backward = searchFlags.backward != back;
   searchFlags.fromBeginning = false;
Index: mimetypes/application/x-gettext.desktop
===================================================================
--- mimetypes/application/x-gettext.desktop	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ mimetypes/application/x-gettext.desktop	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -6,6 +6,7 @@
 Comment[az]=GNU Gettext İsmarış Kataloqu
 Comment[bg]=Съобщения за превод (GNU Gettext)
 Comment[bn]=গনিউ গেট-টেক্সট বার্তা ক্যাটালগ
+Comment[br]=Katalog ar c'hemennadoù GNU Gettext
 Comment[bs]=GNU Gettext katalog poruka
 Comment[ca]=Catàleg de missatges Gettext de GNU
 Comment[cs]=Katalog zpráv (Gettext)
Index: kioslave/ftp/ftp.protocol
===================================================================
--- kioslave/ftp/ftp.protocol	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ kioslave/ftp/ftp.protocol	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -10,6 +10,7 @@
 writing=true
 makedir=true
 deleting=true
+moving=true
 ProxiedBy=http
 Icon=ftp
 maxInstances=2
Index: kio/kio/kmimemagic.cpp
===================================================================
--- kio/kio/kmimemagic.cpp	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ kio/kio/kmimemagic.cpp	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -2262,6 +2262,15 @@
 			tmp += "src";
 		r->setMimeType(tmp);
 	}
+	else
+	if ( tmp == "application/x-sharedlib" )
+	{
+		if ( _filename.find( ".so" ) == -1 ) 
+		{
+			tmp = "application/x-executable";
+			r->setMimeType( tmp );
+		}
+	}
 }
 
 KMimeMagicResult *
Index: kio/misc/kpac/eventsrc
===================================================================
--- kio/misc/kpac/eventsrc	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ kio/misc/kpac/eventsrc	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -77,7 +77,7 @@
 Name[cy]=Sgript dirprwy annilys
 Name[da]=Ugyldigt proxy-script
 Name[de]=Ungültiges Proxy-Skript
-Name[el]=Άκυρο σενάριο διαμεσολαβητή
+Name[el]=Μη έγκυρο σενάριο διαμεσολαβητή
 Name[eo]=Nevalida prokurila skripto
 Name[es]=Procedimiento inválido del proxy
 Name[et]=Vigane proxy skript
@@ -137,7 +137,7 @@
 Comment[cy]=Mae'r sgript ffurfweddu'r dirprwy a chafodd ei lawrlwytho yn annilys
 Comment[da]=Det downloadede proxy-indstillingsscript er ugyldigt
 Comment[de]=Das heruntergeladene Proxy-Einrichtungsskript ist nicht verwendbar
-Comment[el]=Το κατεβασμένο σενάριο ρύθμισης διαμεσολαβητή είναι άκυρο
+Comment[el]=Το κατεβασμένο σενάριο ρύθμισης διαμεσολαβητή δεν είναι έγκυρο
 Comment[eo]=La ŝarĝita prokuragordoskript estas nevalida
 Comment[es]=El procedimiento descargado para la configuración del proxy es inválido
 Comment[et]=Alla laaditud proxy seadistuse skript on vigane
Index: khtml/khtmlview.cpp
===================================================================
--- khtml/khtmlview.cpp	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ khtml/khtmlview.cpp	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -2238,7 +2238,9 @@
             guard = node;
 	}
         // Set focus node on the document
+        QFocusEvent::setReason( QFocusEvent::Shortcut );
         m_part->xmlDocImpl()->setFocusNode(node);
+        QFocusEvent::resetReason();
         if( node != NULL && node->hasOneRef()) // deleted, only held by guard
             return true;
         emit m_part->nodeActivated(Node(node));
Index: khtml/html/html_formimpl.cpp
===================================================================
--- khtml/html/html_formimpl.cpp	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ khtml/html/html_formimpl.cpp	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -954,16 +954,15 @@
 
 	if (!evt->defaultHandled() && m_render && m_render->isWidget()) {
 	    // handle tabbing out, either from a single or repeated key event.
-	    // ### FIXME this is needlessly complicated by our internal KHTML_KEYPRESS_EVENT.
-	    // What is it good for? It should be renamed and only emitted in case of a repeat. -gg
-	    if ( evt->id() == EventImpl::KEYDOWN_EVENT || evt->id() == EventImpl::KHTML_KEYPRESS_EVENT ) {
+	    if ( evt->id() == EventImpl::KHTML_KEYPRESS_EVENT ) {
 	        QKeyEvent* const k = static_cast<TextEventImpl *>(evt)->qKeyEvent();
-	        if ( k && (k->key() == Qt::Key_Tab || k->key() == Qt::Key_BackTab) && 
-	             (evt->id() == EventImpl::KEYDOWN_EVENT || k->isAutoRepeat()) ) {
+	        if ( k && (k->key() == Qt::Key_Tab || k->key() == Qt::Key_BackTab) ) {
 		    QWidget* const widget = static_cast<RenderWidget*>(m_render)->widget();
+		    QFocusEvent::setReason( k->key() == Qt::Key_Tab ? QFocusEvent::Tab : QFocusEvent::Backtab );
 		    if (widget)
                         static_cast<FocusHandleWidget *>(widget)
 			    ->focusNextPrev(k->key() == Qt::Key_Tab);
+		    QFocusEvent::resetReason();
                     evt->setDefaultHandled();
 	        }
             }
Index: khtml/ecma/kjs_events.cpp
===================================================================
--- khtml/ecma/kjs_events.cpp	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ khtml/ecma/kjs_events.cpp	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -635,6 +635,9 @@
     return Boolean(static_cast<DOM::MouseEvent>(event).metaKey());
   case Button:
   {
+    if ( exec->interpreter()->compatMode() == Interpreter::NetscapeCompat ) {
+        return Number(static_cast<DOM::MouseEvent>(event).button());
+    }
     // Tricky. The DOM (and khtml) use 0 for LMB, 1 for MMB and 2 for RMB
     // but MSIE uses 1=LMB, 2=RMB, 4=MMB, as a bitfield
     int domButton = static_cast<DOM::MouseEvent>(event).button();
Index: khtml/rendering/render_list.cpp
===================================================================
--- khtml/rendering/render_list.cpp	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ khtml/rendering/render_list.cpp	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -402,7 +402,7 @@
 
     m_markerWidth = m_width = 0;
 
-    if(m_listImage) {
+    if(m_listImage && !m_listImage->isErrorImage()) {
         m_markerWidth = m_listImage->pixmap().width() + 5;
         if (listPositionInside())
             m_width = m_markerWidth;
Index: khtml/rendering/render_object.cpp
===================================================================
--- khtml/rendering/render_object.cpp	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ khtml/rendering/render_object.cpp	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -1697,21 +1697,22 @@
 {
     // vertical align for table cells has a different meaning
     int vpos = 0;
-    if ( !isTableCell() ) {
+    if ( !isTableCell() && isInline() ) {
         EVerticalAlign va = style()->verticalAlign();
         if ( va == TOP ) {
             vpos = PositionTop;
         } else if ( va == BOTTOM ) {
             vpos = PositionBottom;
-        } else if ( va == LENGTH ) {
-            vpos = -style()->verticalAlignLength().width( lineHeight( firstLine ) );
         } else {
             if (!ref) ref = parent();
-            bool checkParent = ref->isInline() && !ref->isReplacedBlock();
+            bool checkParent = ref->isInline() && !ref->isReplacedBlock() && 
+                               !( ref->style()->verticalAlign() == TOP || ref->style()->verticalAlign() == BOTTOM );
             vpos = checkParent ? ref->verticalPositionHint( firstLine ) : 0;
             // don't allow elements nested inside text-top to have a different valignment.
             if ( va == BASELINE )
                 return vpos;
+            else if ( va == LENGTH )
+                return vpos - style()->verticalAlignLength().width( lineHeight( firstLine ) );
 
             const QFont &f = ref->font( firstLine );
             int fontheight = ref->lineHeight( firstLine );
Index: khtml/rendering/render_replaced.cpp
===================================================================
--- khtml/rendering/render_replaced.cpp	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ khtml/rendering/render_replaced.cpp	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -513,7 +513,7 @@
             QWidget* const w = ::qt_cast<QWidget *>(it.current());
 	    if ( w && !w->isTopLevel() && !w->isHidden()) {
 	        QRect r2 = w->geometry();
-	        blit.subtract( r2 );
+	        blit -= r2;
 	        r2 = r2.intersect( r );
 	        r2.moveBy(-w->x(), -w->y());
 	        cr.append(r2);
Index: khtml/java/kjava.jar
===================================================================
Cannot display: file marked as a binary type.
svn:mime-type = application/x-jar
Index: kresources/kresources.desktop
===================================================================
--- kresources/kresources.desktop	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ kresources/kresources.desktop	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -14,6 +14,7 @@
 Name[af]=KDE Hulpbronne
 Name[bg]=Системни ресурси
 Name[bn]=কে.ডি.ই রিসোর্সসমূহ
+Name[br]=Danvezioù KDE
 Name[bs]=KDE resursi
 Name[ca]=Recursos KDE
 Name[cs]=Zdroje KDE
@@ -56,6 +57,7 @@
 Comment[be]=Настроіць рэсурсы KDE
 Comment[bg]=Настройки на системните ресурсите
 Comment[bn]=কে.ডি.ই. রিসোর্সসমূহ কনফিগার করো
+Comment[br]=Kefluniañ danvezioù KDE
 Comment[bs]=Podesite KDE resurse
 Comment[ca]=Configura els recursos KDE
 Comment[cs]=Nastavení zdrojů KDE
Index: Makefile.cvs
===================================================================
--- Makefile.cvs	(.../tags/KDE/3.4.2/kdelibs)	(revision 0)
+++ Makefile.cvs	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -0,0 +1,15 @@
+
+all:
+	@echo "This Makefile is only for the CVS repository"
+	@echo ""
+	@if test ! -d admin; then \
+	   echo "Please recheckout this module!" ;\
+	   echo "for cvs: use checkout once and after that update again" ;\
+	   echo "for cvsup: checkout kde-common from cvsup and" ;\
+	   echo "   link kde-common/admin to ./admin" ;\
+	   exit 1 ;\
+	fi
+	$(MAKE) -f admin/Makefile.common cvs
+
+.SILENT:
+.PHONY: all

Property changes on: Makefile.cvs
___________________________________________________________________
Name: svn:keywords
   + Author Date Id Revision

Index: configure.in.in
===================================================================
--- configure.in.in	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ configure.in.in	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -26,7 +26,6 @@
 
 dnl Checks for programs.
 AC_CHECK_COMPILERS
-KDE_ENABLE_HIDDEN_VISIBILITY
 AC_LIBLTDL_CONVENIENCE
 
 AC_ENABLE_SHARED(yes)
@@ -59,6 +58,8 @@
 
 dnl Checks for libraries etc
 AC_BASE_PATH_KDE([don't test]) dnl kdelibs is a special case
+KDE_ENABLE_HIDDEN_VISIBILITY
+
 AC_CREATE_KFSSTND(default)
 AC_SUBST_KFSSTND
 
Index: kdeui/ksyntaxhighlighter.cpp
===================================================================
--- kdeui/ksyntaxhighlighter.cpp	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ kdeui/ksyntaxhighlighter.cpp	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -102,6 +102,7 @@
     int wordCount, errorCount;
     int checksRequested, checksDone;
     int disablePercentage;
+    int disableWordCount;
     bool completeRehighlightRequired;
     bool active, automatic, autoReady;
     bool globalConfig, spellReady;
@@ -293,6 +294,7 @@
     KConfigGroupSaver cs( config, "KSpell" );
     d->disablePercentage = config->readNumEntry( "KSpell_AsYouTypeDisablePercentage", 42 );
     d->disablePercentage = QMIN( d->disablePercentage, 101 );
+    d->disableWordCount = config->readNumEntry( "KSpell_AsYouTypeDisableWordCount", 100 );
 
     textEdit->installEventFilter( this );
     textEdit->viewport()->installEventFilter( this );
@@ -565,7 +567,7 @@
 
     if ( d->automatic ) {
 	// tme = Too many errors
-	bool tme = d->errorCount * 100 >= d->disablePercentage * d->wordCount;
+	bool tme = d->wordCount >= d->disableWordCount && d->errorCount * 100 >= d->disablePercentage * d->wordCount;
 	if ( d->active && tme )
 	    d->active = false;
 	else if ( !d->active && !tme )
Index: kdeui/klistview.cpp
===================================================================
--- kdeui/klistview.cpp	(.../tags/KDE/3.4.2/kdelibs)	(revision 442272)
+++ kdeui/klistview.cpp	(.../branches/KDE/3.4/kdelibs)	(revision 442272)
@@ -369,10 +369,10 @@
         QListViewItem *i=item;
         col=0;
         item=0;
-        hide(); // will call focusOutEvent, that's why we set item=0 before
+        p->setFocus();// will call focusOutEvent, that's why we set item=0 before
+        hide();
         if (commit)
             emit done(i,c);
-        i->listView()->setFocus();
     }
 }
 

Property changes on: .
___________________________________________________________________
Name: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.4/kde-common/admin


