--- kdelibs-030718/kdecore/kglobalsettings.cpp~	2003-07-18 00:31:01.000000000 +0200
+++ kdelibs-030718/kdecore/kglobalsettings.cpp	2003-07-18 01:25:04.000000000 +0200
@@ -282,8 +282,8 @@
         return *_generalFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _generalFont = new QFont("helvetica", 12);
-    _generalFont->setPointSize(12);
+    _generalFont = new QFont("helvetica", 9);
+    _generalFont->setPointSize(9);
     _generalFont->setStyleHint(QFont::SansSerif);
 
     KConfig *c = KGlobal::config();
@@ -299,8 +299,8 @@
         return *_fixedFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _fixedFont = new QFont("courier", 12);
-    _fixedFont->setPointSize(12);
+    _fixedFont = new QFont("courier", 9);
+    _fixedFont->setPointSize(9);
     _fixedFont->setStyleHint(QFont::TypeWriter);
 
     KConfig *c = KGlobal::config();
@@ -316,8 +316,8 @@
         return *_toolBarFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _toolBarFont = new QFont("helvetica", 10);
-    _toolBarFont->setPointSize(10);
+    _toolBarFont = new QFont("helvetica", 8);
+    _toolBarFont->setPointSize(8);
     _toolBarFont->setStyleHint(QFont::SansSerif);
 
     KConfig *c = KGlobal::config();
@@ -333,8 +333,8 @@
         return *_menuFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _menuFont = new QFont("helvetica", 12);
-    _menuFont->setPointSize(12);
+    _menuFont = new QFont("helvetica", 9);
+    _menuFont->setPointSize(9);
     _menuFont->setStyleHint(QFont::SansSerif);
 
     KConfig *c = KGlobal::config();
