diff -Nur kdelibs-2.0.org/kdecore/kstddirs.cpp kdelibs-2.0/kdecore/kstddirs.cpp
--- kdelibs-2.0.org/kdecore/kstddirs.cpp	Thu Sep 21 20:01:04 2000
+++ kdelibs-2.0/kdecore/kstddirs.cpp	Mon Nov 27 22:21:41 2000
@@ -588,9 +588,9 @@
     if (!strcmp(type, "data"))
 	return "share/apps/";
     if (!strcmp(type, "html"))
-	return "share/doc/HTML/";
+	return "../share/doc/kde/HTML/";
     if (!strcmp(type, "icon"))
-	return "share/icons/";
+	return "share/pixmaps/";
     if (!strcmp(type, "config"))
 	return "share/config/";
     if (!strcmp(type, "pixmap"))
