--- kdelibs-030317/kded/vfolder_menu.cpp~	Sat Mar 15 00:31:36 2003
+++ kdelibs-030317/kded/vfolder_menu.cpp	Sun Mar 23 23:20:28 2003
@@ -569,7 +569,8 @@
 
    if (result.isEmpty() && !m_docInfo.baseDir.startsWith("/"))
    {
-      result = locate("apps", m_docInfo.baseDir+"default_"+fileName);
+    //  result = locate("apps", m_docInfo.baseDir+"default_"+fileName);
+	result = "/etc/X11/desktop/menus/" + fileName;
    }
    return result;
 }
