--- kdelibs-3.0.98/kdecore/kiconloader.cpp.orig	Mon Nov  4 13:36:39 2002
+++ kdelibs-3.0.98/kdecore/kiconloader.cpp	Tue Nov  5 23:19:53 2002
@@ -609,8 +609,14 @@
                 // Try "User" icon too. Some apps expect this.
                 if (!name.isEmpty())
                     pix = loadIcon(name, KIcon::User, size, state, path_store, true);
-                if (!pix.isNull() || canReturnNull)
-                    return pix;
+                if (!pix.isNull() || canReturnNull) {
+                    if ((group == KIcon::Small) && (pix.width() > 20 || pix.height() > 20)) {
+                        QImage tmp = pix.convertToImage();
+                        tmp = tmp.smoothScale(20, 20);
+                        pix.convertFromImage(tmp);
+                    }
+		    return pix;
+		}
 
                 icon = findMatchingIcon(str_unknown, size);
                 if (!icon.isValid())
