Index: kate/part/katesupercursor.cpp
===================================================================
--- kate/part/katesupercursor.cpp	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ kate/part/katesupercursor.cpp	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -186,7 +186,7 @@
 {
   if (newLine)
   {
-    if (m_line > int(line) || (m_line == int(line) && m_col > int(col)))
+    if (m_line > int(line) || (m_line == int(line) && m_col >= int(col)))
     {
       if(m_line == int(line))
          m_col -= col;
Index: mimetypes/audio/x-oggflac.desktop
===================================================================
--- mimetypes/audio/x-oggflac.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-oggflac.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment[af]=Ogg FLAC Klank
 Comment[ar]=ملف Ogg FLAC صوتي
 Comment[be]=Гукавы файл Ogg FLAC
-Comment[bg]=Аудио файл Ogg FLAC
+Comment[bg]=Ogg FLAC аудио файл
 Comment[bn]=অগ এফ-এল-এ-সি (FLAC) অডিও
 Comment[br]=Klevet Ogg FLAC
 Comment[bs]=Ogg FLAC audio
Index: mimetypes/audio/vnd.rn-realaudio.desktop
===================================================================
--- mimetypes/audio/vnd.rn-realaudio.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/vnd.rn-realaudio.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -10,7 +10,7 @@
 Comment[ar]=ملف Real Audio
 Comment[az]=RealAudio Faylı
 Comment[be]=Гукавы файл RealAudio
-Comment[bg]=Аудио файл RealAudio
+Comment[bg]=RealAudio аудио файл
 Comment[bn]=রিয়েলঅডিও ফাইল
 Comment[br]=Restr RealAudio
 Comment[bs]=RealAudio datoteka
Index: mimetypes/audio/x-vorbis.desktop
===================================================================
--- mimetypes/audio/x-vorbis.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-vorbis.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment[af]=Ogg Vorbis Klank
 Comment[ar]=ملف Ogg Vorbis صوتي
 Comment[be]=Гукавы файл Ogg Vorbis
-Comment[bg]=Аудио файл Ogg Vorbis
+Comment[bg]=Ogg Vorbis аудио файл
 Comment[bn]=অগ ভরবিস অডিও
 Comment[br]=Klevet Ogg Vorbis
 Comment[bs]=Ogg Vorbis audio
Index: mimetypes/audio/x-flac.desktop
===================================================================
--- mimetypes/audio/x-flac.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-flac.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -5,7 +5,7 @@
 Comment[af]=FLAC Klank
 Comment[ar]=ملف FLAC صوتي
 Comment[be]=Гукавы файл FLAC
-Comment[bg]=Аудио файл FLAC
+Comment[bg]=FLAC аудио файл
 Comment[bn]=এফ-এল-এ-সি (FLAC) অডিও
 Comment[br]=Klevet FLAC
 Comment[bs]=FLAC audio
Index: mimetypes/audio/x-aiff.desktop
===================================================================
--- mimetypes/audio/x-aiff.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-aiff.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -5,7 +5,7 @@
 Comment[af]=Aiff/Amiga Klank
 Comment[ar]=صوت AIFF/Amiga
 Comment[be]=Гукавы файл AIFF/Amiga
-Comment[bg]=Аудио файл AIFF/Amiga
+Comment[bg]=AIFF/Amiga аудио файл
 Comment[bn]=এ-আই-এফ-এফ (AIFF)/অ্যামিগা অডিও
 Comment[br]=Klevet AIFF/Amiga
 Comment[bs]=AIFF/Amiga audio
Index: mimetypes/audio/mpegurl.desktop
===================================================================
--- mimetypes/audio/mpegurl.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/mpegurl.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -10,7 +10,7 @@
 Comment[ar]=صوت انسيابي من MPEG الطبقة 3
 Comment[az]=Axıcı MPEG Layer 3 Audio
 Comment[be]=Плынны гук MPEG Layer 3
-Comment[bg]=Поточен файл MP3
+Comment[bg]=Поточен MP3 файл
 Comment[bn]=স্ট্রিমিং এমপেগ লেয়ার-৩ অডিও
 Comment[br]=Klevet lanv MPEG gwelead 3
 Comment[bs]=Streaming MPEG Layer 3 audio
Index: mimetypes/audio/x-musepack.desktop
===================================================================
--- mimetypes/audio/x-musepack.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-musepack.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -7,7 +7,7 @@
 Comment=Musepack Audio
 Comment[af]=Musepack Klank
 Comment[be]=Гукавы файл Musepack
-Comment[bg]=Аудио файл Musepack
+Comment[bg]=Musepack аудио файл
 Comment[bn]=মিউস-প্যাক অডিও
 Comment[br]=Klevet Musepack
 Comment[bs]=Musepack audio
Index: mimetypes/audio/x-mpegurl.desktop
===================================================================
--- mimetypes/audio/x-mpegurl.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-mpegurl.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment[ar]=صوت انسيابي من MPEG الطبقة 3
 Comment[az]=Axıcı MPEG Layer 3 Audio
 Comment[be]=Плынны гук MPEG Layer 3
-Comment[bg]=Поточен файл MP3
+Comment[bg]=Поточен MP3 файл
 Comment[bn]=স্ট্রিমিং এমপেগ লেয়ার-৩ অডিও
 Comment[br]=Klevet lanv MPEG gwelead 3
 Comment[bs]=Streaming MPEG Layer 3 audio
Index: mimetypes/audio/basic.desktop
===================================================================
--- mimetypes/audio/basic.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/basic.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -5,7 +5,7 @@
 Comment[af]=Ulaw (Sun) Audio
 Comment[ar]=صوت ULAW (صن) 
 Comment[be]=Гукавы файл ULAW (Sun)
-Comment[bg]=Аудио файл ULAW (Sun)
+Comment[bg]=ULAW (Sun) аудио файл
 Comment[bn]=ULAW (সান) অডিও
 Comment[br]=Klevet ULAW (Sun)
 Comment[bs]=ULAW (Sun) audio
Index: mimetypes/audio/prs.sid.desktop
===================================================================
--- mimetypes/audio/prs.sid.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/prs.sid.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -7,7 +7,7 @@
 Comment=C64 SID Music
 Comment[af]=C64 SID Musiek
 Comment[be]=Музычны файл C64 SID
-Comment[bg]=Аудио файл C64 SID
+Comment[bg]=C64 SID аудио файл
 Comment[bn]=C64 SID মিউসিক
 Comment[br]=Klevet C64 SID
 Comment[bs]=C64 SID muzika
Index: mimetypes/audio/x-mod.desktop
===================================================================
--- mimetypes/audio/x-mod.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-mod.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -7,7 +7,7 @@
 Comment=Amiga Soundtracker Audio
 Comment[ar]=صوت Amiga Soundtracker
 Comment[be]=Гукавы файл Amiga Soundtracker
-Comment[bg]=Аудио файл Amiga Soundtracker
+Comment[bg]=Amiga Soundtracker аудио файл
 Comment[bn]=অ্যামিগা সাউণ্ডট্র্যাকার অডিও
 Comment[br]=Klevet Amiga soundtracker
 Comment[bs]=Amiga Soundtracker audio
Index: mimetypes/audio/aac.desktop
===================================================================
--- mimetypes/audio/aac.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/aac.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -9,7 +9,7 @@
 Comment[ar]=ملف AAC صوتي
 Comment[az]=AAC Səsi
 Comment[be]=Гукавы файл AAC
-Comment[bg]=Аудио файл AAC
+Comment[bg]=AAC аудио файл
 Comment[bn]= এ-এ-সি অডিও
 Comment[br]=Klevet AAC
 Comment[bs]=AAC zvuk
Index: mimetypes/audio/x-pn-realaudio.desktop
===================================================================
--- mimetypes/audio/x-pn-realaudio.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-pn-realaudio.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment[ar]=ملف Real Audio
 Comment[az]=RealAudio Faylı
 Comment[be]=Гукавы файл RealAudio
-Comment[bg]=Аудио файл RealAudio
+Comment[bg]=RealAudio аудио файл
 Comment[bn]=রিয়েলঅডিও ফাইল
 Comment[br]=Restr RealAudio
 Comment[bs]=RealAudio datoteka
Index: mimetypes/audio/x-matroska.desktop
===================================================================
--- mimetypes/audio/x-matroska.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-matroska.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment[af]=Matroska Klank
 Comment[ar]=ملف ماتروسكا صوتي
 Comment[be]=Гукавы файл Matroska
-Comment[bg]=Аудио файл Matroska
+Comment[bg]=Matroska аудио файл
 Comment[bn]=ম্যাট্রস্কা অডিও
 Comment[br]=Klevet Matroska
 Comment[bs]=Matroska audio
Index: mimetypes/audio/x-midi.desktop
===================================================================
--- mimetypes/audio/x-midi.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-midi.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment[af]=MIDI Klank
 Comment[ar]=صوت MIDI
 Comment[be]=Гукавы файл MIDI
-Comment[bg]=Аудио файл MIDI
+Comment[bg]=MIDI аудио файл
 Comment[bn]=মিডি অডিও
 Comment[br]=Klevet MIDI
 Comment[bs]=MIDI audio
Index: mimetypes/audio/mpeg.desktop
===================================================================
--- mimetypes/audio/mpeg.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/mpeg.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -9,7 +9,7 @@
 Comment=MPEG Audio
 Comment[af]=MPEG Klank
 Comment[be]=Гукавы файл MPEG
-Comment[bg]=Аудио файл MPEG
+Comment[bg]=MPEG аудио файл
 Comment[bn]=এমপেগ অডিও
 Comment[br]=Klevet MPEG
 Comment[bs]=MPEG audio
Index: mimetypes/audio/x-adpcm.desktop
===================================================================
--- mimetypes/audio/x-adpcm.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-adpcm.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -5,7 +5,7 @@
 Comment[af]=PCM Klank
 Comment[ar]=صوت PCM
 Comment[be]=Гукавы файл PCM
-Comment[bg]=Аудио файл PCM
+Comment[bg]=PCM аудио файл
 Comment[bn]=পি-সি-এম অডিও
 Comment[br]=Klevet PCM
 Comment[bs]=PCM audio
Index: mimetypes/audio/x-speex.desktop
===================================================================
--- mimetypes/audio/x-speex.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-speex.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment[af]=Speex Klank
 Comment[ar]=ملف Speex صوتي
 Comment[be]=Гукавы файл Speex
-Comment[bg]=Аудио файл Speex
+Comment[bg]=Speex аудио файл
 Comment[bn]=স্পীক্স অডিও
 Comment[br]=Klevet Speex
 Comment[bs]=Speex audio
Index: mimetypes/audio/mp4.desktop
===================================================================
--- mimetypes/audio/mp4.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/mp4.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment[af]=MPEG-4 Klank
 Comment[ar]=صوت MPEG-4
 Comment[be]=Гукавы файл MPEG-4
-Comment[bg]=Аудио файл MPEG-4
+Comment[bg]=MPEG-4 аудио файл
 Comment[bn]=এমপেগ-৪ অডিও
 Comment[br]=Klevet MPEG-4
 Comment[bs]=MPEG-4 audio
Index: mimetypes/audio/x-wav.desktop
===================================================================
--- mimetypes/audio/x-wav.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-wav.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment[af]=Wav Audio
 Comment[ar]=صوت WAV
 Comment[be]=Гукавы файл WAV
-Comment[bg]=Аудио файл WAV
+Comment[bg]=WAV аудио файл
 Comment[bn]=ওয়েভ অডিও
 Comment[br]=Klevet WAV
 Comment[bs]=WAV audio
Index: mimetypes/audio/x-mp2.desktop
===================================================================
--- mimetypes/audio/x-mp2.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-mp2.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment=MPEG Layer 2 Audio
 Comment[af]=MPEG Vlak 2 Klank
 Comment[be]=Гукавы файл MPEG Layer 2
-Comment[bg]=Аудио файл MP2
+Comment[bg]=MP2 аудио файл
 Comment[bn]=এমপেগ লেয়ার ২ অডিও
 Comment[br]=Klevet MPEG gwelead 2
 Comment[bs]=MPEG Layer 2 audio
Index: mimetypes/audio/vorbis.desktop
===================================================================
--- mimetypes/audio/vorbis.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/vorbis.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment[af]=Ogg Vorbis Klank
 Comment[ar]=ملف Ogg Vorbis صوتي
 Comment[be]=Гукавы файл Ogg Vorbis
-Comment[bg]=Аудио файл Ogg Vorbis
+Comment[bg]=Ogg Vorbis аудио файл
 Comment[bn]=অগ ভরবিস অডিও
 Comment[br]=Klevet Ogg Vorbis
 Comment[bs]=Ogg Vorbis audio
Index: mimetypes/audio/x-mp3.desktop
===================================================================
--- mimetypes/audio/x-mp3.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-mp3.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -9,7 +9,7 @@
 Comment[af]=MPEG Vlak 3 Klank
 Comment[ar]=صوت MPEG الطبقة 3
 Comment[be]=Гукавы файл MPEG Layer 3
-Comment[bg]=Аудио файл MP3
+Comment[bg]=MP3 аудио файл
 Comment[bn]=এমপেগ লেয়ার ৩ অডিও
 Comment[br]=Klevet MPEG gwelead 3
 Comment[bs]=MPEG Layer 3 audio
Index: mimetypes/audio/x-ms-wma.desktop
===================================================================
--- mimetypes/audio/x-ms-wma.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/audio/x-ms-wma.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -7,7 +7,7 @@
 Comment=Windows Media Audio
 Comment[af]=Windows Media Klank
 Comment[be]=Гукавы файл Windows Media
-Comment[bg]=Аудио файл Windows Media
+Comment[bg]=Windows Media аудио файл
 Comment[bn]=উইন্ডোস মিডিয়া অডিও
 Comment[br]=Klevet Windows Media
 Comment[bs]=Windows Media audio
Index: mimetypes/text/vnd.abc.desktop
===================================================================
--- mimetypes/text/vnd.abc.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/text/vnd.abc.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -3,7 +3,7 @@
 Comment=ABC Music File
 Comment[af]=ABC Musiek Lêer
 Comment[be]=Музычны файл ABC
-Comment[bg]=Аудио файл ABC
+Comment[bg]=ABC аудио файл
 Comment[bn]=এ-বি-সি মিউজিক ফাইল
 Comment[br]=Restr ar sonerezh ABC
 Comment[bs]=ABC muzička datoteka
Index: mimetypes/video/x-ogm.desktop
===================================================================
--- mimetypes/video/x-ogm.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/video/x-ogm.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment=Ogg/Ogm Video
 Comment[ar]=ملف مرئي Ogg/Ogm
 Comment[be]=Відэафайл Ogg/Ogm
-Comment[bg]=Видео файл Ogg/Ogm
+Comment[bg]=Ogg/Ogm видео файл
 Comment[bn]=অগ/ও-জি-এম ভিডিও
 Comment[br]=Video Ogg/Ogm
 Comment[bs]=Ogg/Ogm video
Index: mimetypes/video/x-theora.desktop
===================================================================
--- mimetypes/video/x-theora.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/video/x-theora.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -7,7 +7,7 @@
 Comment=Theora Video
 Comment[ar]=ملف فيديو Theora
 Comment[be]=Відэафайл Theora
-Comment[bg]=Видео файл Theora
+Comment[bg]=Theora видео файл
 Comment[bn]=থিওরা ভিডিও
 Comment[br]=Video Theora
 Comment[bs]=Theora video
Index: mimetypes/video/mpeg.desktop
===================================================================
--- mimetypes/video/mpeg.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/video/mpeg.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -9,7 +9,7 @@
 Comment[ar]=ملف مرئي MPEG
 Comment[az]=MPEG Video Faylı
 Comment[be]=Відэафайл MPEG
-Comment[bg]=Видео файл MPEG
+Comment[bg]=MPEG видео файл
 Comment[bn]=এমপেগ ভিডিও
 Comment[br]=Video MPEG
 Comment[ca]=Vídeo MPEG
Index: mimetypes/video/quicktime.desktop
===================================================================
--- mimetypes/video/quicktime.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/video/quicktime.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -7,7 +7,7 @@
 Comment=Quicktime Video
 Comment[ar]=ملف مرئي Quicktime
 Comment[be]=Відэафайл Quicktime
-Comment[bg]=Видео файл Quicktime
+Comment[bg]=Quicktime видео файл
 Comment[bn]=কুইকটাইম ভিডিও
 Comment[br]=Video Quicktime
 Comment[ca]=Vídeo Quicktime
Index: mimetypes/video/x-ms-wmv.desktop
===================================================================
--- mimetypes/video/x-ms-wmv.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/video/x-ms-wmv.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -5,7 +5,7 @@
 Comment[ar]=ملف مرئي Windows Media
 Comment[az]=Windows Mediya Video Faylı
 Comment[be]=Відэафайл Windows Media
-Comment[bg]=Видео файл Windows Media
+Comment[bg]=Windows Media видео файл
 Comment[bn]=উইন্ডোস মিডিয়া ভিডিও
 Comment[br]=Video Windows Media
 Comment[bs]=Windows Media video
Index: mimetypes/video/x-msvideo.desktop
===================================================================
--- mimetypes/video/x-msvideo.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/video/x-msvideo.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -8,7 +8,7 @@
 Comment[af]=Microsoft Avi Video
 Comment[ar]=مرئي Microsoft AVI
 Comment[be]=Відэафайл Microsoft AVI
-Comment[bg]=Видео файл AVI
+Comment[bg]=AVI видео файл
 Comment[bn]=মাইক্রোসফট এ-ভি-আই ভিডিও
 Comment[br]=Video Microsoft AVI
 Comment[bs]=Microsoft AVI video
Index: mimetypes/video/vnd.rn-realvideo.desktop
===================================================================
--- mimetypes/video/vnd.rn-realvideo.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/video/vnd.rn-realvideo.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -9,7 +9,7 @@
 Comment[ar]=ملف RealVideo
 Comment[az]=RealVideo Faylı
 Comment[be]=Відэафайл RealVideo
-Comment[bg]=Видео файл RealVideo
+Comment[bg]=RealVideo файл
 Comment[bn]=রিয়েলভিডিও ফাইল
 Comment[br]=Restr RealVideo
 Comment[bs]=RealVideo datoteka
Index: mimetypes/video/avi.desktop
===================================================================
--- mimetypes/video/avi.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/video/avi.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -9,7 +9,7 @@
 Comment[af]=Microsoft Avi Video
 Comment[ar]=مرئي Microsoft AVI
 Comment[be]=Відэафайл Microsoft AVI
-Comment[bg]=Видео файл AVI
+Comment[bg]=AVI видео файл
 Comment[bn]=মাইক্রোসফট এ-ভি-আই ভিডিও
 Comment[br]=Video Microsoft AVI
 Comment[bs]=Microsoft AVI video
Index: mimetypes/video/mp4.desktop
===================================================================
--- mimetypes/video/mp4.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/video/mp4.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -7,7 +7,7 @@
 Comment=MPEG-4 Video
 Comment[ar]=ملف مرئي MPEG-4
 Comment[be]=Відэафайл MPEG-4
-Comment[bg]=Видео файл MPEG-4
+Comment[bg]=MPEG-4 видео файл
 Comment[bn]=এমপেগ-৪ ভিডিও
 Comment[br]=Video MPEG-4
 Comment[bs]=MPEG-4 video
Index: mimetypes/video/x-flic.desktop
===================================================================
--- mimetypes/video/x-flic.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/video/x-flic.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -9,7 +9,7 @@
 Comment[ar]=ملف FLIC لأوتوديسك
 Comment[az]=Autodesk FLIC Faylı
 Comment[be]=Файл Autodesk FLIC
-Comment[bg]=Видео файл Autodesk's FLIC
+Comment[bg]=Autodesk's FLIC видео файл
 Comment[bn]=অটোডেস্কের এফ-এল-আই-সি (FLIC) ফাইল
 Comment[br]=Restr FLIC Autodesk
 Comment[bs]=Autodesk FLIC datoteke
Index: mimetypes/video/x-matroska.desktop
===================================================================
--- mimetypes/video/x-matroska.desktop	(.../tags/KDE/3.5.9/kdelibs)	(wersja 781012)
+++ mimetypes/video/x-matroska.desktop	(.../branches/KDE/3.5/kdelibs)	(wersja 781012)
@@ -7,7 +7,7 @@
 Comment=Matroska Video
 Comment[ar]=مرئي Matroska
 Comment[be]=Відэафайл Matroska
-Comment[bg]=Видео файл Matroska
+Comment[bg]=Matroska видео файл
 Comment[bn]=মাত্রোস্কা ভিডিও
 Comment[br]=Video Matroska
 Comment[bs]=Matroska video

Zmiany atrybutów dla: .
___________________________________________________________________
Nazwa: svn:mergeinfo
   - 
Nazwa: svn:externals
   + admin https://svn.kde.org/home/kde/branches/KDE/3.5/kde-common/admin


